From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/levelgen/structure/structures/OceanRuinPieces.java b/net/minecraft/world/level/levelgen/structure/structures/OceanRuinPieces.java
index 5e91817a99478a5da91b96f3f056240fd52d9516..922ff49039689dea8d9b939cc26fe3899a442779 100644
--- a/net/minecraft/world/level/levelgen/structure/structures/OceanRuinPieces.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/OceanRuinPieces.java
@@ -321,16 +321,22 @@ public class OceanRuinPieces {
             final String markerId, final BlockPos position, final ServerLevelAccessor level, final RandomSource random, final BoundingBox chunkBB
         ) {
             if ("chest".equals(markerId)) {
-                level.setBlock(
-                    position,
-                    Blocks.CHEST.defaultBlockState().setValue(ChestBlock.WATERLOGGED, level.getFluidState(position).is(FluidTags.WATER)),
-                    Block.UPDATE_CLIENTS
-                );
-                if (level.getBlockEntity(position) instanceof ChestBlockEntity chestBlockEntity) {
-                    chestBlockEntity.setLootTable(
-                        this.isLarge ? BuiltInLootTables.UNDERWATER_RUIN_BIG : BuiltInLootTables.UNDERWATER_RUIN_SMALL, random.nextLong()
-                    );
-                }
+                // CraftBukkit start - transform block to ensure loot table is accessible
+                // level.setBlock(
+                //     position,
+                //     Blocks.CHEST.defaultBlockState().setValue(ChestBlock.WATERLOGGED, level.getFluidState(position).is(FluidTags.WATER)),
+                //     Block.UPDATE_CLIENTS
+                // );
+                // if (level.getBlockEntity(position) instanceof ChestBlockEntity chestBlockEntity) {
+                //     chestBlockEntity.setLootTable(
+                //         this.isLarge ? BuiltInLootTables.UNDERWATER_RUIN_BIG : BuiltInLootTables.UNDERWATER_RUIN_SMALL, random.nextLong()
+                //     );
+                // }
+                org.bukkit.craftbukkit.block.CraftChest craftChest = (org.bukkit.craftbukkit.block.CraftChest) org.bukkit.craftbukkit.block.CraftBlockStates.getBlockState(level, position, Blocks.CHEST.defaultBlockState().setValue(ChestBlock.WATERLOGGED, level.getFluidState(position).is(FluidTags.WATER)), null);
+                craftChest.setSeed(random.nextLong());
+                craftChest.setLootTable(org.bukkit.craftbukkit.CraftLootTable.minecraftToBukkit(this.isLarge ? BuiltInLootTables.UNDERWATER_RUIN_BIG : BuiltInLootTables.UNDERWATER_RUIN_SMALL));
+                this.placeCraftBlockEntity(level, position, craftChest, Block.UPDATE_CLIENTS);
+                // CraftBukkit end
             } else if ("drowned".equals(markerId)) {
                 Drowned drowned = EntityTypes.DROWNED.create(level.getLevel(), EntitySpawnReason.STRUCTURE);
                 if (drowned != null) {
