From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/levelgen/structure/structures/StrongholdPieces.java b/net/minecraft/world/level/levelgen/structure/structures/StrongholdPieces.java
index 6dbe60e3eb2e1843b9b94ce08f6a258d6add56b1..e0cc438e2f9a6cddde9f55bcd74d8792ec225611 100644
--- a/net/minecraft/world/level/levelgen/structure/structures/StrongholdPieces.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/StrongholdPieces.java
@@ -900,10 +900,13 @@ public class StrongholdPieces {
                 BlockPos pos = this.getWorldPos(5, 3, 6);
                 if (chunkBB.isInside(pos)) {
                     this.hasPlacedSpawner = true;
-                    level.setBlock(pos, Blocks.SPAWNER.defaultBlockState(), Block.UPDATE_CLIENTS);
-                    if (level.getBlockEntity(pos) instanceof SpawnerBlockEntity spawner) {
-                        spawner.setEntityId(EntityTypes.SILVERFISH, random);
-                    }
+                    // CraftBukkit start
+                    // level.setBlock(pos, Blocks.SPAWNER.defaultBlockState(), Block.UPDATE_CLIENTS);
+                    // if (level.getBlockEntity(pos) instanceof SpawnerBlockEntity spawner) {
+                    //     spawner.setEntityId(EntityTypes.SILVERFISH, random);
+                    // }
+                    this.placeCraftSpawner(level, pos, org.bukkit.entity.EntityType.SILVERFISH, Block.UPDATE_CLIENTS);
+                    // CraftBukkit end
                 }
             }
         }
