From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/protocol/handshake/ClientIntentionPacket.java b/net/minecraft/network/protocol/handshake/ClientIntentionPacket.java
index bcc08811ec7a9709e5c847d5a335056411306f6f..d74e631ca39d7cdbe5e1482d8de9eebf75fe14e1 100644
--- a/net/minecraft/network/protocol/handshake/ClientIntentionPacket.java
+++ b/net/minecraft/network/protocol/handshake/ClientIntentionPacket.java
@@ -16,7 +16,7 @@ public record ClientIntentionPacket(int protocolVersion, String hostName, int po
     }
 
     private ClientIntentionPacket(final FriendlyByteBuf input) {
-        this(input.readVarInt(), input.readUtf(255), input.readUnsignedShort(), ClientIntent.byId(input.readVarInt()));
+        this(input.readVarInt(), input.readUtf(Short.MAX_VALUE), input.readUnsignedShort(), ClientIntent.byId(input.readVarInt())); // Spigot - increase max hostName length
     }
 
     private void write(final FriendlyByteBuf output) {
