From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/util/datafix/DataFixTypes.java b/net/minecraft/util/datafix/DataFixTypes.java
index d570fbb3528a7184a07110146566140ef90c436e..4168626dcff36763e5354fc165044d96ca661572 100644
--- a/net/minecraft/util/datafix/DataFixTypes.java
+++ b/net/minecraft/util/datafix/DataFixTypes.java
@@ -14,6 +14,7 @@ import net.minecraft.nbt.NbtOps;
 import net.minecraft.util.datafix.fixes.References;
 
 public enum DataFixTypes {
+    NONE(null), // Paper - add no-op fix for custom types until we actually need fixers for them
     LEVEL(References.LEVEL),
     LEVEL_SUMMARY(References.LIGHTWEIGHT_LEVEL),
     PLAYER(References.PLAYER),
@@ -76,6 +77,7 @@ public enum DataFixTypes {
     }
 
     public <T> Dynamic<T> update(final DataFixer fixerUpper, final Dynamic<T> input, final int fromVersion, final int toVersion) {
+        if (this.type == null) return input; // Paper - add no-op fix for custom types until we actually need fixers for them
         return fixerUpper.update(this.type, input, fromVersion, toVersion);
     }
 
