From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/com/mojang/math/OctahedralGroup.java b/com/mojang/math/OctahedralGroup.java
index aa80bcca7013cf000922b05a801f602c6bfccd33..da4e2fdcb8397b7bc7c8a73a8d0c708315dca30f 100644
--- a/com/mojang/math/OctahedralGroup.java
+++ b/com/mojang/math/OctahedralGroup.java
@@ -99,6 +99,14 @@ public enum OctahedralGroup implements StringRepresentable {
         .map(f -> Arrays.stream(values()).filter(s -> f.compose(s) == IDENTITY).findAny().get())
         .toArray(OctahedralGroup[]::new);
 
+    // Paper start - Avoid Lazy Initialization for Enum Fields
+    static {
+        for (OctahedralGroup octahedralGroup : values()) {
+            octahedralGroup.initializeRotationDirections();
+        }
+    }
+    // Paper end - Avoid Lazy Initialization for Enum Fields
+
     OctahedralGroup(final String name, final SymmetricGroup3 permutation, final boolean invertX, final boolean invertY, final boolean invertZ) {
         this.name = name;
         this.invertX = invertX;
@@ -139,7 +147,7 @@ public enum OctahedralGroup implements StringRepresentable {
         return this.name;
     }
 
-    public Direction rotate(final Direction direction) {
+    public void initializeRotationDirections() { // Paper - Avoid Lazy Initialization for Enum Fields
         if (this.rotatedDirections == null) {
             this.rotatedDirections = Util.makeEnumMap(Direction.class, facing -> {
                 Direction.Axis oldAxis = facing.getAxis();
@@ -150,6 +158,11 @@ public enum OctahedralGroup implements StringRepresentable {
             });
         }
 
+    // Paper start - Avoid Lazy Initialization for Enum Fields
+    }
+
+    public Direction rotate(Direction direction) {
+    // Paper end - Avoid Lazy Initialization for Enum Fields
         return this.rotatedDirections.get(direction);
     }
 
