From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/attributes/AttributeInstance.java b/net/minecraft/world/entity/ai/attributes/AttributeInstance.java
index b70d2f58a27d3b62f90a6c2449399669bb348d6e..82fb9b7092ee37bb4b46ad9a352df64821259ddc 100644
--- a/net/minecraft/world/entity/ai/attributes/AttributeInstance.java
+++ b/net/minecraft/world/entity/ai/attributes/AttributeInstance.java
@@ -149,20 +149,20 @@ public class AttributeInstance {
         double base = this.getBaseValue();
 
         for (AttributeModifier modifier : this.getModifiersOrEmpty(AttributeModifier.Operation.ADD_VALUE)) {
-            base += modifier.amount();
+            base += modifier.amount(); // Paper - destroy speed API - diff on change
         }
 
         double result = base;
 
         for (AttributeModifier modifier : this.getModifiersOrEmpty(AttributeModifier.Operation.ADD_MULTIPLIED_BASE)) {
-            result += base * modifier.amount();
+            result += base * modifier.amount(); // Paper - destroy speed API - diff on change
         }
 
         for (AttributeModifier modifier : this.getModifiersOrEmpty(AttributeModifier.Operation.ADD_MULTIPLIED_TOTAL)) {
-            result *= 1.0 + modifier.amount();
+            result *= 1.0 + modifier.amount(); // Paper - destroy speed API - diff on change
         }
 
-        return this.attribute.value().sanitizeValue(result);
+        return this.attribute.value().sanitizeValue(result); // Paper - destroy speed API - diff on change
     }
 
     private Collection<AttributeModifier> getModifiersOrEmpty(final AttributeModifier.Operation operation) {
