From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/permissions/PermissionProviderCheck.java b/net/minecraft/server/permissions/PermissionProviderCheck.java
index 00b314b4449d82c94f41bbe031c6260f5e874b96..e1b1331bb8a75f01420d7bc962ce414e79407c98 100644
--- a/net/minecraft/server/permissions/PermissionProviderCheck.java
+++ b/net/minecraft/server/permissions/PermissionProviderCheck.java
@@ -2,9 +2,21 @@ package net.minecraft.server.permissions;
 
 import java.util.function.Predicate;
 
-public record PermissionProviderCheck<T extends PermissionSetSupplier>(PermissionCheck test) implements Predicate<T> {
+public record PermissionProviderCheck<T extends PermissionSetSupplier>(PermissionCheck test, java.util.concurrent.atomic.AtomicReference<com.mojang.brigadier.tree.CommandNode<net.minecraft.commands.CommandSourceStack>> vanillaNode) implements Predicate<T> { // Paper - Vanilla Command permission checking
     @Override
     public boolean test(final T t) {
+        // Paper start - Vanilla Command permission checking
+        com.mojang.brigadier.tree.CommandNode<net.minecraft.commands.CommandSourceStack> currentCommand = vanillaNode.get();
+        if (currentCommand != null && t instanceof net.minecraft.commands.CommandSourceStack commandSourceStack && this.test instanceof net.minecraft.server.permissions.PermissionCheck.Require req) {
+            return commandSourceStack.hasPermission(req.permission(), org.bukkit.craftbukkit.command.VanillaCommandWrapper.getPermission(currentCommand));
+        }
+        // Paper end - Vanilla Command permission checking
         return this.test.check(t.permissions());
     }
+
+    // Paper start - Vanilla Command permission checking
+    public PermissionProviderCheck(final PermissionCheck test) {
+        this(test, new java.util.concurrent.atomic.AtomicReference<>(null));
+    }
+    // Paper end - Vanilla Command permission checking
 }
