From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/util/worldupdate/WorldUpgrader.java b/net/minecraft/util/worldupdate/WorldUpgrader.java
index 646f0906e6936c1135d422a6bc68e3aaf0cd5750..a855a47dbad1770c5409e6d6c5840b0cf2cf5d27 100644
--- a/net/minecraft/util/worldupdate/WorldUpgrader.java
+++ b/net/minecraft/util/worldupdate/WorldUpgrader.java
@@ -68,7 +68,7 @@ public class WorldUpgrader implements AutoCloseable {
 
     public static CompoundTag getDataFixContextTag(final Registry<LevelStem> dimensions, final ResourceKey<Level> dimension) {
         ChunkGenerator generator = dimensions.getValueOrThrow(Registries.levelToLevelStem(dimension)).generator();
-        return ChunkMap.getChunkDataFixContextTag(dimension, generator.getTypeNameForDataFixer());
+        return ChunkMap.getChunkDataFixContextTag(dimension, generator.getTypeNameForDataFixer(), Registries.levelToLevelStem(dimension)); // Paper - pass level stem
     }
 
     public static boolean verifyChunkPosAndEraseCache(final ChunkPos pos, final CompoundTag upgradedTag) {
