From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/util/SpawnUtil.java b/net/minecraft/util/SpawnUtil.java
index 4bb099a49b228d912832c128d798c1407090ec4f..6d46d844c63f51b3c056417cca79d984f64eb6e0 100644
--- a/net/minecraft/util/SpawnUtil.java
+++ b/net/minecraft/util/SpawnUtil.java
@@ -16,6 +16,7 @@ import net.minecraft.world.level.block.StainedGlassPaneBlock;
 import net.minecraft.world.level.block.state.BlockState;
 
 public class SpawnUtil {
+
     public static <T extends Mob> Optional<T> trySpawnMob(
         final EntityType<T> entityType,
         final EntitySpawnReason spawnReason,
@@ -27,6 +28,24 @@ public class SpawnUtil {
         final SpawnUtil.Strategy strategy,
         final boolean checkCollisions
     ) {
+        // CraftBukkit start
+        return trySpawnMob(entityType, spawnReason, level, start, spawnAttempts, spawnRangeXZ, spawnRangeY, strategy, checkCollisions, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.DEFAULT, null); // Paper - pre creature spawn event
+    }
+
+    public static <T extends Mob> Optional<T> trySpawnMob(
+        final EntityType<T> entityType,
+        final EntitySpawnReason spawnReason,
+        final ServerLevel level,
+        final BlockPos start,
+        final int spawnAttempts,
+        final int spawnRangeXZ,
+        final int spawnRangeY,
+        final SpawnUtil.Strategy strategy,
+        final boolean checkCollisions,
+        final org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason reason,
+        final @org.jspecify.annotations.Nullable Runnable onAbort // Paper - pre creature spawn event
+    ) {
+        // CraftBukkit end
         BlockPos.MutableBlockPos searchPos = start.mutable();
         RandomSource random = level.getRandom();
 
@@ -37,15 +56,32 @@ public class SpawnUtil {
             if (level.getWorldBorder().isWithinBounds(searchPos)
                 && moveToPossibleSpawnPosition(level, spawnRangeY, searchPos, strategy)
                 && (!checkCollisions || level.noCollision(entityType.getSpawnAABB(searchPos.getX() + 0.5, searchPos.getY(), searchPos.getZ() + 0.5)))) {
+                // Paper start - PreCreatureSpawnEvent
+                final com.destroystokyo.paper.event.entity.PreCreatureSpawnEvent event = new com.destroystokyo.paper.event.entity.PreCreatureSpawnEvent(
+                    org.bukkit.craftbukkit.util.CraftLocation.toBukkit(start, level),
+                    org.bukkit.craftbukkit.entity.CraftEntityType.minecraftToBukkit(entityType),
+                    reason
+                );
+                if (!event.callEvent()) {
+                    if (event.shouldAbortSpawn()) {
+                        if (onAbort != null) {
+                            onAbort.run();
+                        }
+                        return Optional.empty();
+                    }
+                    break;
+                }
+                // Paper end - PreCreatureSpawnEvent
                 T mob = (T)entityType.create(level, null, searchPos, spawnReason, false, false);
                 if (mob != null) {
                     if (mob.checkSpawnRules(level, spawnReason) && mob.checkSpawnObstruction(level)) {
-                        level.addFreshEntityWithPassengers(mob);
+                        level.addFreshEntityWithPassengers(mob, reason); // CraftBukkit
+                        if (mob.isRemoved()) return Optional.empty(); // CraftBukkit
                         mob.playAmbientSound();
                         return Optional.of(mob);
                     }
 
-                    mob.discard();
+                    mob.discard(null); // CraftBukkit - add Bukkit remove cause
                 }
             }
         }
