From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/util/StringUtil.java b/net/minecraft/util/StringUtil.java
index 7a9f7942723c4a37c6c58e45d2c723211abecab1..7957e0cfc43909c5268698a11c4933d20b4d9155 100644
--- a/net/minecraft/util/StringUtil.java
+++ b/net/minecraft/util/StringUtil.java
@@ -85,6 +85,25 @@ public class StringUtil {
         return builder.toString();
     }
 
+    // Paper start - Username validation
+    public static boolean isReasonablePlayerName(final String name) {
+        if (name.isEmpty() || name.length() > 16) {
+            return false;
+        }
+
+        for (int i = 0, len = name.length(); i < len; ++i) {
+            final char c = name.charAt(i);
+            if ((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || (c == '_' || c == '.')) {
+                continue;
+            }
+
+            return false;
+        }
+
+        return true;
+    }
+    // Paper end - Username validation
+
     public static boolean isWhitespace(final int codepoint) {
         return Character.isWhitespace(codepoint) || Character.isSpaceChar(codepoint);
     }
