From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/util/TickThrottler.java b/net/minecraft/util/TickThrottler.java
index cb37c72ae25749653abb373ae134998e86323a81..49ded8671febedb8a0ca977f7bf444e6f28ecfc2 100644
--- a/net/minecraft/util/TickThrottler.java
+++ b/net/minecraft/util/TickThrottler.java
@@ -3,7 +3,7 @@ package net.minecraft.util;
 public class TickThrottler {
     private final int incrementStep;
     private final int threshold;
-    private int count;
+    private final java.util.concurrent.atomic.AtomicInteger count = new java.util.concurrent.atomic.AtomicInteger(); // CraftBukkit - multithreaded field
 
     public TickThrottler(final int incrementStep, final int threshold) {
         this.incrementStep = incrementStep;
@@ -11,16 +11,31 @@ public class TickThrottler {
     }
 
     public void increment() {
-        this.count = this.count + this.incrementStep;
+        this.count.addAndGet(this.incrementStep); // CraftBukkit - use thread-safe field access instead
     }
 
     public void tick() {
+        // CraftBukkit start
+        for (int val; (val = this.count.get()) > 0 && !this.count.compareAndSet(val, val - 1); ) ;
+        /* Use thread-safe field access instead
         if (this.count > 0) {
             this.count--;
         }
+        */
+        // CraftBukkit end
     }
 
     public boolean isUnderThreshold() {
-        return this.threshold <= 0 || this.count < this.threshold;
+        // CraftBukkit start - use thread-safe field access instead
+        return this.threshold <= 0 || this.count.get() < this.threshold;
+    }
+
+    public boolean isIncrementAndUnderThreshold() {
+        return this.isIncrementAndUnderThreshold(this.incrementStep, this.threshold);
+    }
+
+    public boolean isIncrementAndUnderThreshold(int incrementStep, int threshold) {
+        return this.count.addAndGet(incrementStep) < threshold;
+        // CraftBukkit end
     }
 }
