From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/com/mojang/brigadier/exceptions/CommandSyntaxException.java b/com/mojang/brigadier/exceptions/CommandSyntaxException.java
index 3370731ee064d2693b972a0765c13dd4fd69f66a..585e0396c899eb909ffd4fd8c5241daf1e3b2816 100644
--- a/com/mojang/brigadier/exceptions/CommandSyntaxException.java
+++ b/com/mojang/brigadier/exceptions/CommandSyntaxException.java
@@ -5,7 +5,7 @@ package com.mojang.brigadier.exceptions;
 
 import com.mojang.brigadier.Message;
 
-public class CommandSyntaxException extends Exception {
+public class CommandSyntaxException extends Exception implements net.kyori.adventure.util.ComponentMessageThrowable { // Paper - Brigadier API
     public static final int CONTEXT_AMOUNT = 10;
     public static boolean ENABLE_COMMAND_STACK_TRACES = true;
     public static BuiltInExceptionProvider BUILT_IN_EXCEPTIONS = new BuiltInExceptions();
@@ -73,4 +73,11 @@ public class CommandSyntaxException extends Exception {
     public int getCursor() {
         return cursor;
     }
+
+    // Paper start - Brigadier API
+    @Override
+    public net.kyori.adventure.text.@org.jspecify.annotations.Nullable Component componentMessage() {
+        return io.papermc.paper.brigadier.PaperBrigadier.componentFromMessage(this.message);
+    }
+    // Paper end - Brigadier API
 }
