From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/chat/ChatDecorator.java b/net/minecraft/network/chat/ChatDecorator.java
index 7f729a53115ed5598df04e7dc5a98f5b68e58322..a11631e7162807d64614bfb9eec8cda658bea54e 100644
--- a/net/minecraft/network/chat/ChatDecorator.java
+++ b/net/minecraft/network/chat/ChatDecorator.java
@@ -5,7 +5,14 @@ import org.jspecify.annotations.Nullable;
 
 @FunctionalInterface
 public interface ChatDecorator {
-    ChatDecorator PLAIN = (player, plain) -> plain;
+    ChatDecorator PLAIN = (player, plain) -> java.util.concurrent.CompletableFuture.completedFuture(plain); // Paper - adventure; support async chat decoration events
 
-    Component decorate(@Nullable ServerPlayer player, Component plain);
+    @Deprecated @io.papermc.paper.annotation.DoNotUse // Paper - adventure; support chat decoration events (callers should use the overload with CommandSourceStack)
+    java.util.concurrent.CompletableFuture<Component> decorate(@Nullable ServerPlayer player, Component plain); // Paper - adventure; support async chat decoration events
+
+    // Paper start - adventure; support async chat decoration events
+    default java.util.concurrent.CompletableFuture<Component> decorate(@Nullable ServerPlayer sender, net.minecraft.commands.@Nullable CommandSourceStack commandSourceStack, Component plain) {
+        throw new UnsupportedOperationException("Must override this implementation");
+    }
+    // Paper end - adventure; support async chat decoration events
 }
