From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/chat/OutgoingChatMessage.java b/net/minecraft/network/chat/OutgoingChatMessage.java
index 899a0d9aac12e491e456fe96aa4cb660335b3666..259cbcbdd3cd47907629772dc7565212bb434729 100644
--- a/net/minecraft/network/chat/OutgoingChatMessage.java
+++ b/net/minecraft/network/chat/OutgoingChatMessage.java
@@ -7,6 +7,12 @@ public interface OutgoingChatMessage {
 
     void sendToPlayer(ServerPlayer player, boolean filtered, ChatType.Bound chatType);
 
+    // Paper start
+    default void sendToPlayer(final ServerPlayer player, final boolean filtered, final ChatType.Bound chatType, final @org.jspecify.annotations.Nullable Component unsigned) {
+        this.sendToPlayer(player, filtered, chatType);
+    }
+    // Paper end
+
     static OutgoingChatMessage create(final PlayerChatMessage message) {
         return message.isSystem() ? new OutgoingChatMessage.Disguised(message.decoratedContent()) : new OutgoingChatMessage.Player(message);
     }
@@ -14,7 +20,13 @@ public interface OutgoingChatMessage {
     record Disguised(@Override Component content) implements OutgoingChatMessage {
         @Override
         public void sendToPlayer(final ServerPlayer player, final boolean filtered, final ChatType.Bound chatType) {
-            player.connection.sendDisguisedChatMessage(this.content, chatType);
+            // Paper start
+            this.sendToPlayer(player, filtered, chatType, null);
+        }
+        @Override
+        public void sendToPlayer(final ServerPlayer player, final boolean filtered, final ChatType.Bound chatType, final @org.jspecify.annotations.Nullable Component unsigned) {
+            player.connection.sendDisguisedChatMessage(unsigned != null ? unsigned : this.content, chatType);
+            // Paper end
         }
     }
 
@@ -26,7 +38,14 @@ public interface OutgoingChatMessage {
 
         @Override
         public void sendToPlayer(final ServerPlayer player, final boolean filtered, final ChatType.Bound chatType) {
+            // Paper start
+            this.sendToPlayer(player, filtered, chatType, null);
+        }
+        @Override
+        public void sendToPlayer(final ServerPlayer player, final boolean filtered, final ChatType.Bound chatType, final @org.jspecify.annotations.Nullable Component unsigned) {
+            // Paper end
             PlayerChatMessage filteredMessage = this.message.filter(filtered);
+            filteredMessage = unsigned != null ? filteredMessage.withUnsignedContent(unsigned) : filteredMessage; // Paper
             if (!filteredMessage.isFullyFiltered()) {
                 player.connection.sendPlayerChatMessage(filteredMessage, chatType);
             }
