From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/core/dispenser/MinecartDispenseItemBehavior.java b/net/minecraft/core/dispenser/MinecartDispenseItemBehavior.java
index 6923988c81a16e34392f6260c5fa0b27f5271833..48f9dd43b12575916d645838a47efb445a0ff8d5 100644
--- a/net/minecraft/core/dispenser/MinecartDispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/MinecartDispenseItemBehavior.java
@@ -58,12 +58,37 @@ public class MinecartDispenseItemBehavior extends DefaultDispenseItemBehavior {
         }
 
         Vec3 spawnPos = new Vec3(spawnX, spawnY + yOffset, spawnZ);
+        // CraftBukkit start
+        ItemStack singleItemStack = dispensed.copyWithCount(1); // Paper - shrink below and single item in event
+        org.bukkit.block.Block block = org.bukkit.craftbukkit.block.CraftBlock.at(level, source.pos());
+        org.bukkit.craftbukkit.inventory.CraftItemStack craftItem = org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(singleItemStack);
+
+        org.bukkit.event.block.BlockDispenseEvent event = new org.bukkit.event.block.BlockDispenseEvent(block, craftItem.clone(), org.bukkit.craftbukkit.util.CraftVector.toBukkit(spawnPos));
+        level.getCraftServer().getPluginManager().callEvent(event);
+
+        if (event.isCancelled()) {
+            return dispensed;
+        }
+
+        boolean shrink = true;
+        if (!event.getItem().equals(craftItem)) {
+            shrink = false;
+            // Chain to handler for new item
+            ItemStack eventStack = org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(event.getItem());
+            DispenseItemBehavior dispenseBehavior = DispenserBlock.getDispenseBehavior(source, eventStack);
+            if (dispenseBehavior != DispenseItemBehavior.NOOP && dispenseBehavior != this) {
+                dispenseBehavior.dispense(source, eventStack);
+                return dispensed;
+            }
+        }
+
+        singleItemStack = org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(event.getItem());
         AbstractMinecart minecart = AbstractMinecart.createMinecart(
-            level, spawnPos.x, spawnPos.y, spawnPos.z, this.entityType, EntitySpawnReason.DISPENSER, dispensed, null
+            level, event.getVelocity().getX(), event.getVelocity().getY(), event.getVelocity().getZ(), this.entityType, EntitySpawnReason.DISPENSER, singleItemStack, null
         );
         if (minecart != null) {
-            level.addFreshEntity(minecart);
-            dispensed.shrink(1);
+            if (level.addFreshEntity(minecart) && shrink) dispensed.shrink(1); // Paper - if entity add was successful and supposed to shrink
+            // CraftBukkit end
         }
 
         return dispensed;
