From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/core/dispenser/ShulkerBoxDispenseBehavior.java b/net/minecraft/core/dispenser/ShulkerBoxDispenseBehavior.java
index 7269556a4bcf0716c7cc857bebdbf5eb70db607f..736a6b6aac3a56756afbc9b0da210614a4b0bf5b 100644
--- a/net/minecraft/core/dispenser/ShulkerBoxDispenseBehavior.java
+++ b/net/minecraft/core/dispenser/ShulkerBoxDispenseBehavior.java
@@ -20,8 +20,34 @@ public class ShulkerBoxDispenseBehavior extends OptionalDispenseItemBehavior {
             BlockPos relativePos = source.pos().relative(facing);
             Direction clickedFace = source.level().isEmptyBlock(relativePos.below()) ? facing : Direction.UP;
 
+            // CraftBukkit start
+            org.bukkit.block.Block bukkitBlock = org.bukkit.craftbukkit.block.CraftBlock.at(source.level(), source.pos());
+            org.bukkit.craftbukkit.inventory.CraftItemStack craftItem = org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(dispensed.copyWithCount(1));
+
+            org.bukkit.event.block.BlockDispenseEvent event = new org.bukkit.event.block.BlockDispenseEvent(bukkitBlock, craftItem.clone(), org.bukkit.craftbukkit.util.CraftVector.toBukkit(relativePos));
+            source.level().getCraftServer().getPluginManager().callEvent(event);
+
+            if (event.isCancelled()) {
+                return dispensed;
+            }
+
+            if (!event.getItem().equals(craftItem)) {
+                // Chain to handler for new item
+                ItemStack eventStack = org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(event.getItem());
+                DispenseItemBehavior dispenseBehavior = DispenserBlock.getDispenseBehavior(source, eventStack);
+                if (dispenseBehavior != DispenseItemBehavior.NOOP && dispenseBehavior != this) {
+                    dispenseBehavior.dispense(source, eventStack);
+                    return dispensed;
+                }
+            }
+            // CraftBukkit end
             try {
-                this.setSuccess(blockItem.place(new DirectionalPlaceContext(source.level(), relativePos, facing, dispensed, clickedFace)).consumesAction());
+                // Paper start - track changed items in the dispense event
+                this.setSuccess(blockItem.place(new DirectionalPlaceContext(source.level(), relativePos, facing, org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(event.getItem()), clickedFace)).consumesAction());
+                if (this.isSuccess()) {
+                    dispensed.shrink(1); // vanilla shrink is in the place function above, manually handle it here
+                }
+                // Paper end - track changed items in the dispense event
             } catch (Exception e) {
                 LOGGER.error("Error trying to place shulker box at {}", relativePos, e);
             }
