From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/core/dispenser/SulfurCubeBlockDispenseItemBehavior.java b/net/minecraft/core/dispenser/SulfurCubeBlockDispenseItemBehavior.java
index 981a193c61e3c6788b29f6589a7d19c6be9411e4..f94ff9370ae3f1350e183e39f7fd677ea705fb41 100644
--- a/net/minecraft/core/dispenser/SulfurCubeBlockDispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/SulfurCubeBlockDispenseItemBehavior.java
@@ -12,12 +12,19 @@ public class SulfurCubeBlockDispenseItemBehavior extends DefaultDispenseItemBeha
 
     @Override
     protected ItemStack execute(final BlockSource source, final ItemStack dispensed) {
-        return dispenseBlock(source.level(), source.pos().relative(source.state().getValue(DispenserBlock.FACING)), dispensed)
+        return dispenseBlock(source.level(), source.pos().relative(source.state().getValue(DispenserBlock.FACING)), dispensed, source, this) // Paper - Call BlockDispenseEvent
             ? dispensed
             : super.execute(source, dispensed);
     }
 
-    public static boolean dispenseBlock(final ServerLevel level, final BlockPos pos, final ItemStack dispensed) {
+    // Paper start - Call BlockDispenseEvent
+    public static boolean dispenseBlock(final ServerLevel level, final BlockPos pos, final ItemStack dispensed, final BlockSource source, final @org.jspecify.annotations.Nullable DispenseItemBehavior currentBehavior) {
+        ItemStack result = org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockDispenseEvent(source, pos, dispensed, currentBehavior);
+        if (result != null) { // todo - snapshot - setItem should probably override the equipped item + possible event in the for loop
+            return false;
+        }
+        // Paper end - Call BlockDispenseEvent
+
         for (SulfurCube entity : level.getEntitiesOfClass(SulfurCube.class, new AABB(pos))) {
             if (entity.equipItem(dispensed)) {
                 dispensed.shrink(1);
