From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/gameevent/vibrations/VibrationSystem.java b/net/minecraft/world/level/gameevent/vibrations/VibrationSystem.java
index dd0abf0c8bb7c4ca6e0e0681477313a28822b333..7364d6ae99772912bf353f7a004dd26f00fa80ae 100644
--- a/net/minecraft/world/level/gameevent/vibrations/VibrationSystem.java
+++ b/net/minecraft/world/level/gameevent/vibrations/VibrationSystem.java
@@ -124,7 +124,7 @@ public interface VibrationSystem {
         public static final Codec<VibrationSystem.Data> CODEC = RecordCodecBuilder.create(
             i -> i.group(
                     VibrationInfo.CODEC.lenientOptionalFieldOf("event").forGetter(o -> Optional.ofNullable(o.currentVibration)),
-                    VibrationSelector.CODEC.fieldOf("selector").forGetter(VibrationSystem.Data::getSelectionStrategy),
+                    VibrationSelector.CODEC.optionalFieldOf("selector").xmap(o -> o.orElseGet(VibrationSelector::new), Optional::of).forGetter(VibrationSystem.Data::getSelectionStrategy), // Paper - fix MapLike spam for missing "selector" in 1.19.2
                     ExtraCodecs.optionalAlwaysPresentFieldOf(ExtraCodecs.NON_NEGATIVE_INT, "event_delay", 0)
                         .forGetter(VibrationSystem.Data::getTravelTimeInTicks)
                 )
@@ -225,7 +225,14 @@ public interface VibrationSystem {
             }
 
             Vec3 destination = listenerSourcePos.get();
-            if (!user.canReceiveVibration(level, BlockPos.containing(sourcePosition), event, context)) {
+            // CraftBukkit start
+            boolean defaultCancel = !user.canReceiveVibration(level, BlockPos.containing(sourcePosition), event, context);
+            Entity entity = context.sourceEntity();
+            org.bukkit.event.block.BlockReceiveGameEvent event1 = new org.bukkit.event.block.BlockReceiveGameEvent(org.bukkit.craftbukkit.CraftGameEvent.minecraftHolderToBukkit(event), org.bukkit.craftbukkit.block.CraftBlock.at(level, BlockPos.containing(destination)), (entity == null) ? null : entity.getBukkitEntity());
+            event1.setCancelled(defaultCancel);
+            level.getCraftServer().getPluginManager().callEvent(event1);
+            if (event1.isCancelled()) {
+                // CraftBukkit end
                 return false;
             }
 
