From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/core/registries/BuiltInRegistries.java b/net/minecraft/core/registries/BuiltInRegistries.java
index 7cf33bfded8825676c1b170fb688c6d9e2daa2a8..56327b51490b5657f22bece5c0c3e449ae023241 100644
--- a/net/minecraft/core/registries/BuiltInRegistries.java
+++ b/net/minecraft/core/registries/BuiltInRegistries.java
@@ -365,6 +365,11 @@ public class BuiltInRegistries {
     public static final Registry<MapCodec<? extends SlotSource>> SLOT_SOURCE_TYPE = registerSimple(Registries.SLOT_SOURCE_TYPE, SlotSources::bootstrap);
     public static final Registry<Consumer<GameTestHelper>> TEST_FUNCTION = registerSimple(Registries.TEST_FUNCTION, BuiltinTestFunctions::bootstrap);
     public static final Registry<? extends Registry<?>> REGISTRY = WRITABLE_REGISTRY;
+    // Paper start - add built-in registry conversions
+    public static final io.papermc.paper.registry.data.util.Conversions STATIC_ACCESS_CONVERSIONS = new io.papermc.paper.registry.data.util.Conversions(
+        new net.minecraft.resources.RegistryOps.HolderLookupAdapter(net.minecraft.server.RegistryLayer.STATIC_ACCESS)
+    );
+    // Paper end - add built-in registry conversions
 
     private static <T> Registry<T> registerSimple(final ResourceKey<? extends Registry<T>> name, final BuiltInRegistries.RegistryBootstrap<T> loader) {
         return internalRegister(name, new MappedRegistry<>(name, Lifecycle.stable(), false), loader);
@@ -392,6 +397,7 @@ public class BuiltInRegistries {
         final ResourceKey<? extends Registry<T>> name, final R registry, final BuiltInRegistries.RegistryBootstrap<T> loader
     ) {
         Bootstrap.checkBootstrapCalled(() -> "registry " + name.identifier());
+        io.papermc.paper.registry.PaperRegistryAccess.instance().registerRegistry(registry); // Paper - initialize API registry
         Identifier key = name.identifier();
         LOADERS.put(key, () -> loader.run(registry));
         WRITABLE_REGISTRY.register((ResourceKey)name, registry, RegistrationInfo.BUILT_IN);
@@ -399,16 +405,34 @@ public class BuiltInRegistries {
     }
 
     public static void bootStrap() {
+        // Paper start
+        bootStrap(() -> {});
+    }
+    public static void bootStrap(Runnable runnable) {
+        // Paper end
+        REGISTRY.freeze(); // Paper - freeze main registry early
         createContents();
+        runnable.run(); // Paper
         freeze();
         validate(REGISTRY);
     }
 
     private static void createContents() {
+        // Paper start - class-load org.bukkit.Registry
+        // we have to class-load Registry here to create all the CraftRegistry instances
+        // that will be created when Registry is class-loaded before RegistryAccess#getRegistry
+        // would try to create them in lockReferenceHolder
+        try {
+            Class.forName(org.bukkit.Registry.class.getName());
+        } catch (final ClassNotFoundException ex) {
+            throw new RuntimeException(ex);
+        }
+        // Paper end - class-load org.bukkit.Registry
         LOADERS.forEach((key, value) -> {
             if (value.get() == null) {
                 LOGGER.error("Unable to bootstrap registry '{}'", key);
             }
+            io.papermc.paper.registry.PaperRegistryAccess.instance().lockReferenceHolders(ResourceKey.createRegistryKey(key)); // Paper - lock reference holder creation
         });
     }
 
@@ -417,6 +441,7 @@ public class BuiltInRegistries {
 
         for (Registry<?> registry : REGISTRY) {
             bindBootstrappedTagsToEmpty(registry);
+            io.papermc.paper.registry.PaperRegistryListenerManager.INSTANCE.runFreezeListeners(registry.key(), STATIC_ACCESS_CONVERSIONS); // Paper
             registry.freeze();
         }
     }
