From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/projectile/hurtingprojectile/windcharge/WindCharge.java b/net/minecraft/world/entity/projectile/hurtingprojectile/windcharge/WindCharge.java
index 03b9a4b45511e7026225bda8b58193f013d08e25..c91adb74bfdc93c5e163b37f7517f530703f6ad5 100644
--- a/net/minecraft/world/entity/projectile/hurtingprojectile/windcharge/WindCharge.java
+++ b/net/minecraft/world/entity/projectile/hurtingprojectile/windcharge/WindCharge.java
@@ -60,6 +60,12 @@ public class WindCharge extends AbstractWindCharge {
 
     @Override
     public void explode(final Vec3 position) {
+        // Paper start - Fire event for WindCharge explosions
+        org.bukkit.event.entity.ExplosionPrimeEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callExplosionPrimeEvent(this, RADIUS, false);
+        if (event.isCancelled()) {
+            return;
+        }
+        // Paper end - Fire event for WindCharge explosions
         this.level()
             .explode(
                 this,
@@ -68,8 +74,8 @@ public class WindCharge extends AbstractWindCharge {
                 position.x(),
                 position.y(),
                 position.z(),
-                1.2F,
-                false,
+                event.getRadius(), // Paper - Fire event for WindCharge explosions
+                event.getFire(), // Paper - Fire event for WindCharge explosions
                 Level.ExplosionInteraction.TRIGGER,
                 ParticleTypes.GUST_EMITTER_SMALL,
                 ParticleTypes.GUST_EMITTER_LARGE,
