From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/enchantment/effects/ApplyExhaustion.java b/net/minecraft/world/item/enchantment/effects/ApplyExhaustion.java
index 989b7d86dd0b307173fb07a3def541d959a14daf..2de2504228c872b854107ca030a8db7d623c9c83 100644
--- a/net/minecraft/world/item/enchantment/effects/ApplyExhaustion.java
+++ b/net/minecraft/world/item/enchantment/effects/ApplyExhaustion.java
@@ -17,7 +17,7 @@ public record ApplyExhaustion(LevelBasedValue amount) implements EnchantmentEnti
     @Override
     public void apply(final ServerLevel serverLevel, final int enchantmentLevel, final EnchantedItemInUse item, final Entity entity, final Vec3 position) {
         if (entity instanceof Player livingEntity) {
-            livingEntity.causeFoodExhaustion(this.amount.calculate(enchantmentLevel));
+            livingEntity.causeFoodExhaustion(this.amount.calculate(enchantmentLevel), org.bukkit.event.entity.EntityExhaustionEvent.ExhaustionReason.ENCHANTMENT_EFFECT); // Paper - add ExhaustionReason
         }
     }
 
