From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/enchantment/effects/SummonEntityEffect.java b/net/minecraft/world/item/enchantment/effects/SummonEntityEffect.java
index 1fbdac73f1e663a401d818bdba38fa8f9e155f68..cbe1850a0ea435174fd57f4e24ff5d30d6b5aa69 100644
--- a/net/minecraft/world/item/enchantment/effects/SummonEntityEffect.java
+++ b/net/minecraft/world/item/enchantment/effects/SummonEntityEffect.java
@@ -34,11 +34,18 @@ public record SummonEntityEffect(HolderSet<EntityType<?>> entityTypes, boolean j
         if (Level.isInSpawnableBounds(blockPos)) {
             Optional<Holder<EntityType<?>>> entityType = this.entityTypes().getRandomElement(serverLevel.getRandom());
             if (!entityType.isEmpty()) {
-                Entity spawned = entityType.get().value().spawn(serverLevel, blockPos, EntitySpawnReason.TRIGGERED);
+                Entity spawned = entityType.get().value().create(serverLevel, null, blockPos, EntitySpawnReason.TRIGGERED, false, false); // CraftBukkit
                 if (spawned != null) {
                     if (spawned instanceof LightningBolt lightningBolt && item.owner() instanceof ServerPlayer player) {
                         lightningBolt.setCause(player);
                     }
+                    // CraftBukkit start
+                    if (spawned instanceof LightningBolt) {
+                        serverLevel.strikeLightning(spawned, (item.itemStack().is(net.minecraft.world.item.Items.TRIDENT)) ? org.bukkit.event.weather.LightningStrikeEvent.Cause.TRIDENT : org.bukkit.event.weather.LightningStrikeEvent.Cause.ENCHANTMENT);
+                    } else {
+                        serverLevel.addFreshEntityWithPassengers(spawned, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.ENCHANTMENT);
+                    }
+                    // CraftBukkit end
 
                     if (this.joinTeam && entity.getTeam() != null) {
                         serverLevel.getScoreboard().addPlayerToTeam(spawned.getScoreboardName(), entity.getTeam());
