From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java b/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java
index 239b7fa504652d2644cff35b521191fe42311e49..0c0228b7e6b50f43cc546e5cb30a48141988a6b4 100644
--- a/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java
+++ b/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java
@@ -14,8 +14,25 @@ import net.minecraft.resources.RegistryOps;
 
 public record ClientboundLoginDisconnectPacket(Component reason) implements Packet<ClientLoginPacketListener> {
     private static final RegistryOps<JsonElement> OPS = RegistryAccess.EMPTY.createSerializationContext(JsonOps.INSTANCE);
-    public static final StreamCodec<ByteBuf, ClientboundLoginDisconnectPacket> STREAM_CODEC = StreamCodec.composite(
-        ByteBufCodecs.lenientJson(262144).apply(ByteBufCodecs.fromCodec(OPS, ComponentSerialization.CODEC)),
+    // Paper start - localized codec
+    // In the login phase, buffer.adventure$locale field is most likely null, but plugins may use internals to set it via the channel attribute
+    public static final StreamCodec<net.minecraft.network.FriendlyByteBuf, ClientboundLoginDisconnectPacket> STREAM_CODEC = StreamCodec.composite(
+        new StreamCodec<>() {
+
+            private static final net.minecraft.network.codec.StreamCodec<ByteBuf, JsonElement> LENIENT_JSON = ByteBufCodecs.lenientJson(net.minecraft.network.FriendlyByteBuf.MAX_COMPONENT_STRING_LENGTH);
+            @Override
+            public Component decode(final net.minecraft.network.FriendlyByteBuf buffer) {
+                java.util.Locale bufLocale = buffer.adventure$locale;
+                return LENIENT_JSON.apply(ByteBufCodecs.fromCodec(OPS, ComponentSerialization.localizedCodec(bufLocale == null ? java.util.Locale.US : bufLocale))).decode(buffer);
+            }
+
+            @Override
+            public void encode(final net.minecraft.network.FriendlyByteBuf buffer, final Component value) {
+                java.util.Locale bufLocale = buffer.adventure$locale;
+                LENIENT_JSON.apply(ByteBufCodecs.fromCodec(OPS, ComponentSerialization.localizedCodec(bufLocale == null ? java.util.Locale.US : bufLocale))).encode(buffer, value);
+            }
+        },
+        // Paper end - localized codec
         ClientboundLoginDisconnectPacket::reason,
         ClientboundLoginDisconnectPacket::new
     );
