From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/util/filefix/FileFixerUpper.java b/net/minecraft/util/filefix/FileFixerUpper.java
index d8cee36016620a5667821cc6cce4ee0b478a487c..5737848bc52ce0fac3a9b0406b8ad37b57284afa 100644
--- a/net/minecraft/util/filefix/FileFixerUpper.java
+++ b/net/minecraft/util/filefix/FileFixerUpper.java
@@ -82,6 +82,7 @@ public class FileFixerUpper {
     ) throws FileFixException {
         int loadedVersion = NbtUtils.getDataVersion(levelDataTag);
         if (this.requiresFileFixing(loadedVersion)) {
+            if (!io.papermc.paper.world.migration.WorldFolderMigration.didInitialLoad && worldVersionToFileFixerVersion(loadedVersion) == 0) io.papermc.paper.world.migration.WorldFolderMigration.warnAndDelayStartupMigration(); // Paper - warn and delay migration if performing the intial migration from 1.21.11 to >= 26.1
             LOGGER.info("Starting upgrade for world \"{}\"", worldAccess.getLevelId());
             Path worldFolder = worldAccess.getLevelDirectory().path();
             Path fileFixDirectory = worldFolder.resolve("filefix");
@@ -312,7 +313,7 @@ public class FileFixerUpper {
                     try {
                         Files.deleteIfExists(movesFile);
                     } catch (IOException e2) {
-                        LOGGER.warn("Failed to delete {}", movesFile, e);
+                        LOGGER.warn("Failed to delete {}", movesFile, e2); // Paper - print correct stacktrace (MC-308790)
                     }
                 }
 
@@ -351,8 +352,8 @@ public class FileFixerUpper {
                     try {
                         deleteDirectory(worldFolder);
                     } catch (Exception e2) {
-                        LOGGER.warn("Failed to delete outdated world folder: ", e);
-                        throw new FailedCleanupFileFixException(e, tempWorldTopLevel.getFileName().toString(), fileSystemCapabilities);
+                        LOGGER.warn("Failed to delete outdated world folder: ", e2); // Paper - print correct stacktrace (MC-308790)
+                        throw new FailedCleanupFileFixException(e2, tempWorldTopLevel.getFileName().toString(), fileSystemCapabilities); // Paper - print correct stacktrace (MC-308790)
                     }
                 }
 
