From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/levelgen/structure/StructureCheck.java b/net/minecraft/world/level/levelgen/structure/StructureCheck.java
index 270545bf8db9e45a57d2f9c8064b3586fc49a3fd..2c0e0a2402a76196e74d249308fb5c628741ef0a 100644
--- a/net/minecraft/world/level/levelgen/structure/StructureCheck.java
+++ b/net/minecraft/world/level/levelgen/structure/StructureCheck.java
@@ -52,8 +52,12 @@ public class StructureCheck {
     private final DataFixer fixerUpper;
     private final Long2ObjectMap<Object2IntMap<Structure>> loadedChunks = new Long2ObjectOpenHashMap<>();
     private final Map<Structure, Long2BooleanMap> featureChecks = new HashMap<>();
+    // Paper start - also store level stem
+    private final ResourceKey<net.minecraft.world.level.dimension.LevelStem> levelStem;
 
     public StructureCheck(
+        final ResourceKey<net.minecraft.world.level.dimension.LevelStem> levelStem,
+    // Paper end - also store level stem
         final ChunkScanAccess storageAccess,
         final RegistryAccess registryAccess,
         final StructureTemplateManager structureTemplateManager,
@@ -75,8 +79,23 @@ public class StructureCheck {
         this.biomeSource = biomeSource;
         this.seed = seed;
         this.fixerUpper = fixerUpper;
+        this.levelStem = levelStem; // Paper - also store level stem
     }
 
+    // Paper start - add missing structure salt configs
+    @Nullable
+    private Integer getSaltOverride(Structure type) {
+        if (this.heightAccessor instanceof net.minecraft.server.level.ServerLevel serverLevel) {
+            if (type instanceof net.minecraft.world.level.levelgen.structure.structures.MineshaftStructure) {
+                return serverLevel.spigotConfig.mineshaftSeed;
+            } else if (type instanceof net.minecraft.world.level.levelgen.structure.structures.BuriedTreasureStructure) {
+                return serverLevel.spigotConfig.buriedTreasureSeed;
+            }
+        }
+        return null;
+    }
+    // Paper end - add missing structure seed configs
+
     public StructureCheckResult checkStart(final ChunkPos pos, final Structure structure, final StructurePlacement placement, final boolean requireUnreferenced) {
         long posKey = pos.pack();
         Object2IntMap<Structure> cachedResult = this.loadedChunks.get(posKey);
@@ -89,7 +108,7 @@ public class StructureCheck {
             return storageCheckResult;
         }
 
-        if (!placement.applyAdditionalChunkRestrictions(pos.x(), pos.z(), this.seed)) {
+        if (!placement.applyAdditionalChunkRestrictions(pos.x(), pos.z(), this.seed, this.getSaltOverride(structure))) { // Paper - add missing structure seed configs
             return StructureCheckResult.START_NOT_PRESENT;
         }
 
@@ -135,7 +154,7 @@ public class StructureCheck {
         if (collectFields.getResult() instanceof CompoundTag chunkTag) {
             int version = NbtUtils.getDataVersion(chunkTag);
             SimpleRegionStorage.injectDatafixingContext(
-                chunkTag, ChunkMap.getChunkDataFixContextTag(this.dimension, this.chunkGenerator.getTypeNameForDataFixer())
+                chunkTag, ChunkMap.getChunkDataFixContextTag(this.dimension, this.chunkGenerator.getTypeNameForDataFixer(), this.levelStem) // Paper - pass level stem
             );
 
             CompoundTag fixedChunkTag;
