From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/AbstractFurnaceMenu.java b/net/minecraft/world/inventory/AbstractFurnaceMenu.java
index acac9cfd5f9f3d6617c2f24d5fd2df4d219080ca..19bfba135158f19a6055af611d4fc2b34be9ca17 100644
--- a/net/minecraft/world/inventory/AbstractFurnaceMenu.java
+++ b/net/minecraft/world/inventory/AbstractFurnaceMenu.java
@@ -32,6 +32,21 @@ public abstract class AbstractFurnaceMenu extends RecipeBookMenu {
     protected final Level level;
     private final RecipePropertySet acceptedInputs;
     private final RecipeBookType recipeBookType;
+    // CraftBukkit start
+    private org.bukkit.craftbukkit.inventory.view.@org.jspecify.annotations.Nullable CraftFurnaceView view = null;
+    private final Inventory inventory;
+
+    @Override
+    public org.bukkit.craftbukkit.inventory.view.CraftFurnaceView getBukkitView() {
+        if (this.view != null) {
+            return this.view;
+        }
+
+        org.bukkit.craftbukkit.inventory.CraftInventoryFurnace inventory = new org.bukkit.craftbukkit.inventory.CraftInventoryFurnace((net.minecraft.world.level.block.entity.AbstractFurnaceBlockEntity) this.container);
+        this.view = new org.bukkit.craftbukkit.inventory.view.CraftFurnaceView(this.inventory.player.getBukkitEntity(), inventory, this);
+        return this.view;
+    }
+    // CraftBukkit end
 
     protected AbstractFurnaceMenu(
         final MenuType<?> menuType,
@@ -63,6 +78,7 @@ public abstract class AbstractFurnaceMenu extends RecipeBookMenu {
         this.addSlot(new Slot(container, 0, 56, 17));
         this.addSlot(new FurnaceFuelSlot(this, container, 1, 56, 53));
         this.addSlot(new FurnaceResultSlot(inventory.player, container, 2, 116, 35));
+        this.inventory = inventory; // CraftBukkit
         this.addStandardInventorySlots(inventory, 8, 84);
         this.addDataSlots(data);
     }
@@ -80,6 +96,7 @@ public abstract class AbstractFurnaceMenu extends RecipeBookMenu {
 
     @Override
     public boolean stillValid(final Player player) {
+        if (!this.checkReachable) return true; // CraftBukkit
         return this.container.stillValid(player);
     }
 
