From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/ContainerSynchronizer.java b/net/minecraft/world/inventory/ContainerSynchronizer.java
index 73a03bf5d917b6563123608e22925497b77b4fbb..9b16533a92860376176c55302c9e3b6761347b99 100644
--- a/net/minecraft/world/inventory/ContainerSynchronizer.java
+++ b/net/minecraft/world/inventory/ContainerSynchronizer.java
@@ -13,4 +13,6 @@ public interface ContainerSynchronizer {
     void sendDataChange(AbstractContainerMenu container, int id, int value);
 
     RemoteSlot createSlot();
+
+    default void sendOffHandSlotChange() {} // Paper - Sync offhand slot in menus
 }
