From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/ContainerListener.java b/net/minecraft/world/inventory/ContainerListener.java
index b508374ee4e955c176cf79af857e54bc9c9786dc..fc0de29dd07d9b8c1aac8132f6e30aa3bef474be 100644
--- a/net/minecraft/world/inventory/ContainerListener.java
+++ b/net/minecraft/world/inventory/ContainerListener.java
@@ -6,4 +6,10 @@ public interface ContainerListener {
     void slotChanged(AbstractContainerMenu container, int slotIndex, ItemStack itemStack);
 
     void dataChanged(AbstractContainerMenu container, int id, int value);
+
+    // Paper start - Add PlayerInventorySlotChangeEvent
+    default void slotChanged(AbstractContainerMenu containerToSend, int dataSlotIndex, ItemStack oldStack, ItemStack stack) {
+        slotChanged(containerToSend, dataSlotIndex, stack);
+    }
+    // Paper end - Add PlayerInventorySlotChangeEvent
 }
