From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/DispenserMenu.java b/net/minecraft/world/inventory/DispenserMenu.java
index 0c88f529658c7c3b841f049343657b3317ba4bb8..dd81046e4ed435b9ce13242041ee5cff42e520f8 100644
--- a/net/minecraft/world/inventory/DispenserMenu.java
+++ b/net/minecraft/world/inventory/DispenserMenu.java
@@ -13,6 +13,10 @@ public class DispenserMenu extends AbstractContainerMenu {
     private static final int USE_ROW_SLOT_START = 36;
     private static final int USE_ROW_SLOT_END = 45;
     private final Container dispenser;
+    // CraftBukkit start
+    private org.bukkit.craftbukkit.inventory.@org.jspecify.annotations.Nullable CraftInventoryView view = null;
+    private final Inventory inventory;
+    // CraftBukkit end
 
     public DispenserMenu(final int containerId, final Inventory inventory) {
         this(containerId, inventory, new SimpleContainer(9));
@@ -20,6 +24,9 @@ public class DispenserMenu extends AbstractContainerMenu {
 
     public DispenserMenu(final int containerId, final Inventory inventory, final Container dispenser) {
         super(MenuType.GENERIC_3x3, containerId);
+        // CraftBukkit start - Save player
+        this.inventory = inventory;
+        // CraftBukkit end
         checkContainerSize(dispenser, 9);
         this.dispenser = dispenser;
         dispenser.startOpen(inventory.player);
@@ -38,6 +45,7 @@ public class DispenserMenu extends AbstractContainerMenu {
 
     @Override
     public boolean stillValid(final Player player) {
+        if (!this.checkReachable) return true; // CraftBukkit
         return this.dispenser.stillValid(player);
     }
 
@@ -77,4 +85,17 @@ public class DispenserMenu extends AbstractContainerMenu {
         super.removed(player);
         this.dispenser.stopOpen(player);
     }
+
+    // CraftBukkit start
+    @Override
+    public org.bukkit.craftbukkit.inventory.CraftInventoryView getBukkitView() {
+        if (this.view != null) {
+            return this.view;
+        }
+
+        org.bukkit.craftbukkit.inventory.CraftInventory inventory = new org.bukkit.craftbukkit.inventory.CraftInventory(this.dispenser);
+        this.view = new org.bukkit.craftbukkit.inventory.CraftInventoryView(this.inventory.player.getBukkitEntity(), inventory, this);
+        return this.view;
+    }
+    // CraftBukkit end
 }
