From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/HopperMenu.java b/net/minecraft/world/inventory/HopperMenu.java
index 668adb389ad35ba123f3d8cf6bdc6fe142c8b921..88495d19e30e5c0fcca205603472e0240d8c2ab0 100644
--- a/net/minecraft/world/inventory/HopperMenu.java
+++ b/net/minecraft/world/inventory/HopperMenu.java
@@ -9,6 +9,21 @@ import net.minecraft.world.item.ItemStack;
 public class HopperMenu extends AbstractContainerMenu {
     public static final int CONTAINER_SIZE = 5;
     private final Container hopper;
+    // CraftBukkit start
+    private org.bukkit.craftbukkit.inventory.@org.jspecify.annotations.Nullable CraftInventoryView view = null;
+    private final Inventory inventory;
+
+    @Override
+    public org.bukkit.craftbukkit.inventory.CraftInventoryView getBukkitView() {
+        if (this.view != null) {
+            return this.view;
+        }
+
+        org.bukkit.craftbukkit.inventory.CraftInventory inventory = new org.bukkit.craftbukkit.inventory.CraftInventory(this.hopper);
+        this.view = new org.bukkit.craftbukkit.inventory.CraftInventoryView(this.inventory.player.getBukkitEntity(), inventory, this);
+        return this.view;
+    }
+    // CraftBukkit end
 
     public HopperMenu(final int containerId, final Inventory inventory) {
         this(containerId, inventory, new SimpleContainer(5));
@@ -17,6 +32,7 @@ public class HopperMenu extends AbstractContainerMenu {
     public HopperMenu(final int containerId, final Inventory inventory, final Container hopper) {
         super(MenuType.HOPPER, containerId);
         this.hopper = hopper;
+        this.inventory = inventory; // CraftBukkit
         checkContainerSize(hopper, 5);
         hopper.startOpen(inventory.player);
 
@@ -29,6 +45,7 @@ public class HopperMenu extends AbstractContainerMenu {
 
     @Override
     public boolean stillValid(final Player player) {
+        if (!this.checkReachable) return true; // CraftBukkit
         return this.hopper.stillValid(player);
     }
 
