From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/ResultSlot.java b/net/minecraft/world/inventory/ResultSlot.java
index 0ad2c1c52a17fdbc65f320d75eb69f9aa677430e..b4c5202efe53b08c2e94bdf28877800631581ea1 100644
--- a/net/minecraft/world/inventory/ResultSlot.java
+++ b/net/minecraft/world/inventory/ResultSlot.java
@@ -40,6 +40,11 @@ public class ResultSlot extends Slot {
     protected void onQuickCraft(final ItemStack picked, final int count) {
         this.removeCount += count;
         this.checkTakeAchievements(picked);
+        // Paper start - add ItemCraftedEvent
+        if (!picked.isEmpty()) {
+            new io.papermc.paper.event.inventory.ItemCraftedEvent((org.bukkit.entity.Player) this.player.getBukkitEntity(), org.bukkit.craftbukkit.inventory.CraftItemStack.asBukkitCopy(picked)).callEvent();
+        }
+        // Paper end - add ItemCraftedEvent
     }
 
     @Override
@@ -80,14 +85,19 @@ public class ResultSlot extends Slot {
     private NonNullList<ItemStack> getRemainingItems(final CraftingInput input, final Level level) {
         return level instanceof ServerLevel serverLevel
             ? serverLevel.recipeAccess()
-                .getRecipeFor(RecipeType.CRAFTING, input, serverLevel)
+                .getRecipeFor(RecipeType.CRAFTING, input, serverLevel, this.craftSlots.getCurrentRecipe()) // Paper - Perf: Improve mass crafting; check last recipe used first
                 .map(recipe -> recipe.value().getRemainingItems(input))
-                .orElseGet(() -> copyAllInputItems(input))
+                .orElseGet(() -> CraftingRecipe.defaultCraftingReminder(input)) // Paper - Fix #1822: Return default remainder when recipe is null
             : CraftingRecipe.defaultCraftingReminder(input);
     }
 
     @Override
     public void onTake(final Player player, final ItemStack carried) {
+        // Paper start - add ItemCraftedEvent
+        if (!carried.isEmpty()) {
+            new io.papermc.paper.event.inventory.ItemCraftedEvent((org.bukkit.entity.Player) player.getBukkitEntity(), org.bukkit.craftbukkit.inventory.CraftItemStack.asBukkitCopy(carried)).callEvent();
+        }
+        // Paper end - add ItemCraftedEvent
         this.checkTakeAchievements(carried);
         CraftingInput.Positioned positionedRecipe = this.craftSlots.asPositionedCraftInput();
         CraftingInput input = positionedRecipe.input();
