From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/ResultContainer.java b/net/minecraft/world/inventory/ResultContainer.java
index c47c93c9ed80890e6ae13b592a393f1d01ceb505..cdd7380aae8e7828a1af296fe801a40864406e23 100644
--- a/net/minecraft/world/inventory/ResultContainer.java
+++ b/net/minecraft/world/inventory/ResultContainer.java
@@ -11,6 +11,55 @@ import org.jspecify.annotations.Nullable;
 public class ResultContainer implements Container, RecipeCraftingHolder {
     private final NonNullList<ItemStack> itemStacks = NonNullList.withSize(1, ItemStack.EMPTY);
     private @Nullable RecipeHolder<?> recipeUsed;
+    // CraftBukkit start
+    private int maxStack = MAX_STACK;
+
+    @Override
+    public java.util.List<ItemStack> getContents() {
+        return this.itemStacks;
+    }
+
+    @Override
+    public org.bukkit.inventory.@Nullable InventoryHolder getOwner() {
+        // Paper start - Add missing InventoryHolders
+        if (this.holder == null && this.holderCreator != null) {
+            this.holder = this.holderCreator.get();
+        }
+        return this.holder; // Result slots don't get an owner
+        // Paper end - Add missing InventoryHolders
+    }
+
+    // Don't need a transaction; the CraftingContainer keeps track of it for us
+    public void onOpen(org.bukkit.craftbukkit.entity.CraftHumanEntity player) {}
+    public void onClose(org.bukkit.craftbukkit.entity.CraftHumanEntity player) {}
+    public java.util.List<org.bukkit.entity.HumanEntity> getViewers() {
+        return new java.util.ArrayList<>();
+    }
+
+    @Override
+    public int getMaxStackSize() {
+        return this.maxStack;
+    }
+
+    public void setMaxStackSize(int size) {
+        this.maxStack = size;
+    }
+
+    @Override
+    public org.bukkit.@Nullable Location getLocation() {
+        return null;
+    }
+    // CraftBukkit end
+    // Paper start - Add missing InventoryHolders
+    private java.util.function.@Nullable Supplier<? extends org.bukkit.inventory.InventoryHolder> holderCreator;
+    private org.bukkit.inventory.@Nullable InventoryHolder holder;
+    public ResultContainer(java.util.function.Supplier<? extends org.bukkit.inventory.InventoryHolder> holderCreator) {
+        this.holderCreator = holderCreator;
+    }
+
+    public ResultContainer() {
+    }
+    // Paper end - Add missing InventoryHolders
 
     @Override
     public int getContainerSize() {
