From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/ShulkerBoxMenu.java b/net/minecraft/world/inventory/ShulkerBoxMenu.java
index 6d565dd6ae347e72d0ac74e8434d86834cefb00e..cb4c8e1f42511ed91ab0fdffddd45b85da0457d8 100644
--- a/net/minecraft/world/inventory/ShulkerBoxMenu.java
+++ b/net/minecraft/world/inventory/ShulkerBoxMenu.java
@@ -9,6 +9,25 @@ import net.minecraft.world.item.ItemStack;
 public class ShulkerBoxMenu extends AbstractContainerMenu {
     private static final int CONTAINER_SIZE = 27;
     private final Container container;
+    // CraftBukkit start
+    private org.bukkit.craftbukkit.inventory.@org.jspecify.annotations.Nullable CraftInventoryView view;
+    private final Inventory inventory;
+
+    @Override
+    public org.bukkit.craftbukkit.inventory.CraftInventoryView getBukkitView() {
+        if (this.view != null) {
+            return this.view;
+        }
+
+        this.view = new org.bukkit.craftbukkit.inventory.CraftInventoryView(this.inventory.player.getBukkitEntity(), new org.bukkit.craftbukkit.inventory.CraftInventory(this.container), this);
+        return this.view;
+    }
+
+    @Override
+    public void startOpen() {
+        this.container.startOpen(this.inventory.player);
+    }
+    // CraftBukkit end
 
     public ShulkerBoxMenu(final int containerId, final Inventory inventory) {
         this(containerId, inventory, new SimpleContainer(27));
@@ -18,7 +37,8 @@ public class ShulkerBoxMenu extends AbstractContainerMenu {
         super(MenuType.SHULKER_BOX, containerId);
         checkContainerSize(container, 27);
         this.container = container;
-        container.startOpen(inventory.player);
+        this.inventory = inventory; // CraftBukkit
+        // container.startOpen(inventory.player); // Paper - don't startOpen until menu actually opens
         int rows = 3;
         int columns = 9;
 
@@ -33,6 +53,7 @@ public class ShulkerBoxMenu extends AbstractContainerMenu {
 
     @Override
     public boolean stillValid(final Player player) {
+        if (!this.checkReachable) return true; // CraftBukkit
         return this.container.stillValid(player);
     }
 
