From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/inventory/SmithingMenu.java b/net/minecraft/world/inventory/SmithingMenu.java
index 0e332e644037cb557ecb149018f6d821e6d25a2e..98dbb4e4cb5e234ddc24449071806e805b50c5bc 100644
--- a/net/minecraft/world/inventory/SmithingMenu.java
+++ b/net/minecraft/world/inventory/SmithingMenu.java
@@ -33,6 +33,9 @@ public class SmithingMenu extends ItemCombinerMenu {
     private final RecipePropertySet templateItemTest;
     private final RecipePropertySet additionItemTest;
     private final DataSlot hasRecipeError = DataSlot.standalone();
+    // CraftBukkit start
+    private org.bukkit.craftbukkit.inventory.CraftInventoryView bukkitEntity;
+    // CraftBukkit end
 
     public SmithingMenu(final int containerId, final Inventory inventory) {
         this(containerId, inventory, ContainerLevelAccess.NULL);
@@ -103,6 +106,7 @@ public class SmithingMenu extends ItemCombinerMenu {
                 && this.getSlot(2).hasItem()
                 && !this.getSlot(this.getResultSlot()).hasItem();
             this.hasRecipeError.set(hasRecipeError ? 1 : 0);
+            org.bukkit.craftbukkit.event.CraftEventFactory.callPrepareResultEvent(this, RESULT_SLOT); // Paper - Add PrepareResultEvent
         }
     }
 
@@ -119,7 +123,9 @@ public class SmithingMenu extends ItemCombinerMenu {
         foundRecipe.ifPresentOrElse(recipe -> {
             ItemStack result = recipe.value().assemble(input);
             this.resultSlots.setRecipeUsed((RecipeHolder<?>)recipe);
-            this.resultSlots.setItem(0, result);
+            // CraftBukkit start
+            org.bukkit.craftbukkit.event.CraftEventFactory.callPrepareSmithingEvent(this.getBukkitView(), result);
+            // CraftBukkit end
         }, () -> {
             this.resultSlots.setRecipeUsed(null);
             this.resultSlots.setItem(0, ItemStack.EMPTY);
@@ -141,4 +147,18 @@ public class SmithingMenu extends ItemCombinerMenu {
     public boolean hasRecipeError() {
         return this.hasRecipeError.get() > 0;
     }
+
+    // CraftBukkit start
+    @Override
+    public org.bukkit.craftbukkit.inventory.CraftInventoryView getBukkitView() {
+        if (this.bukkitEntity != null) {
+            return this.bukkitEntity;
+        }
+
+        org.bukkit.craftbukkit.inventory.CraftInventory inventory = new org.bukkit.craftbukkit.inventory.CraftInventorySmithing(
+                this.access.getLocation(), this.inputSlots, this.resultSlots);
+        this.bukkitEntity = new org.bukkit.craftbukkit.inventory.CraftInventoryView(this.player.getBukkitEntity(), inventory, this);
+        return this.bukkitEntity;
+    }
+    // CraftBukkit end
 }
