From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/entity/BannerBlockEntity.java b/net/minecraft/world/level/block/entity/BannerBlockEntity.java
index 901e4d8888b60bc6b0e940d16f57f920af4dbdc6..ea60053ffd3d974ea1e92c7cb178895fd32a3010 100644
--- a/net/minecraft/world/level/block/entity/BannerBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BannerBlockEntity.java
@@ -48,7 +48,7 @@ public class BannerBlockEntity extends BlockEntity implements Nameable {
     @Override
     protected void saveAdditional(final ValueOutput output) {
         super.saveAdditional(output);
-        if (!this.patterns.equals(BannerPatternLayers.EMPTY)) {
+        if (!this.patterns.equals(BannerPatternLayers.EMPTY) || serialisingForNetwork.get()) { // Paper - always send patterns to client
             output.store("patterns", BannerPatternLayers.CODEC, this.patterns);
         }
 
@@ -59,7 +59,7 @@ public class BannerBlockEntity extends BlockEntity implements Nameable {
     protected void loadAdditional(final ValueInput input) {
         super.loadAdditional(input);
         this.name = parseCustomNameSafe(input, "CustomName");
-        this.patterns = input.read("patterns", BannerPatternLayers.CODEC).orElse(BannerPatternLayers.EMPTY);
+        this.setPatterns(input.read("patterns", BannerPatternLayers.CODEC).orElse(BannerPatternLayers.EMPTY)); // CraftBukkit - apply limits
     }
 
     @Override
@@ -67,9 +67,18 @@ public class BannerBlockEntity extends BlockEntity implements Nameable {
         return ClientboundBlockEntityDataPacket.create(this);
     }
 
+    // Paper start - always send patterns to client
+    ThreadLocal<Boolean> serialisingForNetwork = ThreadLocal.withInitial(() -> Boolean.FALSE);
     @Override
     public CompoundTag getUpdateTag(final HolderLookup.Provider registries) {
+        final Boolean wasSerialisingForNetwork = serialisingForNetwork.get();
+        try {
+            serialisingForNetwork.set(Boolean.TRUE);
         return this.saveWithoutMetadata(registries);
+        } finally {
+            serialisingForNetwork.set(wasSerialisingForNetwork);
+        }
+        // Paper end - always send patterns to client
     }
 
     public BannerPatternLayers getPatterns() {
@@ -89,7 +98,7 @@ public class BannerBlockEntity extends BlockEntity implements Nameable {
     @Override
     protected void applyImplicitComponents(final DataComponentGetter components) {
         super.applyImplicitComponents(components);
-        this.patterns = components.getOrDefault(DataComponents.BANNER_PATTERNS, BannerPatternLayers.EMPTY);
+        this.setPatterns(components.getOrDefault(DataComponents.BANNER_PATTERNS, BannerPatternLayers.EMPTY)); // CraftBukkit - apply limits
         this.name = components.get(DataComponents.CUSTOM_NAME);
     }
 
@@ -105,4 +114,13 @@ public class BannerBlockEntity extends BlockEntity implements Nameable {
         output.discard("patterns");
         output.discard("CustomName");
     }
+
+    // CraftBukkit start
+    public void setPatterns(BannerPatternLayers bannerPatternLayers) {
+        if (bannerPatternLayers.layers().size() > 20) {
+            bannerPatternLayers = new BannerPatternLayers(java.util.List.copyOf(bannerPatternLayers.layers().subList(0, 20)));
+        }
+        this.patterns = bannerPatternLayers;
+    }
+    // CraftBukkit end
 }
