From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/entity/ConduitBlockEntity.java b/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
index 144c9869d917d3807ebeff2385f0932fa60be272..ac0634392c8c3acca4dc089607d6eb148a817797 100644
--- a/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
@@ -165,8 +165,20 @@ public class ConduitBlockEntity extends BlockEntity {
     }
 
     private static void applyEffects(final Level level, final BlockPos worldPosition, final List<BlockPos> effectBlocks) {
+        // CraftBukkit start
+        ConduitBlockEntity.applyEffects(level, worldPosition, ConduitBlockEntity.getRange(effectBlocks));
+    }
+
+    public static int getRange(List<BlockPos> effectBlocks) {
+        // CraftBukkit end
         int activeSize = effectBlocks.size();
         int effectRange = activeSize / 7 * 16;
+        // CraftBukkit start
+        return effectRange;
+    }
+
+    private static void applyEffects(Level level, BlockPos worldPosition, int effectRange) { // i = effect range in blocks
+        // CraftBukkit end
         int x = worldPosition.getX();
         int y = worldPosition.getY();
         int z = worldPosition.getZ();
@@ -175,7 +187,7 @@ public class ConduitBlockEntity extends BlockEntity {
         if (!players.isEmpty()) {
             for (Player player : players) {
                 if (worldPosition.closerThan(player.blockPosition(), effectRange) && player.isInWaterOrRain()) {
-                    player.addEffect(new MobEffectInstance(MobEffects.CONDUIT_POWER, 260, 0, true, true));
+                    player.addEffect(new MobEffectInstance(MobEffects.CONDUIT_POWER, 260, 0, true, true), org.bukkit.event.entity.EntityPotionEffectEvent.Cause.CONDUIT); // CraftBukkit
                 }
             }
         }
@@ -184,13 +196,19 @@ public class ConduitBlockEntity extends BlockEntity {
     private static void updateAndAttackTarget(
         final ServerLevel level, final BlockPos worldPosition, final BlockState blockState, final ConduitBlockEntity entity, final boolean isActive
     ) {
+        // CraftBukkit start - add "damageTarget" boolean
+        updateAndAttackTarget(level, worldPosition, blockState, entity, isActive, true);
+    }
+
+    public static void updateAndAttackTarget(ServerLevel level, BlockPos worldPosition, BlockState blockState, ConduitBlockEntity entity, boolean isActive, boolean damageTarget) {
+        // CraftBukkit end - add "damageTarget" boolean
         EntityReference<LivingEntity> newDestroyTarget = updateDestroyTarget(entity.destroyTarget, level, worldPosition, isActive);
         LivingEntity targetEntity = EntityReference.getLivingEntity(newDestroyTarget, level);
-        if (targetEntity != null) {
+        if (damageTarget && targetEntity != null) { // CraftBukkit
+            if (targetEntity.hurtServer(level, level.damageSources().magic().eventBlockDamager(level, worldPosition), 4.0F)) // CraftBukkit - move up
             level.playSound(
                 null, targetEntity.getX(), targetEntity.getY(), targetEntity.getZ(), SoundEvents.CONDUIT_ATTACK_TARGET, SoundSource.BLOCKS, 1.0F, 1.0F
             );
-            targetEntity.hurtServer(level, level.damageSources().magic(), 4.0F);
         }
 
         if (!Objects.equals(newDestroyTarget, entity.destroyTarget)) {
