From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/entity/DecoratedPotBlockEntity.java b/net/minecraft/world/level/block/entity/DecoratedPotBlockEntity.java
index 55a8c1151a806324b3f002458ea5be565bb9f28f..2d743278ea3bba6844599e89c005ec016715432b 100644
--- a/net/minecraft/world/level/block/entity/DecoratedPotBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/DecoratedPotBlockEntity.java
@@ -25,6 +25,48 @@ import net.minecraft.world.ticks.ContainerSingleItem;
 import org.jspecify.annotations.Nullable;
 
 public class DecoratedPotBlockEntity extends BlockEntity implements ContainerSingleItem.BlockContainerSingleItem, RandomizableContainer {
+
+    // CraftBukkit start - add fields and methods
+    public List<org.bukkit.entity.HumanEntity> transaction = new java.util.ArrayList<>();
+    private int maxStack = MAX_STACK;
+
+    @Override
+    public List<ItemStack> getContents() {
+        return java.util.List.of(this.item);
+    }
+
+    @Override
+    public void onOpen(org.bukkit.craftbukkit.entity.CraftHumanEntity player) {
+        this.transaction.add(player);
+    }
+
+    @Override
+    public void onClose(org.bukkit.craftbukkit.entity.CraftHumanEntity player) {
+        this.transaction.remove(player);
+    }
+
+    @Override
+    public java.util.List<org.bukkit.entity.HumanEntity> getViewers() {
+        return this.transaction;
+    }
+
+    @Override
+    public int getMaxStackSize() {
+        return this.maxStack;
+    }
+
+    @Override
+    public void setMaxStackSize(int size) {
+        this.maxStack = size;
+    }
+
+    @Override
+    public org.bukkit.@Nullable Location getLocation() {
+        if (this.level == null) return null;
+        return org.bukkit.craftbukkit.util.CraftLocation.toBukkit(this.worldPosition, this.level);
+    }
+    // CraftBukkit end
+
     public static final String TAG_SHERDS = "sherds";
     public static final String TAG_ITEM = "item";
     public static final int EVENT_POT_WOBBLES = 1;
@@ -47,8 +89,8 @@ public class DecoratedPotBlockEntity extends BlockEntity implements ContainerSin
             output.store("sherds", PotDecorations.CODEC, this.decorations);
         }
 
-        if (!this.trySaveLootTable(output) && !this.item.isEmpty()) {
-            output.store("item", ItemStack.CODEC, this.item);
+        if (!this.trySaveLootTable(output) && !this.item.isEmpty()) { // Paper - diff on change - hide unnecessary update data
+            output.store("item", ItemStack.CODEC, this.item); // Paper - diff on change - hide unnecessary update data
         }
     }
 
@@ -70,7 +112,14 @@ public class DecoratedPotBlockEntity extends BlockEntity implements ContainerSin
 
     @Override
     public CompoundTag getUpdateTag(final HolderLookup.Provider registries) {
-        return this.saveCustomOnly(registries);
+        // Paper start - hide unnecessary update data
+        // Like chests, decorated pots should not allow clients to inspect their contents without breaking them.
+        final CompoundTag compoundTag = new CompoundTag();
+        if (!this.decorations.equals(PotDecorations.EMPTY)) {
+            compoundTag.store("sherds", PotDecorations.CODEC, this.decorations);
+        }
+        return compoundTag;
+        // Paper end - hide unnecessary update data
     }
 
     public Direction getDirection() {
