From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/entity/TestInstanceBlockEntity.java b/net/minecraft/world/level/block/entity/TestInstanceBlockEntity.java
index 60647bf373857a2e81ee087be48f176ced4e7841..f7233bae65e09b219191433473bb0ea1fb1d8854 100644
--- a/net/minecraft/world/level/block/entity/TestInstanceBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/TestInstanceBlockEntity.java
@@ -163,6 +163,7 @@ public class TestInstanceBlockEntity extends BlockEntity implements BoundingBoxR
 
     @Override
     protected void loadAdditional(final ValueInput input) {
+        super.loadAdditional(input); // Paper - load the PDC
         input.read("data", TestInstanceBlockEntity.Data.CODEC).ifPresent(this::set);
         this.errorMarkers.clear();
         this.errorMarkers.addAll(input.read("errors", TestInstanceBlockEntity.ErrorMarker.LIST_CODEC).orElse(List.of()));
@@ -170,6 +171,7 @@ public class TestInstanceBlockEntity extends BlockEntity implements BoundingBoxR
 
     @Override
     protected void saveAdditional(final ValueOutput output) {
+        super.saveAdditional(output); // Paper - save the PDC
         output.store("data", TestInstanceBlockEntity.Data.CODEC, this.data);
         if (!this.errorMarkers.isEmpty()) {
             output.store("errors", TestInstanceBlockEntity.ErrorMarker.LIST_CODEC, this.errorMarkers);
@@ -344,7 +346,7 @@ public class TestInstanceBlockEntity extends BlockEntity implements BoundingBoxR
     }
 
     private void removeEntities() {
-        this.level.getEntities(null, this.getTestBounds()).stream().filter(entity -> !(entity instanceof Player)).forEach(Entity::discard);
+        this.level.getEntities(null, this.getTestBounds()).stream().filter(entity -> !(entity instanceof Player)).forEach((entity) -> entity.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.DISCARD)); // Paper
     }
 
     private void forceLoadChunks() {
