From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/commands/CommandSource.java b/net/minecraft/commands/CommandSource.java
index dcffa1d811043c281905b7be706ba5a80f6d0230..68b1f048b805eef3902a0cd1a52066a4bd603e6a 100644
--- a/net/minecraft/commands/CommandSource.java
+++ b/net/minecraft/commands/CommandSource.java
@@ -22,6 +22,13 @@ public interface CommandSource {
         public boolean shouldInformAdmins() {
             return false;
         }
+
+        // CraftBukkit start
+        @Override
+        public org.bukkit.command.CommandSender getBukkitSender(CommandSourceStack stack) {
+            return io.papermc.paper.brigadier.NullCommandSender.INSTANCE; // Paper - expose a no-op CommandSender
+        }
+        // CraftBukkit end
     };
 
     void sendSystemMessage(Component message);
@@ -35,4 +42,6 @@ public interface CommandSource {
     default boolean alwaysAccepts() {
         return false;
     }
+
+    org.bukkit.command.CommandSender getBukkitSender(CommandSourceStack wrapper); // CraftBukkit
 }
