From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/jsonrpc/internalapi/MinecraftGameRuleServiceImpl.java b/net/minecraft/server/jsonrpc/internalapi/MinecraftGameRuleServiceImpl.java
index b6cb065a2c420b315b8b17047dc25ff50eb997aa..3082cb0b335a42c816fed9bf82c19b6d1d1ab5b8 100644
--- a/net/minecraft/server/jsonrpc/internalapi/MinecraftGameRuleServiceImpl.java
+++ b/net/minecraft/server/jsonrpc/internalapi/MinecraftGameRuleServiceImpl.java
@@ -14,12 +14,23 @@ import net.minecraft.world.level.gamerules.GameRules;
 public class MinecraftGameRuleServiceImpl implements MinecraftGameRuleService {
     private final NotificationManager notificationManager;
     private final JsonRpcLogger jsonrpcLogger;
+    private @org.jspecify.annotations.Nullable GameRules gameRules; // Paper - per-world game rules
 
     public MinecraftGameRuleServiceImpl(final NotificationManager notificationManager, final JsonRpcLogger jsonrpcLogger) {
         this.notificationManager = notificationManager;
         this.jsonrpcLogger = jsonrpcLogger;
+        this.gameRules = null; // Paper - per-world game rules - cannot get game rules until server is started
     }
 
+    // Paper start - per-world game rules
+    public GameRules getGameRules() {
+        if (this.gameRules == null) {
+            this.gameRules = this.server().overworld().getGameRules();
+        }
+        return this.gameRules;
+    }
+    // Paper end
+
     private DedicatedServer server() {
         return Objects.requireNonNull(this.notificationManager.server());
     }
@@ -28,10 +39,10 @@ public class MinecraftGameRuleServiceImpl implements MinecraftGameRuleService {
     public <T> GameRulesService.GameRuleUpdate<T> updateGameRule(final GameRulesService.GameRuleUpdate<T> update, final ClientInfo clientInfo) {
         GameRule<T> gameRule = update.gameRule();
         MinecraftServer server = this.server();
-        GameRules gameRules = server.getGameRules();
+        GameRules gameRules = this.getGameRules(); // Paper - per-world game rules
         T oldValue = gameRules.get(gameRule);
         T newValue = update.value();
-        gameRules.set(gameRule, newValue, server);
+        gameRules.set(gameRule, newValue, this.server().overworld()); // Paper - per-world game rules - use overworld for vanilla protocol
         this.jsonrpcLogger
             .log(clientInfo, "Game rule '{}' updated from '{}' to '{}'", gameRule.id(), gameRule.serialize(oldValue), gameRule.serialize(newValue));
         return update;
@@ -44,11 +55,11 @@ public class MinecraftGameRuleServiceImpl implements MinecraftGameRuleService {
 
     @Override
     public Stream<GameRule<?>> getAvailableGameRules() {
-        return this.server().getGameRules().availableRules();
+        return this.getGameRules().availableRules(); // Paper - per-world game rules
     }
 
     @Override
     public <T> T getRuleValue(final GameRule<T> gameRule) {
-        return this.server().getGameRules().get(gameRule);
+        return this.getGameRules().get(gameRule); // Paper - per-world game rules
     }
 }
