From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/nbt/SnbtGrammar.java b/net/minecraft/nbt/SnbtGrammar.java
index 687df81b4a67dd61f617bd56146b9a9105cbdfe1..ea95ef086ff548e32ec6bea355fe1248d6509183 100644
--- a/net/minecraft/nbt/SnbtGrammar.java
+++ b/net/minecraft/nbt/SnbtGrammar.java
@@ -619,7 +619,7 @@ public class SnbtGrammar {
             mapEntries, Term.repeatedWithTrailingSeparator(mapEntryRule, mapEntries, StringReaderTerms.character(',')), scope -> scope.getOrThrow(mapEntries)
         );
         Atom<T> mapLiteral = Atom.of("map_literal");
-        rules.put(mapLiteral, Term.sequence(StringReaderTerms.character('{'), rules.named(mapEntries), StringReaderTerms.character('}')), scope -> {
+        rules.put(mapLiteral, Term.sequence(StringReaderTerms.character('{'), Scope.increaseDepth(), rules.named(mapEntries), Scope.decreaseDepth(), StringReaderTerms.character('}')), scope -> { // Paper - track depth
             List<Entry<String, T>> entries = scope.getOrThrow(mapEntries);
             if (entries.isEmpty()) {
                 return emptyMapValue;
@@ -660,9 +660,11 @@ public class SnbtGrammar {
             listLiteral,
             Term.sequence(
                 StringReaderTerms.character('['),
+                Scope.increaseDepth(), // Paper - track depth
                 Term.alternative(
                     Term.sequence(rules.named(arrayPrefix), StringReaderTerms.character(';'), rules.named(intArrayEntries)), rules.named(listEntries)
                 ),
+                Scope.decreaseDepth(), // Paper - track depth
                 StringReaderTerms.character(']')
             ),
             state -> {
