From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/monster/zombie/Husk.java b/net/minecraft/world/entity/monster/zombie/Husk.java
index 28fb35d01c91c31c1e1c72f18c67e0bba1e88dfb..12a0b90c16bf317c6d92843e7fca1129b74bb05c 100644
--- a/net/minecraft/world/entity/monster/zombie/Husk.java
+++ b/net/minecraft/world/entity/monster/zombie/Husk.java
@@ -68,7 +68,7 @@ public class Husk extends Zombie {
         boolean result = super.doHurtTarget(level, target);
         if (result && this.getMainHandItem().isEmpty() && target instanceof LivingEntity livingEntity) {
             float difficulty = level.getCurrentDifficultyAt(this.blockPosition()).getEffectiveDifficulty();
-            livingEntity.addEffect(new MobEffectInstance(MobEffects.HUNGER, 140 * (int)difficulty), this);
+            livingEntity.addEffect(new MobEffectInstance(MobEffects.HUNGER, 140 * (int)difficulty), this, org.bukkit.event.entity.EntityPotionEffectEvent.Cause.ATTACK); // CraftBukkit
         }
 
         return result;
@@ -97,7 +97,7 @@ public class Husk extends Zombie {
         groupData = super.finalizeSpawn(level, difficulty, spawnReason, groupData);
         float difficultyModifier = difficulty.getSpecialMultiplier();
         if (spawnReason != EntitySpawnReason.CONVERSION) {
-            this.setCanPickUpLoot(random.nextFloat() < 0.55F * difficultyModifier);
+            this.setCanPickUpLoot(level.getLevel().paperConfig().entities.behavior.mobsCanAlwaysPickUpLoot.zombies || random.nextFloat() < 0.55F * difficultyModifier); // Paper - Add world settings for mobs picking up loot
         }
 
         if (groupData != null) {
