From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/Container.java b/net/minecraft/world/Container.java
index f6376844fab1d7ee850992e2493f4d91e3ffc6dc..6e08d50794c4af4405f3e164a3fd46f376b3f78f 100644
--- a/net/minecraft/world/Container.java
+++ b/net/minecraft/world/Container.java
@@ -31,9 +31,7 @@ public interface Container extends Clearable, Iterable<ItemStack>, SlotProvider
 
     void setItem(int slot, ItemStack itemStack);
 
-    default int getMaxStackSize() {
-        return 99;
-    }
+    int getMaxStackSize(); // CraftBukkit
 
     default int getMaxStackSize(final ItemStack itemStack) {
         return Math.min(this.getMaxStackSize(), itemStack.getMaxStackSize());
@@ -142,4 +140,22 @@ public interface Container extends Clearable, Iterable<ItemStack>, SlotProvider
             }
         }
     }
+
+    // CraftBukkit start
+    java.util.List<ItemStack> getContents();
+
+    void onOpen(org.bukkit.craftbukkit.entity.CraftHumanEntity player);
+
+    void onClose(org.bukkit.craftbukkit.entity.CraftHumanEntity player);
+
+    java.util.List<org.bukkit.entity.HumanEntity> getViewers();
+
+    org.bukkit.inventory.@Nullable InventoryHolder getOwner();
+
+    void setMaxStackSize(int size);
+
+    org.bukkit.@Nullable Location getLocation();
+
+    int MAX_STACK = Item.ABSOLUTE_MAX_STACK_SIZE;
+    // CraftBukkit end
 }
