From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/LockCode.java b/net/minecraft/world/LockCode.java
index e824970cc7136c4d73290bcac941ac6251c99787..fd35f1859031d25af0641c1e8aa93e7acc72845e 100644
--- a/net/minecraft/world/LockCode.java
+++ b/net/minecraft/world/LockCode.java
@@ -22,8 +22,14 @@ public record LockCode(ItemPredicate predicate) {
         }
     }
 
+    @io.papermc.paper.annotation.DoNotUse @Deprecated // Paper - use new variant for event compatibility
     public boolean canUnlock(final Player player) {
-        return player.isSpectator() || this.unlocksWith(player.getMainHandItem());
+        // Paper start - allow overriding key item
+        return canUnlock(player, player.getMainHandItem());
+    }
+    public boolean canUnlock(Player player, final ItemStack item) {
+        // Paper end - allow overriding key item
+        return player.isSpectator() || this.unlocksWith(item); // Paper - allow overriding key item
     }
 
     public static LockCode fromTag(final ValueInput parent) {
