From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/vehicle/minecart/MinecartCommandBlock.java b/net/minecraft/world/entity/vehicle/minecart/MinecartCommandBlock.java
index d947ddd331a272003f20d5cf6bae7a7d5775e2da..912962e7ba35df57cbb157443fc8b7867a43d5dc 100644
--- a/net/minecraft/world/entity/vehicle/minecart/MinecartCommandBlock.java
+++ b/net/minecraft/world/entity/vehicle/minecart/MinecartCommandBlock.java
@@ -87,7 +87,7 @@ public class MinecartCommandBlock extends AbstractMinecart {
 
     @Override
     public InteractionResult interact(final Player player, final InteractionHand hand, final Vec3 location) {
-        if (!player.canUseGameMasterBlocks()) {
+        if (!player.canUseGameMasterBlocks() && (!player.isCreative() || !player.getBukkitEntity().hasPermission("minecraft.commandblock"))) { // Paper - command block permission
             return InteractionResult.PASS;
         }
 
@@ -125,7 +125,7 @@ public class MinecartCommandBlock extends AbstractMinecart {
                 MinecartCommandBlock.this.position(),
                 MinecartCommandBlock.this.getRotationVector(),
                 level,
-                LevelBasedPermissionSet.GAMEMASTER,
+                LevelBasedPermissionSet.forLevel(net.minecraft.server.permissions.PermissionLevel.byId(level.paperConfig().commandBlocks.permissionsLevel)), // Paper - configurable command block perm level
                 this.getName().getString(),
                 MinecartCommandBlock.this.getDisplayName(),
                 level.getServer(),
@@ -137,5 +137,17 @@ public class MinecartCommandBlock extends AbstractMinecart {
         public boolean isValid() {
             return !MinecartCommandBlock.this.isRemoved();
         }
+
+        // CraftBukkit start
+        @Override
+        public org.bukkit.command.CommandSender getBukkitSender(CommandSourceStack wrapper) {
+            return MinecartCommandBlock.this.getBukkitEntity();
+        }
+
+        @Override
+        public ServerLevel getLevel() {
+            return (ServerLevel) MinecartCommandBlock.this.level();
+        }
+        // CraftBukkit end
     }
 }
