From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/vehicle/minecart/OldMinecartBehavior.java b/net/minecraft/world/entity/vehicle/minecart/OldMinecartBehavior.java
index 77c7c2dc1cbbeeee6e6ca2038fbc6f43eb2fd26a..3c8694e75cabd68690d9139ecedc9e079db482c5 100644
--- a/net/minecraft/world/entity/vehicle/minecart/OldMinecartBehavior.java
+++ b/net/minecraft/world/entity/vehicle/minecart/OldMinecartBehavior.java
@@ -375,8 +375,22 @@ public class OldMinecartBehavior extends MinecartBehavior {
                         && !(entity instanceof AbstractMinecart)
                         && !this.minecart.isVehicle()
                         && !entity.isPassenger()) {
+                        // CraftBukkit start
+                        org.bukkit.event.vehicle.VehicleEntityCollisionEvent collisionEvent = new org.bukkit.event.vehicle.VehicleEntityCollisionEvent(
+                            (org.bukkit.entity.Vehicle) this.minecart.getBukkitEntity(), entity.getBukkitEntity()
+                        );
+                        if (!collisionEvent.callEvent()) continue;
+                        // CraftBukkit end
                         entity.startRiding(this.minecart);
                     } else {
+                        // CraftBukkit start
+                        if (!this.minecart.isPassengerOfSameVehicle(entity)) {
+                            org.bukkit.event.vehicle.VehicleEntityCollisionEvent collisionEvent = new org.bukkit.event.vehicle.VehicleEntityCollisionEvent(
+                                (org.bukkit.entity.Vehicle) this.minecart.getBukkitEntity(), entity.getBukkitEntity()
+                            );
+                            if (!collisionEvent.callEvent()) continue;
+                        }
+                        // CraftBukkit end
                         entity.push(this.minecart);
                     }
                 }
@@ -384,6 +398,12 @@ public class OldMinecartBehavior extends MinecartBehavior {
         } else {
             for (Entity entity : this.level().getEntities(this.minecart, hitbox)) {
                 if (!this.minecart.hasPassenger(entity) && entity.isPushable() && entity instanceof AbstractMinecart) {
+                    // CraftBukkit start
+                    org.bukkit.event.vehicle.VehicleEntityCollisionEvent collisionEvent = new org.bukkit.event.vehicle.VehicleEntityCollisionEvent(
+                        (org.bukkit.entity.Vehicle) this.minecart.getBukkitEntity(), entity.getBukkitEntity()
+                    );
+                    if (!collisionEvent.callEvent()) continue;
+                    // CraftBukkit end
                     entity.push(this.minecart);
                 }
             }
@@ -406,11 +426,18 @@ public class OldMinecartBehavior extends MinecartBehavior {
 
     @Override
     public double getMaxSpeed(final ServerLevel level) {
+        // CraftBukkit start
+        Double maxSpeed = this.minecart.maxSpeed;
+        if (maxSpeed != null) {
+            return this.minecart.isInWater() ? maxSpeed / 2.0 : maxSpeed;
+        }
+        // CraftBukkit end
         return this.minecart.isInWater() ? 0.2 : 0.4;
     }
 
     @Override
     public double getSlowdownFactor() {
-        return this.minecart.isVehicle() ? 0.997 : 0.96;
+        if (this.minecart.frictionState == net.kyori.adventure.util.TriState.FALSE) return 1; // Paper
+        return this.minecart.isVehicle() || !this.minecart.slowWhenEmpty ? 0.997 : 0.96; // CraftBukkit - add !this.slowWhenEmpty
     }
 }
