From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/navigation/WallClimberNavigation.java b/net/minecraft/world/entity/ai/navigation/WallClimberNavigation.java
index e7697b583c3395b24a9c4214b3d09259e92c9936..2b5ba8f01b57b5e54bba4a8624f17e1b550d818e 100644
--- a/net/minecraft/world/entity/ai/navigation/WallClimberNavigation.java
+++ b/net/minecraft/world/entity/ai/navigation/WallClimberNavigation.java
@@ -15,9 +15,9 @@ public class WallClimberNavigation extends GroundPathNavigation {
     }
 
     @Override
-    public Path createPath(final BlockPos pos, final int reachRange) {
+    public Path createPath(final BlockPos pos, @Nullable Entity entity, final int reachRange) {
         this.pathToPosition = pos;
-        return super.createPath(pos, reachRange);
+        return super.createPath(pos, entity, reachRange); // Paper - EntityPathfindEvent
     }
 
     @Override
