From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/animal/fish/AbstractSchoolingFish.java b/net/minecraft/world/entity/animal/fish/AbstractSchoolingFish.java
index 1d610be4f2cd75bf592183b334ca699b490c991f..2811ac180e6b3ea991f052bf959d8ad6e33242ae 100644
--- a/net/minecraft/world/entity/animal/fish/AbstractSchoolingFish.java
+++ b/net/minecraft/world/entity/animal/fish/AbstractSchoolingFish.java
@@ -50,6 +50,7 @@ public abstract class AbstractSchoolingFish extends AbstractFish {
     }
 
     public void stopFollowing() {
+        if (this.leader == null) return; // Paper - Avoid NPE, plugins can now set the leader and certain fish goals might cause this method to be called
         this.leader.removeFollower();
         this.leader = null;
     }
