From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/animal/frog/Tadpole.java b/net/minecraft/world/entity/animal/frog/Tadpole.java
index d392a29788b280da32117acad3d092911cabbc22..296cae1d5f5a3a054949ee03e2a2552bfeae4682 100644
--- a/net/minecraft/world/entity/animal/frog/Tadpole.java
+++ b/net/minecraft/world/entity/animal/frog/Tadpole.java
@@ -128,7 +128,7 @@ public class Tadpole extends AbstractFish {
         entityData.define(AGE_LOCKED, false);
     }
 
-    protected void setAgeLocked(final boolean locked) {
+    public void setAgeLocked(final boolean locked) { // Paper - public
         this.entityData.set(AGE_LOCKED, locked);
     }
 
@@ -158,6 +158,11 @@ public class Tadpole extends AbstractFish {
             this.feed(player, itemStack);
             return InteractionResult.SUCCESS;
         } else if (AgeableMob.canUseGoldenDandelion(itemStack, true, this.ageLockParticleTimer, this)) {
+            // Paper start - call PlayerToggleEntityAgeLockEvent
+            if (!org.bukkit.craftbukkit.event.CraftEventFactory.callPlayerToggleEntityAgeLockEvent(player, this, itemStack, hand, !this.isAgeLocked(), AGE_LOCKED)) {
+                return InteractionResult.SUCCESS;
+            }
+            // Paper end - call PlayerToggleEntityAgeLockEvent
             AgeableMob.setAgeLocked(this, this::isAgeLocked, player, itemStack, mob -> this.setAgeLockedData());
             return InteractionResult.SUCCESS;
         } else {
@@ -237,12 +242,17 @@ public class Tadpole extends AbstractFish {
 
     private void ageUp() {
         if (this.level() instanceof ServerLevel serverLevel) {
-            this.convertTo(EntityTypes.FROG, ConversionParams.single(this, false, false), frog -> {
+            Frog converted = this.convertTo(EntityTypes.FROG, ConversionParams.single(this, false, false), frog -> { // CraftBukkit
                 frog.finalizeSpawn(serverLevel, serverLevel.getCurrentDifficultyAt(frog.blockPosition()), EntitySpawnReason.CONVERSION, null);
                 frog.setPersistenceRequired();
                 frog.fudgePositionAfterSizeChange(this.getDimensions(this.getPose()));
                 this.playSound(SoundEvents.TADPOLE_GROW_UP, 0.15F, 1.0F);
-            });
+            // CraftBukkit start
+            }, org.bukkit.event.entity.EntityTransformEvent.TransformReason.METAMORPHOSIS, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.METAMORPHOSIS);
+            if (converted == null) {
+                this.setAge(0); // Sets the age to 0 for avoid a loop if the event is canceled
+            }
+            // CraftBukkit end
         }
     }
 
