From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/players/BanListEntry.java b/net/minecraft/server/players/BanListEntry.java
index ff07e0e3a538588a47c4aae5fadd41a9a5588a10..a3aa9128aeb17e9148a149e93fe514c7661a9e21 100644
--- a/net/minecraft/server/players/BanListEntry.java
+++ b/net/minecraft/server/players/BanListEntry.java
@@ -28,7 +28,7 @@ public abstract class BanListEntry<T> extends StoredUserEntry<T> {
     }
 
     protected BanListEntry(final @Nullable T user, final JsonObject object) {
-        super(user);
+        super(BanListEntry.checkExpiry(user, object)); // CraftBukkit
 
         Date created;
         try {
@@ -106,4 +106,22 @@ public abstract class BanListEntry<T> extends StoredUserEntry<T> {
     public int hashCode() {
         return Objects.hash(this.source, this.expires, this.reason, this.getUser());
     }
+
+    // CraftBukkit start
+    private static <T> T checkExpiry(T object, JsonObject jsonobject) {
+        Date expires = null;
+
+        try {
+            expires = jsonobject.has("expires") ? BanListEntry.DATE_FORMAT.parse(jsonobject.get("expires").getAsString()) : null;
+        } catch (ParseException ex) {
+            // Guess we don't have a date
+        }
+
+        if (expires == null || expires.after(new Date())) {
+            return object;
+        } else {
+            return null;
+        }
+    }
+    // CraftBukkit end
 }
