From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/players/NameAndId.java b/net/minecraft/server/players/NameAndId.java
index 5635924e55038549a1bb47aa2344cebe8ad52a31..fb2c627a74e37f1270274d93d87f1262547dfd59 100644
--- a/net/minecraft/server/players/NameAndId.java
+++ b/net/minecraft/server/players/NameAndId.java
@@ -48,4 +48,10 @@ public record NameAndId(UUID id, String name) {
         UUID id = UUIDUtil.createOfflinePlayerUUID(name);
         return new NameAndId(id, name);
     }
+
+    // Paper start - utility method for common conversion back to the game profile
+    public GameProfile toUncompletedGameProfile() {
+        return new GameProfile(this.id, this.name);
+    }
+    // Paper end - utility method for common conversion back to the game profile
 }
