From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/players/StoredUserList.java b/net/minecraft/server/players/StoredUserList.java
index 01dac844fbf6cc3463322d3fafb6ff074bf151d9..eaf8fb8a9cf94d0d16debf90e5ceda3f91c979ef 100644
--- a/net/minecraft/server/players/StoredUserList.java
+++ b/net/minecraft/server/players/StoredUserList.java
@@ -28,7 +28,7 @@ public abstract class StoredUserList<K, V extends StoredUserEntry<K>> {
     private static final Logger LOGGER = LogUtils.getLogger();
     private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
     private final File file;
-    private final Map<String, V> map = Maps.newHashMap();
+    private final Map<String, V> map = Maps.newConcurrentMap(); // Paper - Use ConcurrentHashMap in StoredUserList
     protected final NotificationService notificationService;
 
     public StoredUserList(final File file, final NotificationService notificationService) {
@@ -59,8 +59,11 @@ public abstract class StoredUserList<K, V extends StoredUserEntry<K>> {
     }
 
     public @Nullable V get(final K user) {
-        this.removeExpired();
-        return this.map.get(this.getKeyForUser(user));
+        // Paper start - Use ConcurrentHashMap in StoredUserList
+        return this.map.computeIfPresent(this.getKeyForUser(user), (key, value) -> {
+            return value.hasExpired() ? null : value;
+        });
+        // Paper end - Use ConcurrentHashMap in StoredUserList
     }
 
     public boolean remove(final K user) {
@@ -105,22 +108,15 @@ public abstract class StoredUserList<K, V extends StoredUserEntry<K>> {
     }
 
     protected boolean contains(final K user) {
+        this.removeExpired(); // CraftBukkit - SPIGOT-7589: Consistently remove expired entries to mirror .get(...)
         return this.map.containsKey(this.getKeyForUser(user));
     }
 
+    // Paper start - Use ConcurrentHashMap in StoredUserList
     private void removeExpired() {
-        List<K> toRemove = Lists.newArrayList();
-
-        for (V entry : this.map.values()) {
-            if (entry.hasExpired()) {
-                toRemove.add(entry.getUser());
-            }
-        }
-
-        for (K user : toRemove) {
-            this.map.remove(this.getKeyForUser(user));
-        }
+        this.map.values().removeIf(StoredUserEntry::hasExpired);
     }
+    // Paper end - Use ConcurrentHashMap in StoredUserList
 
     protected abstract StoredUserEntry<K> createEntry(final JsonObject object);
 
@@ -129,6 +125,7 @@ public abstract class StoredUserList<K, V extends StoredUserEntry<K>> {
     }
 
     public void save() throws IOException {
+        this.removeExpired(); // Paper - remove expired values before saving
         JsonArray result = new JsonArray();
         this.map.values().stream().map(entry -> Util.make(new JsonObject(), entry::serialize)).forEach(result::add);
 
@@ -153,7 +150,14 @@ public abstract class StoredUserList<K, V extends StoredUserEntry<K>> {
                         this.map.put(this.getKeyForUser(entry.getUser()), (V)entry);
                     }
                 }
+            // Spigot start
+            } catch (com.google.gson.JsonParseException | NullPointerException ex) {
+                File backup = new File(this.file + ".backup");
+                LOGGER.warn("Unable to read file {}, backing it up to {} and creating new copy.", this.file, backup, ex);
+                this.file.renameTo(backup);
+                this.file.delete();
             }
+            // Spigot end
         }
     }
 }
