From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/players/UserWhiteList.java b/net/minecraft/server/players/UserWhiteList.java
index f23fd18a82f25f67f1651de71350261ad1a9083f..e42147dca81ef47e15e7407b6136a7121cac3c64 100644
--- a/net/minecraft/server/players/UserWhiteList.java
+++ b/net/minecraft/server/players/UserWhiteList.java
@@ -21,6 +21,11 @@ public class UserWhiteList extends StoredUserList<NameAndId, UserWhiteListEntry>
 
     @Override
     public boolean add(final UserWhiteListEntry infos) {
+        // Paper start - Add whitelist events
+        if (!new io.papermc.paper.event.server.WhitelistStateUpdateEvent(com.destroystokyo.paper.profile.CraftPlayerProfile.asBukkitCopy(infos.getUser().toUncompletedGameProfile()), io.papermc.paper.event.server.WhitelistStateUpdateEvent.WhitelistStatus.ADDED).callEvent()) {
+            return false;
+        }
+        // Paper end - Add whitelist events
         if (super.add(infos)) {
             if (infos.getUser() != null) {
                 this.notificationService.playerAddedToAllowlist(infos.getUser());
@@ -34,6 +39,11 @@ public class UserWhiteList extends StoredUserList<NameAndId, UserWhiteListEntry>
 
     @Override
     public boolean remove(final NameAndId user) {
+        // Paper start - Add whitelist events
+        if (!new io.papermc.paper.event.server.WhitelistStateUpdateEvent(com.destroystokyo.paper.profile.CraftPlayerProfile.asBukkitCopy(user.toUncompletedGameProfile()), io.papermc.paper.event.server.WhitelistStateUpdateEvent.WhitelistStatus.REMOVED).callEvent()) {
+            return false;
+        }
+        // Paper end - Add whitelist events
         if (super.remove(user)) {
             this.notificationService.playerRemovedFromAllowlist(user);
             return true;
