From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/gui/StatsComponent.java b/net/minecraft/server/gui/StatsComponent.java
index e81029f5b07712507e0fafe15fe661aaa4243fec..e9f53520d99454536f32bb491ed1ef6b2a5ee07a 100644
--- a/net/minecraft/server/gui/StatsComponent.java
+++ b/net/minecraft/server/gui/StatsComponent.java
@@ -32,12 +32,21 @@ public class StatsComponent extends JComponent {
 
     private void tick() {
         long usedRam = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory();
+        // Paper start - Improve ServerGUI
+        double[] tps = org.bukkit.Bukkit.getTPS();
+        String[] tpsAvg = new String[tps.length];
+
+        for (int g = 0; g < tps.length; g++) {
+            tpsAvg[g] = format(tps[g]);
+        }
         this.msgs[0] = "Memory use: "
             + usedRam / 1024L / 1024L
             + " mb ("
             + Runtime.getRuntime().freeMemory() * 100L / Runtime.getRuntime().maxMemory()
             + "% free)";
         this.msgs[1] = "Avg tick: " + DECIMAL_FORMAT.format((double)this.server.getAverageTickTimeNanos() / TimeUtil.NANOSECONDS_PER_MILLISECOND) + " ms";
+        this.msgs[2] = "TPS from last 1m, 5m, 15m: " + String.join(", ", tpsAvg);
+        // Paper end - Improve ServerGUI
         this.values[this.vp++ & 0xFF] = (int)(usedRam * 100L / Runtime.getRuntime().maxMemory());
         this.repaint();
     }
@@ -66,4 +75,10 @@ public class StatsComponent extends JComponent {
     public void close() {
         this.timer.stop();
     }
+
+    // Paper start - Improve ServerGUI
+    private static String format(double tps) {
+        return ((tps > 21.0) ? "*" : "") + Math.min(Math.round(tps * 100.0) / 100.0, 20.0); // only print * at 21, we commonly peak to 20.02 as the tick sleep is not accurate enough, stop the noise
+    }
+    // Paper end - Improve ServerGUI
 }
