From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/effect/HealOrHarmMobEffect.java b/net/minecraft/world/effect/HealOrHarmMobEffect.java
index ef31d92c1158ba67f780182eaeb1cea17fb99bd1..f5e89c81a4aae3cbbe85e0ebb0ac1adfc51f0de3 100644
--- a/net/minecraft/world/effect/HealOrHarmMobEffect.java
+++ b/net/minecraft/world/effect/HealOrHarmMobEffect.java
@@ -16,7 +16,7 @@ class HealOrHarmMobEffect extends InstantaneousMobEffect {
     @Override
     public boolean applyEffectTick(final ServerLevel level, final LivingEntity mob, final int amplification) {
         if (this.isHarm == mob.isInvertedHealAndHarm()) {
-            mob.heal(Math.max(4 << amplification, 0));
+            mob.heal(Math.max(4 << amplification, 0), org.bukkit.event.entity.EntityRegainHealthEvent.RegainReason.MAGIC); // CraftBukkit
         } else {
             mob.hurtServer(level, mob.damageSources().magic(), 6 << amplification);
         }
@@ -33,9 +33,10 @@ class HealOrHarmMobEffect extends InstantaneousMobEffect {
         final int amplification,
         final double scale
     ) {
+        if (!new io.papermc.paper.event.entity.EntityEffectTickEvent(mob.getBukkitLivingEntity(), org.bukkit.craftbukkit.potion.CraftPotionEffectType.minecraftToBukkit(this), amplification).callEvent()) { return; } // Paper - Add EntityEffectTickEvent
         if (this.isHarm == mob.isInvertedHealAndHarm()) {
             int amount = (int)(scale * (4 << amplification) + 0.5);
-            mob.heal(amount);
+            mob.heal(amount, org.bukkit.event.entity.EntityRegainHealthEvent.RegainReason.MAGIC); // CraftBukkit
         } else {
             int amount = (int)(scale * (6 << amplification) + 0.5);
             if (source == null) {
