From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/effect/PoisonMobEffect.java b/net/minecraft/world/effect/PoisonMobEffect.java
index b12590c3384b020b18af9d85ea5f82fa83fb27d4..52b93a06bb657d50fa21caf9639305d2ef8b29a7 100644
--- a/net/minecraft/world/effect/PoisonMobEffect.java
+++ b/net/minecraft/world/effect/PoisonMobEffect.java
@@ -13,7 +13,7 @@ public class PoisonMobEffect extends MobEffect {
     @Override
     public boolean applyEffectTick(final ServerLevel level, final LivingEntity mob, final int amplification) {
         if (mob.getHealth() > 1.0F) {
-            mob.hurtServer(level, mob.damageSources().magic(), 1.0F);
+            mob.hurtServer(level, mob.damageSources().magic().knownCause(org.bukkit.event.entity.EntityDamageEvent.DamageCause.POISON), 1.0F); // CraftBukkit
         }
 
         return true;
