From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/sensing/Sensor.java b/net/minecraft/world/entity/ai/sensing/Sensor.java
index eeee386e6ea664ee96e5d943e03c7fb2d8c874b3..745f26d36a30a9f570bc989b97002c2723db4ba8 100644
--- a/net/minecraft/world/entity/ai/sensing/Sensor.java
+++ b/net/minecraft/world/entity/ai/sensing/Sensor.java
@@ -28,8 +28,17 @@ public abstract class Sensor<E extends LivingEntity> {
         .ignoreInvisibilityTesting();
     private final int scanRate;
     private long timeToTick;
+    private final String configKey; // Paper - configurable sensor tick rate and timings
 
     public Sensor(final int scanRate) {
+        // Paper start - configurable sensor tick rate and timings
+        String key = this.getClass().getName();
+        int lastSeparator = key.lastIndexOf('.');
+        if (lastSeparator != -1) {
+            key = key.substring(lastSeparator + 1);
+        }
+        this.configKey = key.toLowerCase(java.util.Locale.ROOT);
+        // Paper end
         this.scanRate = scanRate;
     }
 
@@ -43,7 +52,7 @@ public abstract class Sensor<E extends LivingEntity> {
 
     public final void tick(final ServerLevel level, final E body) {
         if (--this.timeToTick <= 0L) {
-            this.timeToTick = this.scanRate;
+            this.timeToTick = java.util.Objects.requireNonNullElse(level.paperConfig().tickRates.sensor.get(body.getType(), this.configKey), this.scanRate); // Paper - configurable sensor tick rate and timings
             this.updateTargetingConditionRanges(body);
             this.doTick(level, body);
         }
