From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/BellBlock.java b/net/minecraft/world/level/block/BellBlock.java
index 738a80ffc652a0e7540995fb7cda1f0c6280b5dc..c9cf4e7ae4235ce744163abb369f6f17b3092a05 100644
--- a/net/minecraft/world/level/block/BellBlock.java
+++ b/net/minecraft/world/level/block/BellBlock.java
@@ -74,6 +74,11 @@ public class BellBlock extends BaseEntityBlock {
     ) {
         boolean signal = level.hasNeighborSignal(pos);
         if (signal != state.getValue(POWERED)) {
+            // Paper start - Call BlockRedstoneEvent
+            if (!org.bukkit.craftbukkit.event.CraftEventFactory.callBinaryRedstoneChange(level, pos, signal)) {
+                return;
+            }
+            // Paper end - Call BlockRedstoneEvent
             if (signal) {
                 this.attemptToRing(level, pos, null);
             }
@@ -139,6 +144,11 @@ public class BellBlock extends BaseEntityBlock {
                 direction = level.getBlockState(pos).getValue(FACING);
             }
 
+            // CraftBukkit start
+            if (!org.bukkit.craftbukkit.event.CraftEventFactory.handleBellRingEvent(level, pos, direction, ringingEntity)) {
+                return false;
+            }
+            // CraftBukkit end
             bellBlockEntity.onHit(direction);
             level.playSound(null, pos, SoundEvents.BELL_BLOCK, SoundSource.BLOCKS, 2.0F, 1.0F);
             level.gameEvent(ringingEntity, GameEvent.BLOCK_CHANGE, pos);
