From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/CampfireBlock.java b/net/minecraft/world/level/block/CampfireBlock.java
index 575126b685aa5a70b0c80eeb9cf8f19c4ca868d7..9ebb0ed771b1a80beca34390c1b48b9f0d054bfc 100644
--- a/net/minecraft/world/level/block/CampfireBlock.java
+++ b/net/minecraft/world/level/block/CampfireBlock.java
@@ -116,8 +116,9 @@ public class CampfireBlock extends BaseEntityBlock implements SimpleWaterloggedB
         final InsideBlockEffectApplier effectApplier,
         final boolean isPrecise
     ) {
+        if (!new io.papermc.paper.event.entity.EntityInsideBlockEvent(entity.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.at(level, pos)).callEvent()) { return; } // Paper - Add EntityInsideBlockEvent
         if (state.getValue(LIT) && entity instanceof LivingEntity) {
-            entity.hurt(level.damageSources().campfire(), this.fireDamage);
+            entity.hurt(level.damageSources().campfire().eventBlockDamager(level, pos), this.fireDamage); // CraftBukkit
         }
 
         super.entityInside(state, level, pos, entity, effectApplier, isPrecise);
@@ -228,6 +229,11 @@ public class CampfireBlock extends BaseEntityBlock implements SimpleWaterloggedB
             && projectile.mayInteract(serverLevel, pos)
             && !state.getValue(LIT)
             && !state.getValue(WATERLOGGED)) {
+            // CraftBukkit start
+            if (org.bukkit.craftbukkit.event.CraftEventFactory.callBlockIgniteEvent(level, pos, projectile).isCancelled()) {
+                return;
+            }
+            // CraftBukkit end
             level.setBlock(pos, state.setValue(BlockStateProperties.LIT, true), Block.UPDATE_ALL_IMMEDIATE);
         }
     }
