From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/CaveVinesBlock.java b/net/minecraft/world/level/block/CaveVinesBlock.java
index ed8e910f14398347eb203d1288336cae80e97fbd..8cb4adda185a229a9f036355d3984187f60d7585 100644
--- a/net/minecraft/world/level/block/CaveVinesBlock.java
+++ b/net/minecraft/world/level/block/CaveVinesBlock.java
@@ -45,9 +45,17 @@ public class CaveVinesBlock extends GrowingPlantHeadBlock implements CaveVines {
 
     @Override
     protected BlockState getGrowIntoState(final BlockState growFromState, final RandomSource random) {
-        return super.getGrowIntoState(growFromState, random).setValue(BERRIES, random.nextFloat() < 0.11F);
+        // Paper start - Fix Spigot growth modifiers
+        return this.getGrowIntoState(growFromState, random, null);
     }
 
+    @Override
+    protected BlockState getGrowIntoState(BlockState growFromState, RandomSource random, @org.jspecify.annotations.Nullable Level level) {
+        final boolean value = random.nextFloat() < (level != null ? (0.11F * (level.spigotConfig.glowBerryModifier / 100.0F)) : 0.11F);
+        return super.getGrowIntoState(growFromState, random).setValue(CaveVinesBlock.BERRIES, value);
+    }
+    // Paper end - Fix Spigot growth modifiers
+
     @Override
     protected ItemStack getCloneItemStack(final LevelReader level, final BlockPos pos, final BlockState state, final boolean includeData) {
         return new ItemStack(Items.GLOW_BERRIES);
