From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/ChestBlock.java b/net/minecraft/world/level/block/ChestBlock.java
index f35c255bb1e79b8550be330ab43002f363f113b3..39b5e9f8bfe070296aa7364668e40ca429d2ee08 100644
--- a/net/minecraft/world/level/block/ChestBlock.java
+++ b/net/minecraft/world/level/block/ChestBlock.java
@@ -85,7 +85,7 @@ public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements
         @Override
         public Optional<MenuProvider> acceptDouble(final ChestBlockEntity first, final ChestBlockEntity second) {
             final Container container = new CompoundContainer(first, second);
-            return Optional.of(new MenuProvider() {
+            return Optional.of(org.bukkit.craftbukkit.inventory.CraftInventoryDoubleChest.Provider.wrap(new MenuProvider() { // CraftBukkit - wrap for identification
                 @Override
                 public @Nullable AbstractContainerMenu createMenu(final int containerId, final Inventory inventory, final Player player) {
                     if (first.canOpen(player) && second.canOpen(player)) {
@@ -106,10 +106,10 @@ public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements
                     if (first.hasCustomName()) {
                         return first.getDisplayName();
                     } else {
-                        return second.hasCustomName() ? second.getDisplayName() : Component.translatable("container.chestDouble");
+                        return second.hasCustomName() ? second.getDisplayName() : Component.translatable("container.chestDouble"); // Paper - diff on change - CraftDoubleChestInventoryViewBuilder.defaultTitle
                     }
                 }
-            });
+            }, (CompoundContainer) container)); // CraftBukkit - wrap for identification
         }
 
         @Override
@@ -247,8 +247,7 @@ public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements
     ) {
         if (level instanceof ServerLevel serverLevel) {
             MenuProvider menuProvider = this.getMenuProvider(state, level, pos);
-            if (menuProvider != null) {
-                player.openMenu(menuProvider);
+            if (menuProvider != null && player.openMenu(menuProvider).isPresent()) { // Paper - Fix InventoryOpenEvent cancellation
                 player.awardStat(this.getOpenChestStat());
                 PiglinAi.angerNearbyPiglins(serverLevel, player, true);
             }
@@ -289,7 +288,13 @@ public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements
 
     @Override
     public @Nullable MenuProvider getMenuProvider(final BlockState state, final Level level, final BlockPos pos) {
-        return this.combine(state, level, pos, false).apply(MENU_PROVIDER_COMBINER).orElse(null);
+    // CraftBukkit start
+        return this.getMenuProvider(state, level, pos, false);
+    }
+
+    public @Nullable MenuProvider getMenuProvider(final BlockState state, final Level level, final BlockPos pos, final boolean ignoreObstructions) {
+        return this.combine(state, level, pos, ignoreObstructions).apply(MENU_PROVIDER_COMBINER).orElse(null);
+    // CraftBukkit end
     }
 
     public static DoubleBlockCombiner.Combiner<ChestBlockEntity, Float2FloatFunction> opennessCombiner(final LidBlockEntity entity) {
@@ -331,6 +336,11 @@ public class ChestBlock extends AbstractChestBlock<ChestBlockEntity> implements
     }
 
     private static boolean isCatSittingOnChest(final LevelAccessor level, final BlockPos pos) {
+        // Paper start - Option to disable chest cat detection
+        if (level.getMinecraftWorld().paperConfig().entities.behavior.disableChestCatDetection) {
+            return false;
+        }
+        // Paper end - Option to disable chest cat detection
         List<Cat> cats = level.getEntitiesOfClass(Cat.class, new AABB(pos.getX(), pos.getY() + 1, pos.getZ(), pos.getX() + 1, pos.getY() + 2, pos.getZ() + 1));
         if (!cats.isEmpty()) {
             for (Cat cat : cats) {
