From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/ComparatorBlock.java b/net/minecraft/world/level/block/ComparatorBlock.java
index e8dfec5e7eaf7aa3861fefeb640a2ce6c4603661..b867f52d98f127269864775dbeb2481cbbd07a26 100644
--- a/net/minecraft/world/level/block/ComparatorBlock.java
+++ b/net/minecraft/world/level/block/ComparatorBlock.java
@@ -164,8 +164,18 @@ public class ComparatorBlock extends DiodeBlock implements EntityBlock {
             boolean sourceOn = this.shouldTurnOn(level, pos, state);
             boolean isOn = state.getValue(POWERED);
             if (isOn && !sourceOn) {
+                // Paper start - Call BlockRedstoneEvent
+                if (!org.bukkit.craftbukkit.event.CraftEventFactory.callBinaryRedstoneChange(level, pos, false)) {
+                    return;
+                }
+                // Paper end - Call BlockRedstoneEvent
                 level.setBlock(pos, state.setValue(POWERED, false), Block.UPDATE_CLIENTS);
             } else if (!isOn && sourceOn) {
+                // Paper start - Call BlockRedstoneEvent
+                if (!org.bukkit.craftbukkit.event.CraftEventFactory.callBinaryRedstoneChange(level, pos, true)) {
+                    return;
+                }
+                // Paper end - Call BlockRedstoneEvent
                 level.setBlock(pos, state.setValue(POWERED, true), Block.UPDATE_CLIENTS);
             }
 
