From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/DispenserBlock.java b/net/minecraft/world/level/block/DispenserBlock.java
index cb4c22c5bd1c9d29610607d0522cf483041dad1e..f777b67e0d90c1bb25c3aeef8e5b1710ab8ac96e 100644
--- a/net/minecraft/world/level/block/DispenserBlock.java
+++ b/net/minecraft/world/level/block/DispenserBlock.java
@@ -70,8 +70,7 @@ public class DispenserBlock extends BaseEntityBlock {
     protected InteractionResult useWithoutItem(
         final BlockState state, final Level level, final BlockPos pos, final Player player, final BlockHitResult hitResult
     ) {
-        if (!level.isClientSide() && level.getBlockEntity(pos) instanceof DispenserBlockEntity dispenser) {
-            player.openMenu(dispenser);
+        if (!level.isClientSide() && level.getBlockEntity(pos) instanceof DispenserBlockEntity dispenser && player.openMenu(dispenser).isPresent()) { // Paper - Fix InventoryOpenEvent cancellation
             player.awardStat(dispenser instanceof DropperBlockEntity ? Stats.INSPECT_DROPPER : Stats.INSPECT_DISPENSER);
         }
 
@@ -86,18 +85,27 @@ public class DispenserBlock extends BaseEntityBlock {
             BlockSource source = new BlockSource(level, pos, state, blockEntity);
             int slot = blockEntity.getRandomSlot(level.getRandom());
             if (slot < 0) {
+                if (org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockFailedDispenseEvent(level, pos)) { // Paper - Add BlockFailedDispenseEvent
                 level.levelEvent(LevelEvent.SOUND_DISPENSER_FAIL, pos, 0);
                 level.gameEvent(GameEvent.BLOCK_ACTIVATE, pos, GameEvent.Context.of(blockEntity.getBlockState()));
+                } // Paper - Add BlockFailedDispenseEvent
             } else {
                 ItemStack itemStack = blockEntity.getItem(slot);
                 DispenseItemBehavior behavior = this.getDispenseMethod(level, itemStack);
                 if (behavior != DispenseItemBehavior.NOOP) {
+                    if (!org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockPreDispenseEvent(level, pos, itemStack, slot)) return; // Paper - Add BlockPreDispenseEvent
                     blockEntity.setItem(slot, behavior.dispense(source, itemStack));
                 }
             }
         }
     }
 
+    // Paper start - Fix NPE with equippable and items without behavior
+    public static DispenseItemBehavior getDispenseBehavior(BlockSource pointer, ItemStack stack) {
+        return ((DispenserBlock) pointer.state().getBlock()).getDispenseMethod(pointer.level(), stack);
+    }
+    // Paper end - Fix NPE with equippable and items without behavior
+
     protected DispenseItemBehavior getDispenseMethod(final Level level, final ItemStack itemStack) {
         if (!itemStack.isItemEnabled(level.enabledFeatures())) {
             return DEFAULT_BEHAVIOR;
