From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/DoublePlantBlock.java b/net/minecraft/world/level/block/DoublePlantBlock.java
index 3854dd7775898ad4bf193a4cb4e85a978484d169..a65b487e8c32deb0f931815ca7d8e31ce42a5e67 100644
--- a/net/minecraft/world/level/block/DoublePlantBlock.java
+++ b/net/minecraft/world/level/block/DoublePlantBlock.java
@@ -114,11 +114,17 @@ public class DoublePlantBlock extends VegetationBlock {
         final BlockState state,
         final @Nullable BlockEntity blockEntity,
         final ItemStack destroyedWith
+        , boolean includeDrops, boolean dropExp // Paper - fix drops not preventing stats/food exhaustion
     ) {
-        super.playerDestroy(level, player, pos, Blocks.AIR.defaultBlockState(), blockEntity, destroyedWith);
+        super.playerDestroy(level, player, pos, Blocks.AIR.defaultBlockState(), blockEntity, destroyedWith, includeDrops, dropExp); // Paper - fix drops not preventing stats/food exhaustion;
     }
 
     protected static void preventDropFromBottomPart(final Level level, final BlockPos pos, final BlockState state, final Player player) {
+        // CraftBukkit start
+        if (((net.minecraft.server.level.ServerLevel)level).hasPhysicsEvent && org.bukkit.craftbukkit.event.CraftEventFactory.callBlockPhysicsEvent(level, pos).isCancelled()) { // Paper
+            return;
+        }
+        // CraftBukkit end
         DoubleBlockHalf part = state.getValue(HALF);
         if (part == DoubleBlockHalf.UPPER) {
             BlockPos bottomPos = pos.below();
