From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/DoubleBlockCombiner.java b/net/minecraft/world/level/block/DoubleBlockCombiner.java
index c934c7180e079d6c0035f9e6c1a0ccd7e7c868d7..560d63ecfef1d65d6fe1e01b8e273f4b2d609304 100644
--- a/net/minecraft/world/level/block/DoubleBlockCombiner.java
+++ b/net/minecraft/world/level/block/DoubleBlockCombiner.java
@@ -38,7 +38,12 @@ public class DoubleBlockCombiner {
         }
 
         BlockPos neighborPos = pos.relative(connectionResolver.apply(state));
-        BlockState neighbourState = level.getBlockState(neighborPos);
+        // Paper start - Don't load Chunks from Hoppers and other things
+        BlockState neighbourState = level.getBlockStateIfLoaded(neighborPos);
+        if (neighbourState == null) {
+            return new DoubleBlockCombiner.NeighborCombineResult.Single<>(blockEntity);
+        }
+        // Paper end - Don't load Chunks from Hoppers and other things
         if (neighbourState.is(state.getBlock())) {
             DoubleBlockCombiner.BlockType neighbourType = typeResolver.apply(neighbourState);
             if (neighbourType != DoubleBlockCombiner.BlockType.SINGLE
