From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/EnderChestBlock.java b/net/minecraft/world/level/block/EnderChestBlock.java
index afb24368d2a6b52ff946154339fcdd4cffcff414..f1a5b9670dd04b800a8d755f32364458a16d924a 100644
--- a/net/minecraft/world/level/block/EnderChestBlock.java
+++ b/net/minecraft/world/level/block/EnderChestBlock.java
@@ -83,13 +83,16 @@ public class EnderChestBlock extends AbstractChestBlock<EnderChestBlockEntity> i
         PlayerEnderChestContainer container = player.getEnderChestInventory();
         if (container != null && level.getBlockEntity(pos) instanceof EnderChestBlockEntity enderChest) {
             BlockPos above = pos.above();
-            if (level.getBlockState(above).isRedstoneConductor(level, above)) {
+            if (level.getBlockState(above).isRedstoneConductor(level, above)) { // Paper - diff on change; make sure that EnderChest#isBlocked uses the same logic
                 return InteractionResult.SUCCESS;
             }
 
-            if (level instanceof ServerLevel serverLevel) {
-                container.setActiveChest(enderChest);
-                player.openMenu(new SimpleMenuProvider((containerId, inventory, p) -> ChestMenu.threeRows(containerId, inventory, container), CONTAINER_TITLE));
+            // Paper start - Fix InventoryOpenEvent cancellation - moved up;
+            container.setActiveChest(enderChest); // Needs to happen before ChestMenu.threeRows as it is required for opening animations
+            if (level instanceof ServerLevel serverLevel && player.openMenu(
+                new SimpleMenuProvider((containerId, inventory, p) -> ChestMenu.threeRows(containerId, inventory, container), CONTAINER_TITLE)
+            ).isPresent()) {
+            // Paper end - Fix InventoryOpenEvent cancellation - moved up;
                 player.awardStat(Stats.OPEN_ENDERCHEST);
                 PiglinAi.angerNearbyPiglins(serverLevel, player, true);
             }
