From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/EnderChestBlock.java b/net/minecraft/world/level/block/EnderChestBlock.java
index 0cb1a6661f5b25a7418475e952022e6dce7956ba..34e6bb7d246f065698def3b1f7a0d237f304f2aa 100644
--- a/net/minecraft/world/level/block/EnderChestBlock.java
+++ b/net/minecraft/world/level/block/EnderChestBlock.java
@@ -76,13 +76,16 @@ public class EnderChestBlock extends AbstractChestBlock<EnderChestBlockEntity> i
         PlayerEnderChestContainer container = player.getEnderChestInventory();
         if (container != null && level.getBlockEntity(pos) instanceof EnderChestBlockEntity enderChest) {
             BlockPos above = pos.above();
-            if (level.getBlockState(above).isRedstoneConductor(level, above)) {
+            if (level.getBlockState(above).isRedstoneConductor(level, above)) { // Paper - diff on change; make sure that EnderChest#isBlocked uses the same logic
                 return InteractionResult.SUCCESS;
             }
 
-            if (level instanceof ServerLevel serverLevel) {
-                container.setActiveChest(enderChest);
-                player.openMenu(new SimpleMenuProvider((containerId, inventory, p) -> ChestMenu.threeRows(containerId, inventory, container), CONTAINER_TITLE));
+            // Paper start - Fix InventoryOpenEvent cancellation - moved up;
+            container.setActiveChest(enderChest); // Needs to happen before ChestMenu.threeRows as it is required for opening animations
+            if (level instanceof ServerLevel serverLevel && player.openMenu(
+                new SimpleMenuProvider((containerId, inventory, p) -> ChestMenu.threeRows(containerId, inventory, container), CONTAINER_TITLE)
+            ).isPresent()) {
+            // Paper end - Fix InventoryOpenEvent cancellation - moved up;
                 player.awardStat(Stats.OPEN_ENDERCHEST);
                 PiglinAi.angerNearbyPiglins(serverLevel, player, true);
             }
