From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/EndGatewayBlock.java b/net/minecraft/world/level/block/EndGatewayBlock.java
index a3a18c34163ca625c8b5eecdcd1f6a83e5158184..67941c19d97a328dfce7eeecf52518349f27df81 100644
--- a/net/minecraft/world/level/block/EndGatewayBlock.java
+++ b/net/minecraft/world/level/block/EndGatewayBlock.java
@@ -94,10 +94,15 @@ public class EndGatewayBlock extends BaseEntityBlock implements Portal {
         final InsideBlockEffectApplier effectApplier,
         final boolean isPrecise
     ) {
+        if (!new io.papermc.paper.event.entity.EntityInsideBlockEvent(entity.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.at(level, pos)).callEvent()) { return; } // Paper - Add EntityInsideBlockEvent
         if (entity.canUsePortal(false)
             && !level.isClientSide()
             && level.getBlockEntity(pos) instanceof TheEndGatewayBlockEntity endGatewayBlockEntity
             && !endGatewayBlockEntity.isCoolingDown()) {
+            // Paper start - call EntityPortalEnterEvent
+            org.bukkit.event.entity.EntityPortalEnterEvent event = new org.bukkit.event.entity.EntityPortalEnterEvent(entity.getBukkitEntity(), org.bukkit.craftbukkit.util.CraftLocation.toBukkit(pos, level), org.bukkit.PortalType.END_GATEWAY); // Paper - add portal type
+            if (!event.callEvent()) return;
+            // Paper end - call EntityPortalEnterEvent
             entity.setAsInsidePortal(this, pos);
             TheEndGatewayBlockEntity.triggerCooldown(level, pos, state, endGatewayBlockEntity);
         }
@@ -111,7 +116,7 @@ public class EndGatewayBlock extends BaseEntityBlock implements Portal {
                 return null;
             } else {
                 return entity instanceof ThrownEnderpearl
-                    ? new TeleportTransition(currentLevel, teleportPosition, Vec3.ZERO, 0.0F, 0.0F, Set.of(), TeleportTransition.PLACE_PORTAL_TICKET)
+                    ? new TeleportTransition(currentLevel, teleportPosition, Vec3.ZERO, 0.0F, 0.0F, Set.of(), TeleportTransition.PLACE_PORTAL_TICKET).withCause(org.bukkit.event.player.PlayerTeleportEvent.TeleportCause.END_GATEWAY) // CraftBukkit
                     : new TeleportTransition(
                         currentLevel,
                         teleportPosition,
@@ -120,7 +125,7 @@ public class EndGatewayBlock extends BaseEntityBlock implements Portal {
                         0.0F,
                         Relative.union(Relative.DELTA, Relative.ROTATION),
                         TeleportTransition.PLACE_PORTAL_TICKET
-                    );
+                    ).withCause(org.bukkit.event.player.PlayerTeleportEvent.TeleportCause.END_GATEWAY);  // CraftBukkit;
             }
         } else {
             return null;
