From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/FrogspawnBlock.java b/net/minecraft/world/level/block/FrogspawnBlock.java
index 6bf57d0d59d906f81be7ee9b6ca400e0901bbbb5..701a1c80e91dc5c0517819f3af55c1473979b039 100644
--- a/net/minecraft/world/level/block/FrogspawnBlock.java
+++ b/net/minecraft/world/level/block/FrogspawnBlock.java
@@ -99,6 +99,7 @@ public class FrogspawnBlock extends Block {
         final InsideBlockEffectApplier effectApplier,
         final boolean isPrecise
     ) {
+        if (!new io.papermc.paper.event.entity.EntityInsideBlockEvent(entity.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.at(level, pos)).callEvent()) { return; } // Paper - Add EntityInsideBlockEvent
         if (entity.is(EntityTypes.FALLING_BLOCK)) {
             this.destroyBlock(level, pos);
         }
@@ -111,6 +112,11 @@ public class FrogspawnBlock extends Block {
     }
 
     private void hatchFrogspawn(final ServerLevel level, final BlockPos pos, final RandomSource random) {
+        // Paper start - Call BlockFadeEvent
+        if (org.bukkit.craftbukkit.event.CraftEventFactory.callBlockFadeEvent(level, pos, Blocks.AIR.defaultBlockState()).isCancelled()) {
+            return;
+        }
+        // Paper end - Call BlockFadeEven
         this.destroyBlock(level, pos);
         level.playSound(null, pos, SoundEvents.FROGSPAWN_HATCH, SoundSource.BLOCKS, 1.0F, 1.0F);
         this.spawnTadpoles(level, pos, random);
@@ -131,7 +137,7 @@ public class FrogspawnBlock extends Block {
                 int yRot = random.nextInt(1, 361);
                 tadpole.snapTo(xPos, pos.getY() - 0.5, zPos, yRot, 0.0F);
                 tadpole.setPersistenceRequired();
-                level.addFreshEntity(tadpole);
+                level.addFreshEntity(tadpole, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.EGG); // Paper - use correct spawn reason
             }
         }
     }
