From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/GrowingPlantHeadBlock.java b/net/minecraft/world/level/block/GrowingPlantHeadBlock.java
index 060c487a8af28e2c6bfd1398b9e43c78aacc1baf..4fd9e6bb598afdebfda4828e9f18afe556057f63 100644
--- a/net/minecraft/world/level/block/GrowingPlantHeadBlock.java
+++ b/net/minecraft/world/level/block/GrowingPlantHeadBlock.java
@@ -44,14 +44,32 @@ public abstract class GrowingPlantHeadBlock extends GrowingPlantBlock implements
 
     @Override
     protected void randomTick(final BlockState state, final ServerLevel level, final BlockPos pos, final RandomSource random) {
-        if (state.getValue(AGE) < 25 && random.nextDouble() < this.growPerTickProbability) {
+        // Spigot start
+        int modifier = 100;
+        if (this == Blocks.KELP) {
+            modifier = level.spigotConfig.kelpModifier;
+        } else if (this == Blocks.TWISTING_VINES) {
+            modifier = level.spigotConfig.twistingVinesModifier;
+        } else if (this == Blocks.WEEPING_VINES) {
+            modifier = level.spigotConfig.weepingVinesModifier;
+        } else if (this == Blocks.CAVE_VINES) {
+            modifier = level.spigotConfig.caveVinesModifier;
+        }
+        if (state.getValue(AGE) < 25 && random.nextDouble() < ((modifier / 100.0) * this.growPerTickProbability)) { // Spigot - SPIGOT-7159: Better modifier resolution
+            // Spigot end
             BlockPos growthPos = pos.relative(this.growthDirection);
             if (this.canGrowInto(level.getBlockState(growthPos))) {
-                level.setBlockAndUpdate(growthPos, this.getGrowIntoState(state, level.getRandom()));
+                org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockSpreadEvent(level, pos, growthPos, this.getGrowIntoState(state, level.getRandom(), level), Block.UPDATE_ALL); // CraftBukkit // Paper - Fix Spigot growth modifiers
             }
         }
     }
 
+    // Paper start - Fix Spigot growth modifiers
+    protected BlockState getGrowIntoState(BlockState state, RandomSource random, @org.jspecify.annotations.Nullable Level level) {
+        return this.getGrowIntoState(state, random);
+    }
+    // Paper end - Fix Spigot growth modifiers
+
     protected BlockState getGrowIntoState(final BlockState growFromState, final RandomSource random) {
         return growFromState.cycle(AGE);
     }
