From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/LeverBlock.java b/net/minecraft/world/level/block/LeverBlock.java
index e851ad854b37999784e0c89a67d77610107b927c..7b9989c23f248dcde9a0730b5d898fd674c53edf 100644
--- a/net/minecraft/world/level/block/LeverBlock.java
+++ b/net/minecraft/world/level/block/LeverBlock.java
@@ -63,6 +63,12 @@ public class LeverBlock extends FaceAttachedHorizontalDirectionalBlock {
                 makeParticle(stateAfter, level, pos, 1.0F);
             }
         } else {
+            // Paper start - Call BlockRedstoneEvent
+            boolean wasPowered = stateBefore.getValue(POWERED);
+            if (!org.bukkit.craftbukkit.event.CraftEventFactory.callBinaryRedstoneChange(level, pos, !wasPowered)) {
+                return InteractionResult.SUCCESS;
+            }
+            // Paper end - Call BlockRedstoneEvent
             this.pull(stateBefore, level, pos, null);
         }
 
@@ -74,7 +80,12 @@ public class LeverBlock extends FaceAttachedHorizontalDirectionalBlock {
         final BlockState state, final ServerLevel level, final BlockPos pos, final Explosion explosion, final BiConsumer<ItemStack, BlockPos> onHit
     ) {
         if (explosion.canTriggerBlocks()) {
+            // Paper start - Call BlockRedstoneEvent
+            boolean wasPowered = state.getValue(POWERED);
+            if (org.bukkit.craftbukkit.event.CraftEventFactory.callBinaryRedstoneChange(level, pos, !wasPowered)) {
+            // Paper end - Call BlockRedstoneEvent
             this.pull(state, level, pos, null);
+            } // Paper - Call BlockRedstoneEvent
         }
 
         super.onExplosionHit(state, level, pos, explosion, onHit);
