From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/MultifaceSpreader.java b/net/minecraft/world/level/block/MultifaceSpreader.java
index a961fdba62446fbc35a5f7170aec21b9e02c7dc4..2cbd6be3ba3495f9040af1fc2eba6eeb232103c9 100644
--- a/net/minecraft/world/level/block/MultifaceSpreader.java
+++ b/net/minecraft/world/level/block/MultifaceSpreader.java
@@ -178,14 +178,14 @@ public class MultifaceSpreader {
                     level.getChunk(spreadPos.pos()).markPosForPostProcessing(spreadPos.pos());
                 }
 
-                return level.setBlock(spreadPos.pos(), spreadState, Block.UPDATE_CLIENTS);
+                return org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockSpreadEvent(level, spreadPos.source(), spreadPos.pos(), spreadState, Block.UPDATE_CLIENTS, true); // CraftBukkit
             } else {
                 return false;
             }
         }
     }
 
-    public record SpreadPos(BlockPos pos, Direction face) {
+    public record SpreadPos(BlockPos pos, Direction face, BlockPos source) { // CraftBukkit
     }
 
     @FunctionalInterface
@@ -197,19 +197,19 @@ public class MultifaceSpreader {
         SAME_POSITION {
             @Override
             public MultifaceSpreader.SpreadPos getSpreadPos(final BlockPos pos, final Direction spreadDirection, final Direction fromFace) {
-                return new MultifaceSpreader.SpreadPos(pos, spreadDirection);
+                return new MultifaceSpreader.SpreadPos(pos, spreadDirection, pos); // CraftBukkit
             }
         },
         SAME_PLANE {
             @Override
             public MultifaceSpreader.SpreadPos getSpreadPos(final BlockPos pos, final Direction spreadDirection, final Direction fromFace) {
-                return new MultifaceSpreader.SpreadPos(pos.relative(spreadDirection), fromFace);
+                return new MultifaceSpreader.SpreadPos(pos.relative(spreadDirection), fromFace, pos); // CraftBukkit
             }
         },
         WRAP_AROUND {
             @Override
             public MultifaceSpreader.SpreadPos getSpreadPos(final BlockPos pos, final Direction spreadDirection, final Direction fromFace) {
-                return new MultifaceSpreader.SpreadPos(pos.relative(spreadDirection).relative(fromFace), spreadDirection.getOpposite());
+                return new MultifaceSpreader.SpreadPos(pos.relative(spreadDirection).relative(fromFace), spreadDirection.getOpposite(), pos); // CraftBukkit
             }
         };
 
