From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/PowderSnowBlock.java b/net/minecraft/world/level/block/PowderSnowBlock.java
index c7c5d3a6a60c9b00c35df6da7de7975c605737bd..a9313f7bda622e02024b45c1551c4fa27c96b829 100644
--- a/net/minecraft/world/level/block/PowderSnowBlock.java
+++ b/net/minecraft/world/level/block/PowderSnowBlock.java
@@ -60,6 +60,14 @@ public class PowderSnowBlock extends Block implements BucketPickup {
         final InsideBlockEffectApplier effectApplier,
         final boolean isPrecise
     ) {
+        // Paper start - Add EntityInsideBlockEvent
+        if (!new io.papermc.paper.event.entity.EntityInsideBlockEvent(entity.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.at(level, pos)).callEvent()) {
+            if (entity instanceof net.minecraft.server.level.ServerPlayer player) {
+                player.resendPossiblyDesyncedDataValues(java.util.List.of(net.minecraft.world.entity.Entity.DATA_TICKS_FROZEN), player); // prevent desync on cancelled event
+            }
+            return;
+        }
+        // Paper end - Add EntityInsideBlockEvent
         if (!(entity instanceof LivingEntity) || entity.getInBlockState().is(this)) {
             entity.makeStuckInBlock(state, new Vec3(0.9F, 1.5, 0.9F));
             if (level.isClientSide()) {
@@ -85,8 +93,13 @@ public class PowderSnowBlock extends Block implements BucketPickup {
             e -> {
                 if (level instanceof ServerLevel serverLevel
                     && e.isOnFire()
-                    && (serverLevel.getGameRules().get(GameRules.MOB_GRIEFING) || e instanceof Player)
+                        // CraftBukkit - move down
                     && e.mayInteract(serverLevel, position)) {
+                    // CraftBukkit start
+                    if (!org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(e, pos, Blocks.AIR.defaultBlockState(), !(serverLevel.getGameRules().get(GameRules.MOB_GRIEFING) || e instanceof Player))) {
+                        return;
+                    }
+                    // CraftBukkit end
                     level.destroyBlock(position, false);
                 }
             }
