From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/ScaffoldingBlock.java b/net/minecraft/world/level/block/ScaffoldingBlock.java
index d8898aa528470b6ff5610674f35ee6084e9e169a..827cdf0d1649e0a9c841838ef13fd961945cdca3 100644
--- a/net/minecraft/world/level/block/ScaffoldingBlock.java
+++ b/net/minecraft/world/level/block/ScaffoldingBlock.java
@@ -117,7 +117,7 @@ public class ScaffoldingBlock extends Block implements SimpleWaterloggedBlock {
     protected void tick(final BlockState state, final ServerLevel level, final BlockPos pos, final RandomSource random) {
         int distance = getDistance(level, pos);
         BlockState newState = state.setValue(DISTANCE, distance).setValue(BOTTOM, this.isBottom(level, pos, distance));
-        if (newState.getValue(DISTANCE) == 7) {
+        if (newState.getValue(DISTANCE) == 7 && !org.bukkit.craftbukkit.event.CraftEventFactory.callBlockFadeEvent(level, pos, newState.getFluidState().createLegacyBlock()).isCancelled()) { // CraftBukkit - BlockFadeEvent // Paper - fix wrong block state
             if (state.getValue(DISTANCE) == 7) {
                 FallingBlockEntity.fall(level, pos, newState);
             } else {
