From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/ShelfBlock.java b/net/minecraft/world/level/block/ShelfBlock.java
index 59c2ffe912131d07d366e789abc54560ed437744..8a1313a10b6afeee86ea2245da4bd9beb28d1d84 100644
--- a/net/minecraft/world/level/block/ShelfBlock.java
+++ b/net/minecraft/world/level/block/ShelfBlock.java
@@ -111,6 +111,11 @@ public class ShelfBlock extends BaseEntityBlock implements SelectableSlotContain
         if (!level.isClientSide()) {
             boolean signal = level.hasNeighborSignal(pos);
             if (state.getValue(POWERED) != signal) {
+                // Paper start - Call BlockRedstoneEvent
+                if (!org.bukkit.craftbukkit.event.CraftEventFactory.callBinaryRedstoneChange(level, pos, signal)) {
+                    return;
+                }
+                // Paper end - Call BlockRedstoneEvent
                 BlockState newState = state.setValue(POWERED, signal);
                 if (!signal) {
                     newState = newState.setValue(SIDE_CHAIN_PART, SideChainPart.UNCONNECTED);
