From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/SnifferEggBlock.java b/net/minecraft/world/level/block/SnifferEggBlock.java
index fdc14e699f7bf8a4709c7ff2bfaca94fb9800c07..6ad394a0cfb9cec11845fcc17dc1d7fab34cca2d 100644
--- a/net/minecraft/world/level/block/SnifferEggBlock.java
+++ b/net/minecraft/world/level/block/SnifferEggBlock.java
@@ -61,12 +61,31 @@ public class SnifferEggBlock extends Block {
         return this.getHatchLevel(state) == 2;
     }
 
+    // Paper start - Call BlockFadeEvent
+    private void rescheduleTick(ServerLevel level, BlockPos pos) {
+        int baseDelay = hatchBoost(level, pos) ? level.paperConfig().entities.sniffer.boostedHatchTime.or(BOOSTED_HATCH_TIME_TICKS) : level.paperConfig().entities.sniffer.hatchTime.or(REGULAR_HATCH_TIME_TICKS); // Paper - Configure sniffer egg hatch time
+        level.scheduleTick(pos, this, (baseDelay / 3) + level.getRandom().nextInt(RANDOM_HATCH_OFFSET_TICKS));
+        // reschedule to avoid being stuck here and behave like the other calls (see #onPlace)
+    }
+    // Paper end - Call BlockFadeEvent
+
     @Override
     public void tick(final BlockState state, final ServerLevel level, final BlockPos position, final RandomSource random) {
         if (!this.isReadyToHatch(state)) {
+            // Paper start
+            if (!org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(level, position, state.setValue(HATCH, this.getHatchLevel(state) + 1), Block.UPDATE_CLIENTS)) {
+                this.rescheduleTick(level, position);
+                return;
+            }
+            // Paper end
             level.playSound(null, position, SoundEvents.SNIFFER_EGG_CRACK, SoundSource.BLOCKS, 0.7F, 0.9F + random.nextFloat() * 0.2F);
-            level.setBlock(position, state.setValue(HATCH, this.getHatchLevel(state) + 1), Block.UPDATE_CLIENTS);
         } else {
+            // Paper start - Call BlockFadeEvent
+            if (org.bukkit.craftbukkit.event.CraftEventFactory.callBlockFadeEvent(level, position, state.getFluidState().createLegacyBlock()).isCancelled()) {
+                this.rescheduleTick(level, position);
+                return;
+            }
+            // Paper end - Call BlockFadeEvent
             level.playSound(null, position, SoundEvents.SNIFFER_EGG_HATCH, SoundSource.BLOCKS, 0.7F, 0.9F + random.nextFloat() * 0.2F);
             level.destroyBlock(position, false);
             Sniffer sniffer = EntityTypes.SNIFFER.create(level, EntitySpawnReason.BREEDING);
@@ -74,7 +93,7 @@ public class SnifferEggBlock extends Block {
                 Vec3 spawnAt = Vec3.atCenterOf(position);
                 sniffer.setBaby(true);
                 sniffer.snapTo(spawnAt.x(), spawnAt.y(), spawnAt.z(), Mth.wrapDegrees(level.getRandom().nextFloat() * 360.0F), 0.0F);
-                level.addFreshEntity(sniffer);
+                level.addFreshEntity(sniffer, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.EGG); // Paper - use correct spawn reason
             }
         }
     }
@@ -86,7 +105,7 @@ public class SnifferEggBlock extends Block {
             level.levelEvent(LevelEvent.PARTICLES_EGG_CRACK, pos, 0);
         }
 
-        int hatchTime = boosted ? 12000 : 24000;
+        int hatchTime = boosted ? level.paperConfig().entities.sniffer.boostedHatchTime.or(BOOSTED_HATCH_TIME_TICKS) : level.paperConfig().entities.sniffer.hatchTime.or(REGULAR_HATCH_TIME_TICKS); // Paper
         int progressionTickDelay = hatchTime / 3;
         level.gameEvent(GameEvent.BLOCK_PLACE, pos, GameEvent.Context.of(state));
         level.scheduleTick(pos, this, progressionTickDelay + level.getRandom().nextInt(300));
