From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/SpawnerBlock.java b/net/minecraft/world/level/block/SpawnerBlock.java
index c92db59453105b02b2e0550646563479f820ac40..07a3376b2a146ab8241c93b74c205f76e9cc0bd0 100644
--- a/net/minecraft/world/level/block/SpawnerBlock.java
+++ b/net/minecraft/world/level/block/SpawnerBlock.java
@@ -40,10 +40,18 @@ public class SpawnerBlock extends BaseEntityBlock {
     @Override
     protected void spawnAfterBreak(final BlockState state, final ServerLevel level, final BlockPos pos, final ItemStack tool, final boolean dropExperience) {
         super.spawnAfterBreak(state, level, pos, tool, dropExperience);
+        // CraftBukkit start - Delegate to getExpDrop
+    }
+
+    @Override
+    public int getExpDrop(final BlockState state, final ServerLevel level, final BlockPos pos, final ItemStack tool, final boolean dropExperience) {
         if (dropExperience) {
             RandomSource random = level.getRandom();
             int magicCount = 15 + random.nextInt(15) + random.nextInt(15);
-            this.popExperience(level, pos, magicCount);
+            // this.popExperience(level, pos, magicCount);
+            return magicCount;
         }
+        return 0;
+        // CraftBukkit end
     }
 }
