From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/StemBlock.java b/net/minecraft/world/level/block/StemBlock.java
index 255b5dea019b743ed572a389eb0b582fabea6408..b772d913732f9215652eafe156ca3578cbcd9932 100644
--- a/net/minecraft/world/level/block/StemBlock.java
+++ b/net/minecraft/world/level/block/StemBlock.java
@@ -83,11 +83,11 @@ public class StemBlock extends VegetationBlock implements BonemealableBlock {
     protected void randomTick(BlockState state, final ServerLevel level, final BlockPos pos, final RandomSource random) {
         if (level.getRawBrightness(pos, 0) >= 9) {
             float growthSpeed = CropBlock.getGrowthSpeed(this, level, pos);
-            if (random.nextInt((int)(25.0F / growthSpeed) + 1) == 0) {
+            if (random.nextFloat() < ((this == Blocks.PUMPKIN_STEM ? level.spigotConfig.pumpkinModifier : level.spigotConfig.melonModifier) / (100.0F * (Math.floor((25.0F / growthSpeed) + 1))))) { // Spigot - SPIGOT-7159: Better modifier resolution
                 int age = state.getValue(AGE);
                 if (age < 7) {
                     state = state.setValue(AGE, age + 1);
-                    level.setBlock(pos, state, Block.UPDATE_CLIENTS);
+                    org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(level, pos, state, Block.UPDATE_CLIENTS); // CraftBukkit
                 } else {
                     Direction direction = Direction.Plane.HORIZONTAL.getRandomDirection(random);
                     BlockPos relative = pos.relative(direction);
@@ -97,7 +97,11 @@ public class StemBlock extends VegetationBlock implements BonemealableBlock {
                         Optional<Block> fruit = blocks.getOptional(this.fruit);
                         Optional<Block> stem = blocks.getOptional(this.attachedStem);
                         if (fruit.isPresent() && stem.isPresent()) {
-                            level.setBlockAndUpdate(relative, fruit.get().defaultBlockState());
+                            // CraftBukkit start
+                            if (!org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(level, relative, fruit.get().defaultBlockState(), Block.UPDATE_ALL)) {
+                                return;
+                            }
+                            // CraftBukkit end
                             level.setBlockAndUpdate(pos, stem.get().defaultBlockState().setValue(HorizontalDirectionalBlock.FACING, direction));
                         }
                     }
@@ -125,7 +129,7 @@ public class StemBlock extends VegetationBlock implements BonemealableBlock {
     public void performBonemeal(final ServerLevel level, final RandomSource random, final BlockPos pos, final BlockState state) {
         int age = Math.min(7, state.getValue(AGE) + Mth.nextInt(random, 2, 5));
         BlockState newState = state.setValue(AGE, age);
-        level.setBlock(pos, newState, Block.UPDATE_CLIENTS);
+        org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(level, pos, newState, Block.UPDATE_CLIENTS); // CraftBukkit
         if (age == 7) {
             newState.randomTick(level, pos, random);
         }
