From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/WallHangingSignBlock.java b/net/minecraft/world/level/block/WallHangingSignBlock.java
index e204b9552be57a9dd00fd5fbc5b7e67fe80d1ced..6972bcca504ea5d617ea1fe92efd87ba37e6923c 100644
--- a/net/minecraft/world/level/block/WallHangingSignBlock.java
+++ b/net/minecraft/world/level/block/WallHangingSignBlock.java
@@ -172,7 +172,7 @@ public class WallHangingSignBlock extends SignBlock implements HangingSignBlock
 
     @Override
     public <T extends BlockEntity> @Nullable BlockEntityTicker<T> getTicker(final Level level, final BlockState blockState, final BlockEntityType<T> type) {
-        return createTickerHelper(type, BlockEntityTypes.HANGING_SIGN, SignBlockEntity::tick);
+        return null; // CraftBukkit - remove unnecessary sign ticking
     }
 
     @Override
