From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/WeatheringCopperGolemStatueBlock.java b/net/minecraft/world/level/block/WeatheringCopperGolemStatueBlock.java
index e32c0c7e3b27c95459ee9ecb977b7b6dc41a03e7..899a5935c46a23e33f5507ee0ff793741927f016 100644
--- a/net/minecraft/world/level/block/WeatheringCopperGolemStatueBlock.java
+++ b/net/minecraft/world/level/block/WeatheringCopperGolemStatueBlock.java
@@ -64,16 +64,21 @@ public class WeatheringCopperGolemStatueBlock extends CopperGolemStatueBlock imp
                     return InteractionResult.PASS;
                 }
 
-                this.updatePose(level, state, pos, player);
-                return InteractionResult.SUCCESS;
+                return this.updatePose(level, state, pos, player); // Paper - call EntityChangeBlockEvent
             }
 
             if (this.getAge().equals(WeatheringCopper.WeatherState.UNAFFECTED)) {
                 CopperGolem copperGolem = copperGolemStatueBlockEntity.removeStatue(state);
-                itemStack.hurtAndBreak(1, player, hand.asEquipmentSlot());
                 if (copperGolem != null) {
-                    level.addFreshEntity(copperGolem);
-                    level.removeBlock(pos, false);
+                    // Paper start - call EntityChangeBlockEvent and spawnReason
+                    BlockState newState = level.getFluidState(pos).createLegacyBlock();
+                    if (!org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(player, pos, newState)) {
+                        return InteractionResult.PASS;
+                    }
+                    itemStack.hurtAndBreak(1, player, hand.asEquipmentSlot()); // Paper - moved after event
+                    level.addFreshEntity(copperGolem, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.REANIMATE); // Paper - add SpawnReason
+                    level.setBlock(pos, newState, Block.UPDATE_ALL);
+                    // Paper end - call EntityChangeBlockEvent and spawnReason
                     return InteractionResult.SUCCESS;
                 }
             }
