From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/WeatheringCopperGolemStatueBlock.java b/net/minecraft/world/level/block/WeatheringCopperGolemStatueBlock.java
index 9551d2c509b12c06a53fd6071c99fb9b5d5edf34..c154ec71c2f5af67cfe0f0612203fab32cd9df92 100644
--- a/net/minecraft/world/level/block/WeatheringCopperGolemStatueBlock.java
+++ b/net/minecraft/world/level/block/WeatheringCopperGolemStatueBlock.java
@@ -52,16 +52,21 @@ public class WeatheringCopperGolemStatueBlock extends CopperGolemStatueBlock imp
                     return InteractionResult.PASS;
                 }
 
-                this.updatePose(level, state, pos, player);
-                return InteractionResult.SUCCESS;
+                return this.updatePose(level, state, pos, player); // Paper - call EntityChangeBlockEvent
             }
 
             if (this.getAge().equals(WeatheringCopper.WeatherState.UNAFFECTED)) {
                 CopperGolem copperGolem = copperGolemStatueBlockEntity.removeStatue(state);
-                itemStack.hurtAndBreak(1, player, hand.asEquipmentSlot());
                 if (copperGolem != null) {
-                    level.addFreshEntity(copperGolem);
-                    level.removeBlock(pos, false);
+                    // Paper start - call EntityChangeBlockEvent and spawnReason
+                    BlockState newState = level.getFluidState(pos).createLegacyBlock();
+                    if (!org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(player, pos, newState)) {
+                        return InteractionResult.PASS;
+                    }
+                    itemStack.hurtAndBreak(1, player, hand.asEquipmentSlot()); // Paper - moved after event
+                    level.addFreshEntity(copperGolem, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.REANIMATE); // Paper - add SpawnReason
+                    level.setBlock(pos, newState, Block.UPDATE_ALL);
+                    // Paper end - call EntityChangeBlockEvent and spawnReason
                     return InteractionResult.SUCCESS;
                 }
             }
