From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/WebBlock.java b/net/minecraft/world/level/block/WebBlock.java
index ff866c3569b02ab8a5e36a428031573b2417140b..cae80183c36b00c54b2f85c21fc7b5c02021a584 100644
--- a/net/minecraft/world/level/block/WebBlock.java
+++ b/net/minecraft/world/level/block/WebBlock.java
@@ -32,6 +32,7 @@ public class WebBlock extends Block {
         final InsideBlockEffectApplier effectApplier,
         final boolean isPrecise
     ) {
+        if (!new io.papermc.paper.event.entity.EntityInsideBlockEvent(entity.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.at(level, pos)).callEvent()) { return; } // Paper - Add EntityInsideBlockEvent
         Vec3 speedMultiplier = new Vec3(0.25, 0.05F, 0.25);
         if (entity instanceof LivingEntity livingEntity && livingEntity.hasEffect(MobEffects.WEAVING)) {
             speedMultiplier = new Vec3(0.5, 0.25, 0.5);
