From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/block/WitherSkullBlock.java b/net/minecraft/world/level/block/WitherSkullBlock.java
index 6b525573397b737681ea867bedbc63c0a5ab21d0..2e65881cd2db4dab9967a4d2c597365c31939c42 100644
--- a/net/minecraft/world/level/block/WitherSkullBlock.java
+++ b/net/minecraft/world/level/block/WitherSkullBlock.java
@@ -43,7 +43,7 @@ public class WitherSkullBlock extends SkullBlock {
     }
 
     public static void checkSpawn(final Level level, final BlockPos pos) {
-        if (level.getBlockEntity(pos) instanceof SkullBlockEntity placedSkull) {
+        if (!level.captureBlockStates && level.getBlockEntity(pos) instanceof SkullBlockEntity placedSkull) { // CraftBukkit
             checkSpawn(level, pos, placedSkull);
         }
     }
@@ -57,7 +57,7 @@ public class WitherSkullBlock extends SkullBlock {
                 if (match != null) {
                     WitherBoss witherBoss = EntityTypes.WITHER.create(level, EntitySpawnReason.TRIGGERED);
                     if (witherBoss != null) {
-                        CarvedPumpkinBlock.clearPatternBlocks(level, match);
+                        // CarvedPumpkinBlock.clearPatternBlocks(level, match); // CraftBukkit - move down
                         BlockPos spawnPos = match.getBlock(1, 2, 0).getPos();
                         witherBoss.snapTo(
                             spawnPos.getX() + 0.5,
@@ -68,12 +68,23 @@ public class WitherSkullBlock extends SkullBlock {
                         );
                         witherBoss.yBodyRot = match.getForwards().getAxis() == Direction.Axis.X ? 0.0F : 90.0F;
                         witherBoss.makeInvulnerable();
+                        // Paper start
+                        if (!new io.papermc.paper.event.entity.EntityConstructEvent(witherBoss.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.getMatchingBlocks(level, match)).callEvent()) {
+                            return;
+                        }
+                        // Paper end
+                        // CraftBukkit start
+                        if (!level.addFreshEntity(witherBoss, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.BUILD_WITHER)) {
+                            return;
+                        }
+                        CarvedPumpkinBlock.clearPatternBlocks(level, match); // CraftBukkit - from above
+                        // CraftBukkit end
 
                         for (ServerPlayer player : level.getEntitiesOfClass(ServerPlayer.class, witherBoss.getBoundingBox().inflate(50.0))) {
                             CriteriaTriggers.SUMMONED_ENTITY.trigger(player, witherBoss);
                         }
 
-                        level.addFreshEntity(witherBoss);
+                        // level.addFreshEntity(witherBoss); // CraftBukkit - moved up
                         CarvedPumpkinBlock.updatePatternBlocks(level, match);
                     }
                 }
