From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/syncher/EntityDataSerializers.java b/net/minecraft/network/syncher/EntityDataSerializers.java
index 81462fe7122d2ae5f6a39e1ddc8b5dd9c5f0105e..66571c25f5fa35de77740ffd743b4e70b4114385 100644
--- a/net/minecraft/network/syncher/EntityDataSerializers.java
+++ b/net/minecraft/network/syncher/EntityDataSerializers.java
@@ -59,10 +59,27 @@ public class EntityDataSerializers {
     public static final EntityDataSerializer<Optional<Component>> OPTIONAL_COMPONENT = EntityDataSerializer.forValueType(
         ComponentSerialization.TRUSTED_OPTIONAL_STREAM_CODEC
     );
+    // Paper start - do not obfuscate items sent as entity data
+    public static final StreamCodec<RegistryFriendlyByteBuf, ItemStack> OVERSIZED_ITEM_CODEC = new net.minecraft.network.codec.StreamCodec<>() {
+        @Override
+        public net.minecraft.world.item.ItemStack decode(final net.minecraft.network.RegistryFriendlyByteBuf buffer) {
+            return ItemStack.OPTIONAL_STREAM_CODEC.decode(buffer);
+        }
+
+        @Override
+        public void encode(final net.minecraft.network.RegistryFriendlyByteBuf buffer, final net.minecraft.world.item.ItemStack value) {
+            // If the codec is called during an obfuscation session, downgrade the context's obf level to OVERSIZED if it isn't already.
+            // Entity data cannot be fully obfuscated as entities might render out specific values (e.g. count or custom name).
+            try (final io.papermc.paper.util.SafeAutoClosable ignored = io.papermc.paper.util.sanitizer.ItemObfuscationSession.withContext(c -> c.level(io.papermc.paper.util.sanitizer.ItemObfuscationSession.ObfuscationLevel.OVERSIZED))) {
+                ItemStack.OPTIONAL_STREAM_CODEC.encode(buffer, value);
+            }
+        }
+    };
+    // Paper end - do not obfuscate items sent as entity data
     public static final EntityDataSerializer<ItemStack> ITEM_STACK = new EntityDataSerializer<ItemStack>() {
         @Override
         public StreamCodec<? super RegistryFriendlyByteBuf, ItemStack> codec() {
-            return ItemStack.OPTIONAL_STREAM_CODEC;
+            return OVERSIZED_ITEM_CODEC; // Paper - do not obfuscate items sent as entity data
         }
 
         @Override
