From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/syncher/SynchedEntityData.java b/net/minecraft/network/syncher/SynchedEntityData.java
index 4a4524873797ab75c2a3d1f39f1b2576575f0ab4..bdb4f9f5052fa5c919783b9cbe744c4c55cf9cae 100644
--- a/net/minecraft/network/syncher/SynchedEntityData.java
+++ b/net/minecraft/network/syncher/SynchedEntityData.java
@@ -67,6 +67,13 @@ public class SynchedEntityData {
         }
     }
 
+    // CraftBukkit start - add method from above
+    public <T> void markDirty(final EntityDataAccessor<T> accessor) {
+        this.getItem(accessor).setDirty(true);
+        this.isDirty = true;
+    }
+    // CraftBukkit end
+
     public boolean isDirty() {
         return this.isDirty;
     }
@@ -172,6 +179,19 @@ public class SynchedEntityData {
         }
     }
 
+    // Paper start
+    // We need to pack all as we cannot rely on "non default values" or "dirty" ones.
+    // Because these values can possibly be desynced on the client.
+    public List<SynchedEntityData.DataValue<?>> packAll() {
+        final List<SynchedEntityData.DataValue<?>> values = new ArrayList<>(this.itemsById.length);
+        for (final DataItem<?> item : this.itemsById) {
+            values.add(item.value());
+        }
+
+        return values;
+    }
+    // Paper end
+
     public static class DataItem<T> {
         private final EntityDataAccessor<T> accessor;
         private T value;
