From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/decoration/LeashFenceKnotEntity.java b/net/minecraft/world/entity/decoration/LeashFenceKnotEntity.java
index 5b1c76d6e1c8e6bd870f9a5ed70f2857b8629151..ab44beed7767ebe73053df2b6ddd3b2f2fb391b6 100644
--- a/net/minecraft/world/entity/decoration/LeashFenceKnotEntity.java
+++ b/net/minecraft/world/entity/decoration/LeashFenceKnotEntity.java
@@ -87,7 +87,7 @@ public class LeashFenceKnotEntity extends BlockAttachedEntity {
         boolean attachedMob = false;
 
         for (Leashable leashable : Leashable.leashableLeashedTo(player)) {
-            if (leashable.canHaveALeashAttachedTo(this)) {
+            if (leashable.canHaveALeashAttachedTo(this) && org.bukkit.craftbukkit.event.CraftEventFactory.handlePlayerLeashEntityEvent(leashable, this, player, hand)) { // Paper - leash event
                 leashable.setLeashedTo(this, true);
                 attachedMob = true;
             }
@@ -96,7 +96,7 @@ public class LeashFenceKnotEntity extends BlockAttachedEntity {
         boolean anyDropped = false;
         if (!attachedMob && !player.isSecondaryUseActive()) {
             for (Leashable mob : Leashable.leashableLeashedTo(this)) {
-                if (mob.canHaveALeashAttachedTo(player)) {
+                if (mob.canHaveALeashAttachedTo(player) && org.bukkit.craftbukkit.event.CraftEventFactory.handlePlayerLeashEntityEvent(mob, player, player, hand)) { // Paper - leash event
                     mob.setLeashedTo(player, true);
                     anyDropped = true;
                 }
@@ -115,7 +115,7 @@ public class LeashFenceKnotEntity extends BlockAttachedEntity {
     @Override
     public void notifyLeasheeRemoved(final Leashable entity) {
         if (Leashable.leashableLeashedTo(this).isEmpty()) {
-            this.discard();
+            this.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.DROP); // CraftBukkit - add Bukkit remove cause
         }
     }
 
