From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/animal/equine/AbstractChestedHorse.java b/net/minecraft/world/entity/animal/equine/AbstractChestedHorse.java
index 90d68510057773c6a1f2b764c4da26a0b1641c1a..15f95a42be96d73a1a86210553098ad5a92d8900 100644
--- a/net/minecraft/world/entity/animal/equine/AbstractChestedHorse.java
+++ b/net/minecraft/world/entity/animal/equine/AbstractChestedHorse.java
@@ -74,7 +74,7 @@ public abstract class AbstractChestedHorse extends AbstractHorse {
         super.dropEquipment(level);
         if (this.hasChest()) {
             this.spawnAtLocation(level, Blocks.CHEST);
-            this.setChest(false);
+            this.postDeathEventTasks.add(() -> this.setChest(false)); // Paper
         }
     }
 
