From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/level/DistanceManager.java b/net/minecraft/server/level/DistanceManager.java
index 47041aaca7defab030773e6c63c8711a7bc4febb..31badcd3a478f770e1a8d296dbb5d9e4b529ae7d 100644
--- a/net/minecraft/server/level/DistanceManager.java
+++ b/net/minecraft/server/level/DistanceManager.java
@@ -72,6 +72,12 @@ public abstract class DistanceManager {
         }
 
         if (!this.chunksToUpdateFutures.isEmpty()) {
+            // CraftBukkit start - SPIGOT-7780: Call chunk unload events before updateHighestAllowedStatus
+            for (final ChunkHolder chunksToUpdateFuture : this.chunksToUpdateFutures) {
+                chunksToUpdateFuture.callEventIfUnloading(scheduler);
+            }
+            // CraftBukkit end - SPIGOT-7780: Call chunk unload events before updateHighestAllowedStatus
+
             for (ChunkHolder chunksToUpdateFuture : this.chunksToUpdateFutures) {
                 chunksToUpdateFuture.updateHighestAllowedStatus(scheduler);
             }
@@ -119,8 +125,10 @@ public abstract class DistanceManager {
         ChunkPos chunk = pos.chunk();
         long chunkPos = chunk.pack();
         ObjectSet<ServerPlayer> chunkPlayers = this.playersPerChunk.get(chunkPos);
-        chunkPlayers.remove(player);
-        if (chunkPlayers.isEmpty()) {
+        // Paper start - some state corruption happens here, don't crash, clean up gracefully
+        if (chunkPlayers != null) chunkPlayers.remove(player);
+        if (chunkPlayers == null || chunkPlayers.isEmpty()) {
+        // Paper end - some state corruption happens here, don't crash, clean up gracefully
             this.playersPerChunk.remove(chunkPos);
             this.naturalSpawnChunkCounter.update(chunkPos, Integer.MAX_VALUE, false);
             this.playerTicketManager.update(chunkPos, Integer.MAX_VALUE, false);
