From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/crafting/CustomRecipe.java b/net/minecraft/world/item/crafting/CustomRecipe.java
index 02139aac87faa27974240b4d64ba47ec559e33e7..af19d0ba60f094fc2a6a23ffbef2f9125ae58bc7 100644
--- a/net/minecraft/world/item/crafting/CustomRecipe.java
+++ b/net/minecraft/world/item/crafting/CustomRecipe.java
@@ -28,4 +28,15 @@ public abstract class CustomRecipe implements CraftingRecipe {
 
     @Override
     public abstract RecipeSerializer<? extends CustomRecipe> getSerializer();
+
+    // CraftBukkit start
+    @Override
+    public org.bukkit.inventory.Recipe toBukkitRecipe(org.bukkit.NamespacedKey id) {
+        org.bukkit.craftbukkit.inventory.CraftComplexRecipe recipe = new org.bukkit.craftbukkit.inventory.CraftComplexRecipe(id, org.bukkit.inventory.ItemStack.empty(), this);
+        recipe.setGroup(this.group());
+        recipe.setCategory(org.bukkit.craftbukkit.inventory.CraftRecipe.getCategory(this.category()));
+
+        return recipe;
+    }
+    // CraftBukkit end
 }
