From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/crafting/DyeRecipe.java b/net/minecraft/world/item/crafting/DyeRecipe.java
index 457ecd6ac530bb489dac9c0e730f4ec91ef9f644..f211efc81d34ee9573ec9fa7b5881c85a6790e27 100644
--- a/net/minecraft/world/item/crafting/DyeRecipe.java
+++ b/net/minecraft/world/item/crafting/DyeRecipe.java
@@ -148,4 +148,15 @@ public class DyeRecipe extends NormalCraftingRecipe {
             )
         );
     }
+
+    // CraftBukkit start
+    @Override
+    public org.bukkit.inventory.Recipe toBukkitRecipe(org.bukkit.NamespacedKey id) {
+        org.bukkit.craftbukkit.inventory.CraftItemStack result = org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(this.result.create());
+        org.bukkit.craftbukkit.inventory.CraftComplexRecipe recipe = new org.bukkit.craftbukkit.inventory.CraftComplexRecipe(id, result, this);
+        recipe.setGroup(this.group());
+        recipe.setCategory(org.bukkit.craftbukkit.inventory.CraftRecipe.getCategory(this.category()));
+        return recipe;
+    }
+    // CraftBukkit end
 }
