From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/crafting/RecipeHolder.java b/net/minecraft/world/item/crafting/RecipeHolder.java
index 72302238b58a30c78bcaa42e9f76ceffe0172ab7..3649aa75fcd666f57e7b8e4b0bd0faf946b4fcd1 100644
--- a/net/minecraft/world/item/crafting/RecipeHolder.java
+++ b/net/minecraft/world/item/crafting/RecipeHolder.java
@@ -10,6 +10,12 @@ public record RecipeHolder<T extends Recipe<?>>(ResourceKey<Recipe<?>> id, T val
         ResourceKey.streamCodec(Registries.RECIPE), RecipeHolder::id, Recipe.STREAM_CODEC, RecipeHolder::value, RecipeHolder::new
     );
 
+    // CraftBukkit start
+    public org.bukkit.inventory.Recipe toBukkitRecipe() {
+        return this.value.toBukkitRecipe(org.bukkit.craftbukkit.util.CraftNamespacedKey.fromMinecraft(this.id.identifier()));
+    }
+    // CraftBukkit end
+
     @Override
     public boolean equals(final Object obj) {
         return this == obj || obj instanceof RecipeHolder<?> holder && this.id == holder.id;
