From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/crafting/SmithingTrimRecipe.java b/net/minecraft/world/item/crafting/SmithingTrimRecipe.java
index c5ab70d38d3fd040aef2005a726cb7f37e570377..93347bd9fbab41888c5d63eb40e7928e74a1f9af 100644
--- a/net/minecraft/world/item/crafting/SmithingTrimRecipe.java
+++ b/net/minecraft/world/item/crafting/SmithingTrimRecipe.java
@@ -47,11 +47,20 @@ public class SmithingTrimRecipe extends SimpleSmithingRecipe {
     private final Ingredient base;
     private final Ingredient addition;
     private final Holder<TrimPattern> pattern;
+    final boolean copyDataComponents; // Paper - Option to prevent data components copy
 
     public SmithingTrimRecipe(
         final Recipe.CommonInfo commonInfo, final Ingredient template, final Ingredient base, final Ingredient addition, final Holder<TrimPattern> pattern
     ) {
+        // Paper start - Option to prevent data components copy
+        this(commonInfo, template, base, addition, pattern, true);
+    }
+    public SmithingTrimRecipe(
+        final Recipe.CommonInfo commonInfo, final Ingredient template, final Ingredient base, final Ingredient addition, final Holder<TrimPattern> pattern, final boolean copyDataComponents
+    ) {
+        // Paper end - Option to prevent data components copy
         super(commonInfo);
+        this.copyDataComponents = copyDataComponents; // Paper
         this.template = template;
         this.base = base;
         this.addition = addition;
@@ -60,10 +69,15 @@ public class SmithingTrimRecipe extends SimpleSmithingRecipe {
 
     @Override
     public ItemStack assemble(final SmithingRecipeInput input) {
-        return applyTrim(input.base(), input.addition(), this.pattern);
+        return applyTrim(input.base(), input.addition(), this.pattern, this.copyDataComponents); // Paper start - Option to prevent data components copy
     }
 
     public static ItemStack applyTrim(final ItemStack baseItem, final ItemStack materialItem, final Holder<TrimPattern> pattern) {
+        // Paper start - Option to prevent data components copy
+        return applyTrim(baseItem, materialItem, pattern, true);
+    }
+    public static ItemStack applyTrim(final ItemStack baseItem, final ItemStack materialItem, final Holder<TrimPattern> pattern, final boolean copyDataComponents) {
+        // Paper end - Option to prevent data components copy
         Holder<TrimMaterial> material = materialItem.get(DataComponents.PROVIDES_TRIM_MATERIAL);
         if (material != null) {
             ArmorTrim existingTrim = baseItem.get(DataComponents.TRIM);
@@ -72,7 +86,7 @@ public class SmithingTrimRecipe extends SimpleSmithingRecipe {
                 return ItemStack.EMPTY;
             }
 
-            ItemStack trimmedItem = baseItem.copyWithCount(1);
+            ItemStack trimmedItem = copyDataComponents ? baseItem.copyWithCount(1) : new ItemStack(baseItem.getItem(), 1); // Paper - Option to prevent data components copy
             trimmedItem.set(DataComponents.TRIM, newTrim);
             return trimmedItem;
         } else {
@@ -120,4 +134,10 @@ public class SmithingTrimRecipe extends SimpleSmithingRecipe {
             )
         );
     }
+    // CraftBukkit start
+    @Override
+    public org.bukkit.inventory.Recipe toBukkitRecipe(org.bukkit.NamespacedKey id) {
+        return new org.bukkit.craftbukkit.inventory.CraftSmithingTrimRecipe(id, org.bukkit.craftbukkit.inventory.CraftRecipe.toChoice(this.template), org.bukkit.craftbukkit.inventory.CraftRecipe.toChoice(this.base), org.bukkit.craftbukkit.inventory.CraftRecipe.toChoice(this.addition), org.bukkit.craftbukkit.inventory.trim.CraftTrimPattern.minecraftHolderToBukkit(this.pattern), this.copyDataComponents); // Paper - Option to prevent data components copy
+    }
+    // CraftBukkit end
 }
