From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/crafting/SmithingTransformRecipe.java b/net/minecraft/world/item/crafting/SmithingTransformRecipe.java
index 34694705fbb74c49acfe2536ecc0809848cfa13a..33f9870a4f966a887023b1debd4ab9ef9d24378f 100644
--- a/net/minecraft/world/item/crafting/SmithingTransformRecipe.java
+++ b/net/minecraft/world/item/crafting/SmithingTransformRecipe.java
@@ -42,6 +42,7 @@ public class SmithingTransformRecipe extends SimpleSmithingRecipe {
     private final Ingredient base;
     private final Optional<Ingredient> addition;
     private final ItemStackTemplate result;
+    private final boolean copyDataComponents; // Paper - Option to prevent data components copy
 
     public SmithingTransformRecipe(
         final Recipe.CommonInfo commonInfo,
@@ -49,8 +50,22 @@ public class SmithingTransformRecipe extends SimpleSmithingRecipe {
         final Ingredient base,
         final Optional<Ingredient> addition,
         final ItemStackTemplate result
+    ) {
+    // Paper start - Option to prevent data components copy
+        this(commonInfo, template, base, addition, result, true);
+    }
+
+    public SmithingTransformRecipe(
+        final Recipe.CommonInfo commonInfo,
+        final Optional<Ingredient> template,
+        final Ingredient base,
+        final Optional<Ingredient> addition,
+        final ItemStackTemplate result,
+        final boolean copyDataComponents
     ) {
         super(commonInfo);
+        this.copyDataComponents = copyDataComponents;
+    // Paper end - Option to prevent data components copy
         this.template = template;
         this.base = base;
         this.addition = addition;
@@ -59,7 +74,7 @@ public class SmithingTransformRecipe extends SimpleSmithingRecipe {
 
     @Override
     public ItemStack assemble(final SmithingRecipeInput input) {
-        return TransmuteRecipe.createWithOriginalComponents(this.result, input.base());
+        return this.copyDataComponents ? TransmuteRecipe.createWithOriginalComponents(this.result, input.base()) : this.result.create(); // Paper - Option to prevent data components copy
     }
 
     @Override
@@ -99,4 +114,13 @@ public class SmithingTransformRecipe extends SimpleSmithingRecipe {
             )
         );
     }
+
+    // CraftBukkit start
+    @Override
+    public org.bukkit.inventory.Recipe toBukkitRecipe(org.bukkit.NamespacedKey id) {
+        org.bukkit.craftbukkit.inventory.CraftItemStack result = org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(this.result.create());
+
+        return new org.bukkit.craftbukkit.inventory.CraftSmithingTransformRecipe(id, result, org.bukkit.craftbukkit.inventory.CraftRecipe.toChoice(this.template), org.bukkit.craftbukkit.inventory.CraftRecipe.toChoice(this.base), org.bukkit.craftbukkit.inventory.CraftRecipe.toChoice(this.addition), this.copyDataComponents);
+    }
+    // CraftBukkit end
 }
