From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/ChunkPos.java b/net/minecraft/world/level/ChunkPos.java
index 914449c7de9ec05141db9114b20a0b43aa0f2ea5..3136dd4a5d71cc421e850df7d422bd2e51314d09 100644
--- a/net/minecraft/world/level/ChunkPos.java
+++ b/net/minecraft/world/level/ChunkPos.java
@@ -16,7 +16,16 @@ import net.minecraft.util.Util;
 import net.minecraft.world.level.chunk.status.ChunkPyramid;
 import org.jspecify.annotations.Nullable;
 
-public record ChunkPos(int x, int z) {
+// Paper start
+public record ChunkPos(int x, int z, long longKey) {
+    public ChunkPos(int x, int z) {
+        this(x, z, pack(x, z));
+    }
+
+    public ChunkPos {
+        longKey = pack(x, z); // override if people mess around with longKey
+    }
+    // Paper end
     public static final Codec<ChunkPos> CODEC = Codec.INT_STREAM
         .<ChunkPos>comapFlatMap(input -> Util.fixedSize(input, 2).map(ints -> new ChunkPos(ints[0], ints[1])), pos -> IntStream.of(pos.x, pos.z))
         .stable();
@@ -69,7 +78,7 @@ public record ChunkPos(int x, int z) {
     }
 
     public long pack() {
-        return pack(this.x, this.z);
+        return this.longKey; // Paper
     }
 
     public static long pack(final int x, final int z) {
