From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/StructureManager.java b/net/minecraft/world/level/StructureManager.java
index 88803a3d3a5b7489fd33ada2aa6d7dc9ae0d7289..a8f00da4d5c7cca9b6948d7c2d5114f84fdacc00 100644
--- a/net/minecraft/world/level/StructureManager.java
+++ b/net/minecraft/world/level/StructureManager.java
@@ -48,7 +48,13 @@ public class StructureManager {
     }
 
     public List<StructureStart> startsForStructure(final ChunkPos pos, final Predicate<Structure> matcher) {
-        Map<Structure, LongSet> allReferences = this.level.getChunk(pos.x(), pos.z(), ChunkStatus.STRUCTURE_REFERENCES).getAllReferences();
+        // Paper start - Fix swamp hut cat generation deadlock
+        return this.startsForStructure(pos, matcher, null);
+    }
+
+    public List<StructureStart> startsForStructure(final ChunkPos pos, final Predicate<Structure> matcher, final @Nullable ServerLevelAccessor levelAccessor) {
+        Map<Structure, LongSet> allReferences = (levelAccessor == null ? this.level : levelAccessor).getChunk(pos.x(), pos.z(), ChunkStatus.STRUCTURE_REFERENCES).getAllReferences();
+        // Paper end - Fix swamp hut cat generation deadlock
         Builder<StructureStart> result = ImmutableList.builder();
 
         for (Entry<Structure, LongSet> entry : allReferences.entrySet()) {
@@ -111,14 +117,29 @@ public class StructureManager {
     }
 
     public StructureStart getStructureWithPieceAt(final BlockPos blockPos, final HolderSet<Structure> structures) {
-        return this.getStructureWithPieceAt(blockPos, structures::contains);
+        // Paper start - Fix swamp hut cat generation deadlock
+        return this.getStructureWithPieceAt(blockPos, structures, null);
+    }
+    public StructureStart getStructureWithPieceAt(final BlockPos blockPos, final HolderSet<Structure> structures, final @Nullable ServerLevelAccessor levelAccessor) {
+        return this.getStructureWithPieceAt(blockPos, structures::contains, levelAccessor);
+        // Paper end - Fix swamp hut cat generation deadlock
     }
 
     public StructureStart getStructureWithPieceAt(final BlockPos blockPos, final Predicate<Holder<Structure>> predicate) {
+        // Paper start - Fix swamp hut cat generation deadlock
+        return this.getStructureWithPieceAt(blockPos, predicate, null);
+    }
+
+    public StructureStart getStructureWithPieceAt(final BlockPos blockPos, final TagKey<Structure> structureTag, final @Nullable ServerLevelAccessor levelAccessor) {
+        return this.getStructureWithPieceAt(blockPos, structure -> structure.is(structureTag), levelAccessor);
+    }
+
+    public StructureStart getStructureWithPieceAt(final BlockPos blockPos, final Predicate<Holder<Structure>> predicate, final @Nullable ServerLevelAccessor levelAccessor) {
+        // Paper end - Fix swamp hut cat generation deadlock
         Registry<Structure> structures = this.registryAccess().lookupOrThrow(Registries.STRUCTURE);
 
         for (StructureStart structureStart : this.startsForStructure(
-            ChunkPos.containing(blockPos), s -> structures.get(structures.getId(s)).map(predicate::test).orElse(false)
+            ChunkPos.containing(blockPos), s -> structures.get(structures.getId(s)).map(predicate::test).orElse(false), levelAccessor // Paper - Fix swamp hut cat generation deadlock
         )) {
             if (this.structureHasPieceAt(blockPos, structureStart)) {
                 return structureStart;
