From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/util/datafix/fixes/ItemStackTheFlatteningFix.java b/net/minecraft/util/datafix/fixes/ItemStackTheFlatteningFix.java
index 45c71c4a5ffaf33316ce2b971cd85e6cb5a2928a..677648be1661e1aafc9fa1c9037c6b6c2c30ade0 100644
--- a/net/minecraft/util/datafix/fixes/ItemStackTheFlatteningFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemStackTheFlatteningFix.java
@@ -424,7 +424,7 @@ public class ItemStackTheFlatteningFix extends DataFix {
             if (DAMAGE_IDS.contains(id.get().getSecond())) {
                 Typed<?> tag = input.getOrCreateTyped(tagF);
                 Dynamic<?> tagRest = tag.get(DSL.remainderFinder());
-                tagRest = tagRest.set("Damage", tagRest.createInt(data));
+                if (data != 0) tagRest = tagRest.set("Damage", tagRest.createInt(data)); // CraftBukkit
                 output = output.set(tagF, tag.set(DSL.remainderFinder(), tagRest));
             }
 
