From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/AgeableMob.java b/net/minecraft/world/entity/AgeableMob.java
index 52db37aa97a5d4257761060f01cdd5029f002647..dad587ec02f434c0ce91244d9668664a9681f354 100644
--- a/net/minecraft/world/entity/AgeableMob.java
+++ b/net/minecraft/world/entity/AgeableMob.java
@@ -68,6 +68,11 @@ public abstract class AgeableMob extends PathfinderMob {
     protected InteractionResult mobInteract(final Player player, final InteractionHand hand) {
         ItemStack itemInHand = player.getItemInHand(hand);
         if (canUseGoldenDandelion(itemInHand, this.isBaby(), this.ageLockParticleTimer, this)) {
+            // Paper start - call PlayerToggleEntityAgeLockEvent
+            if (!org.bukkit.craftbukkit.event.CraftEventFactory.callPlayerToggleEntityAgeLockEvent(player, this, itemInHand, hand, !this.isAgeLocked(), AGE_LOCKED)) {
+                return InteractionResult.SUCCESS;
+            }
+            // Paper end - call PlayerToggleEntityAgeLockEvent
             setAgeLocked(this, this::isAgeLocked, player, itemInHand, mob -> this.setAgeLockedData());
             return InteractionResult.SUCCESS;
         } else {
