From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/Bucketable.java b/net/minecraft/world/entity/Bucketable.java
index fb699dff1db30a5664e7c4d97554e0e51b301adb..f88d8791c3deb1ec0e022e4b422c7d73da8f1893 100644
--- a/net/minecraft/world/entity/Bucketable.java
+++ b/net/minecraft/world/entity/Bucketable.java
@@ -80,9 +80,25 @@ public interface Bucketable {
     ) {
         ItemStack itemStack = player.getItemInHand(hand);
         if (pickupEntity.canBePickedUpWithBucket(itemStack) && pickupEntity.isAlive()) {
-            pickupEntity.playSound(pickupEntity.getPickupSound(), 1.0F, 1.0F);
+            // CraftBukkit start
+            // pickupEntity.playSound(pickupEntity.getPickupSound(), 1.0F, 1.0F); // CraftBukkit - moved down
             ItemStack bucket = pickupEntity.getBucketItemStack();
             pickupEntity.saveToBucketTag(bucket);
+            org.bukkit.event.player.PlayerBucketEntityEvent playerBucketFishEvent = org.bukkit.craftbukkit.event.CraftEventFactory.callPlayerFishBucketEvent(pickupEntity, player, itemStack, bucket, hand);
+            bucket = org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(playerBucketFishEvent.getEntityBucket());
+            if (playerBucketFishEvent.isCancelled()) {
+                // Paper start - Fix inventory desync
+                if (player.hasInfiniteMaterials() || itemStack.getCount() > 1) {
+                    player.containerMenu.sendAllDataToRemote();
+                } else {
+                    player.containerMenu.forceHeldSlot(hand);
+                }
+                // Paper end - Fix inventory desync
+                pickupEntity.resendPossiblyDesyncedEntityData((ServerPlayer) player); // Paper
+                return Optional.of(InteractionResult.FAIL);
+            }
+            pickupEntity.playSound(pickupEntity.getPickupSound(), 1.0F, 1.0F);
+            // CraftBukkit end
             ItemStack result = ItemUtils.createFilledResult(itemStack, player, bucket, false);
             player.setItemInHand(hand, result);
             Level level = pickupEntity.level();
@@ -94,7 +110,7 @@ public interface Bucketable {
                 leashable.dropLeash();
             }
 
-            pickupEntity.discard();
+            pickupEntity.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.PICKUP); // CraftBukkit - add Bukkit remove cause
             return Optional.of(InteractionResult.SUCCESS);
         } else {
             return Optional.empty();
