From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ConversionType.java b/net/minecraft/world/entity/ConversionType.java
index 3313ad9465b596166851b0748424556a03b7bb7c..154e3e0283a58d17f6ac254efe1a41d003473f37 100644
--- a/net/minecraft/world/entity/ConversionType.java
+++ b/net/minecraft/world/entity/ConversionType.java
@@ -24,7 +24,7 @@ public enum ConversionType {
 
                 for (Entity passenger : to.getPassengers()) {
                     passenger.stopRiding();
-                    passenger.remove(Entity.RemovalReason.DISCARDED);
+                    passenger.remove(Entity.RemovalReason.DISCARDED, org.bukkit.event.entity.EntityRemoveEvent.Cause.TRANSFORMATION); // CraftBukkit - add Bukkit remove cause
                 }
 
                 rootPassenger.startRiding(to);
@@ -73,6 +73,7 @@ public enum ConversionType {
             if (leashHolder != null) {
                 from.dropLeash();
             }
+            to.aware = from.aware; // Paper - Fix nerfed slime when splitting
 
             ConversionType.convertCommon(from, to, params);
         }
