From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/EntitySelector.java b/net/minecraft/world/entity/EntitySelector.java
index 61fc625262933ada9f2d93c0b2d5c3857d0ef3ac..a3c13e421768f5e1a703b5b1167fa1c912b6f126 100644
--- a/net/minecraft/world/entity/EntitySelector.java
+++ b/net/minecraft/world/entity/EntitySelector.java
@@ -17,6 +17,22 @@ public final class EntitySelector {
     public static final Predicate<Entity> NO_SPECTATORS = entity -> !entity.isSpectator();
     public static final Predicate<Entity> CAN_BE_COLLIDED_WITH = NO_SPECTATORS.and(entity -> entity.canBeCollidedWith(null));
     public static final Predicate<Entity> CAN_BE_PICKED = Entity::isPickable;
+    // Paper start - Ability to control player's insomnia and phantoms
+    public static Predicate<Player> IS_INSOMNIAC = (player) -> {
+        int playerInsomniaTicks = player.level().paperConfig().entities.behavior.playerInsomniaStartTicks;
+        if (playerInsomniaTicks <= 0) {
+            return false;
+        }
+
+        net.minecraft.server.level.ServerPlayer serverPlayer = (net.minecraft.server.level.ServerPlayer) player;
+        return net.minecraft.util.Mth.clamp(serverPlayer.getStats().getValue(net.minecraft.stats.Stats.CUSTOM.get(net.minecraft.stats.Stats.TIME_SINCE_REST)), 1, Integer.MAX_VALUE) >= playerInsomniaTicks;
+    };
+    // Paper end - Ability to control player's insomnia and phantoms
+    // Paper start - Affects Spawning API
+    public static final Predicate<Entity> PLAYER_AFFECTS_SPAWNING = (entity) -> {
+        return !entity.isSpectator() && entity.isAlive() && entity instanceof Player player && player.affectsSpawning;
+    };
+    // Paper end - Affects Spawning API
 
     private EntitySelector() {
     }
@@ -33,14 +49,14 @@ public final class EntitySelector {
             ? Predicates.alwaysFalse()
             : NO_SPECTATORS.and(
                 input -> {
-                    if (!input.isPushable()) {
+                    if (!input.isPushable() || !input.canCollideWithBukkit(entity) || !entity.canCollideWithBukkit(input)) { // CraftBukkit - collidable API // Paper - Climbing should not bypass cramming gamerule
                         return false;
                     }
 
                     if (!entity.level().isClientSide() || input instanceof Player player && player.isLocalPlayer()) {
                         Team theirTeam = input.getTeam();
                         Team.CollisionRule theirCollisionRule = theirTeam == null ? Team.CollisionRule.ALWAYS : theirTeam.getCollisionRule();
-                        if (theirCollisionRule == Team.CollisionRule.NEVER) {
+                        if (theirCollisionRule == Team.CollisionRule.NEVER || (input instanceof Player && !io.papermc.paper.configuration.GlobalConfiguration.get().collisions.enablePlayerCollisions)) { // Paper - Configurable player collision
                             return false;
                         }
 
