From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/EntityEquipment.java b/net/minecraft/world/entity/EntityEquipment.java
index 126b127e860d3d75971cd2036ed62f469b6a9aea..20041325ae77f16a5b77f5de7255788b52ebb3a5 100644
--- a/net/minecraft/world/entity/EntityEquipment.java
+++ b/net/minecraft/world/entity/EntityEquipment.java
@@ -70,4 +70,11 @@ public class EntityEquipment {
     public void clear() {
         this.items.replaceAll((s, v) -> ItemStack.EMPTY);
     }
+
+    // Paper start - EntityDeathEvent
+    // Needed to not set ItemStack.EMPTY to not existent slot.
+    public boolean has(final EquipmentSlot slot) {
+        return this.items.containsKey(slot);
+    }
+    // Paper end - EntityDeathEvent
 }
