From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/EntityReference.java b/net/minecraft/world/entity/EntityReference.java
index 04a505270a3eff9f1ae400835ddbf75de7aac544..b7a74cfa848511b756187568fb30177a445b5026 100644
--- a/net/minecraft/world/entity/EntityReference.java
+++ b/net/minecraft/world/entity/EntityReference.java
@@ -142,4 +142,15 @@ public final class EntityReference<StoredEntityType extends UniquelyIdentifyable
     public int hashCode() {
         return this.getUUID().hashCode();
     }
+
+    // Paper start - utility for reading an entity from a nullable entity reference via a UUIDLookup
+    @Nullable
+    public static <StoredEntityType extends UniquelyIdentifyable> StoredEntityType getEntity(
+        final @Nullable EntityReference<StoredEntityType> ref,
+        final UUIDLookup<? extends UniquelyIdentifyable> uuidLookup,
+        final Class<StoredEntityType> entityClass
+    ) {
+        return ref == null ? null : ref.getEntity(uuidLookup, entityClass);
+    }
+    // Paper end - utility for reading an entity from a nullable entity reference via a UUIDLookup
 }
