From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/InsideBlockEffectType.java b/net/minecraft/world/entity/InsideBlockEffectType.java
index c68898581888a60f1fc1aceb09db2ef8ca7b5653..6414b5deab539d9e7af26e0f95dcdd7eab989c11 100644
--- a/net/minecraft/world/entity/InsideBlockEffectType.java
+++ b/net/minecraft/world/entity/InsideBlockEffectType.java
@@ -6,22 +6,35 @@ import net.minecraft.world.level.block.BaseFireBlock;
 public enum InsideBlockEffectType {
     FREEZE(entity -> {
         entity.setIsInPowderSnow(true);
-        if (entity.canFreeze()) {
+        if (entity.canFreeze() && !entity.freezeLocked) { // Paper - Freeze Tick Lock API
             entity.setTicksFrozen(Math.min(entity.getTicksRequiredToFreeze(), entity.getTicksFrozen() + 1));
         }
     }),
     CLEAR_FREEZE(Entity::clearFreeze),
     FIRE_IGNITE(BaseFireBlock::fireIgnite),
-    LAVA_IGNITE(Entity::lavaIgnite),
+    LAVA_IGNITE((entity, pos) -> entity.lavaIgnite(pos)), // Paper - track lava contact
     EXTINGUISH(Entity::clearFire);
 
-    private final Consumer<Entity> effect;
+    private final Applier effect; // Paper - track position inside effect was triggered on
 
     InsideBlockEffectType(final Consumer<Entity> effect) {
+    // Paper start - track position inside effect was triggered on
+        this((entity, block) -> effect.accept(entity));
+    }
+    InsideBlockEffectType(final Applier effect) {
+    // Paper end - track position inside effect was triggered on
         this.effect = effect;
     }
 
-    public Consumer<Entity> effect() {
+    public Applier effect() { // Paper - track position inside effect was triggered on
         return this.effect;
     }
+
+    // Paper start - track position inside effect was triggered on
+    // Use over biconsumer for less fqn spamming.
+    @FunctionalInterface
+    public interface Applier {
+        void affect(final Entity entity, final net.minecraft.core.BlockPos blockPos);
+    }
+    // Paper end - track position inside effect was triggered on
 }
