From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ItemBasedSteering.java b/net/minecraft/world/entity/ItemBasedSteering.java
index 2da6e45deaf9e8d99d54eb1f74f12450c6107106..f5171c9e33377b1e2a479ba7de9db1b87487e5ed 100644
--- a/net/minecraft/world/entity/ItemBasedSteering.java
+++ b/net/minecraft/world/entity/ItemBasedSteering.java
@@ -34,6 +34,14 @@ public class ItemBasedSteering {
         return true;
     }
 
+    // CraftBukkit start
+    public void boost(final int boostTime) {
+        this.boosting = true;
+        this.boostTime = 0;
+        this.entityData.set(this.boostTimeAccessor, boostTime);
+    }
+    // CraftBukkit end
+
     public void tickBoost() {
         if (this.boosting && this.boostTime++ > this.boostTimeTotal()) {
             this.boosting = false;
