From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/gametest/framework/StructureUtils.java b/net/minecraft/gametest/framework/StructureUtils.java
index 949f2eafa4f03710c9f3a8f4f6d4e736abe2b685..ed6d659c0925cfc6c7c1ce2aaaa48a3a06224d34 100644
--- a/net/minecraft/gametest/framework/StructureUtils.java
+++ b/net/minecraft/gametest/framework/StructureUtils.java
@@ -75,7 +75,7 @@ public class StructureUtils {
         level.clearBlockEvents(structureBoundingBox);
         AABB bounds = AABB.of(structureBoundingBox);
         List<Entity> livingEntities = level.getEntitiesOfClass(Entity.class, bounds, mob -> !(mob instanceof Player));
-        livingEntities.forEach(Entity::discard);
+        livingEntities.forEach(entity -> entity.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.DISCARD)); // Paper
     }
 
     public static BlockPos getTransformedFarCorner(final BlockPos structurePosition, final Vec3i size, final Rotation rotation) {
