From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/gametest/framework/TestEnvironmentDefinition.java b/net/minecraft/gametest/framework/TestEnvironmentDefinition.java
index 3597e72e1fcc0f83ec1d0348a05895701d8fd05d..58dd562fad900466a572e6b2fb429e338e817916 100644
--- a/net/minecraft/gametest/framework/TestEnvironmentDefinition.java
+++ b/net/minecraft/gametest/framework/TestEnvironmentDefinition.java
@@ -184,13 +184,13 @@ public interface TestEnvironmentDefinition<SavedDataType> {
         @Override
         public Difficulty setup(final ServerLevel level) {
             Difficulty oldDifficulty = level.getDifficulty();
-            level.getServer().setDifficulty(this.difficulty, true);
+            level.getServer().setDifficulty(level, this.difficulty, null, true); // Paper
             return oldDifficulty;
         }
 
         @Override
         public void teardown(final ServerLevel level, final Difficulty saveData) {
-            level.getServer().setDifficulty(saveData, true);
+            level.getServer().setDifficulty(level, saveData, null, true); // Paper
         }
 
         @Override
@@ -210,7 +210,7 @@ public interface TestEnvironmentDefinition<SavedDataType> {
             GameRuleMap originalState = GameRuleMap.of();
             GameRules gameRules = level.getGameRules();
             this.gameRulesMap.keySet().forEach(rule -> setFromActive(originalState, (GameRule<?>)rule, gameRules));
-            gameRules.setAll(this.gameRulesMap, level.getServer());
+            gameRules.setAll(this.gameRulesMap, level); // Paper - per-world game rules
             return originalState;
         }
 
@@ -220,7 +220,7 @@ public interface TestEnvironmentDefinition<SavedDataType> {
 
         @Override
         public void teardown(final ServerLevel level, final GameRuleMap saveData) {
-            level.getGameRules().setAll(saveData, level.getServer());
+            level.getGameRules().setAll(saveData, level); // Paper - per-world game rules
         }
 
         @Override
@@ -312,7 +312,7 @@ public interface TestEnvironmentDefinition<SavedDataType> {
             }
 
             public void apply(final ServerLevel level) {
-                level.getServer().setWeatherParameters(this.clearTime, this.rainTime, this.raining, this.thundering);
+                level.getServer().setWeatherParameters(level, this.clearTime, this.rainTime, this.raining, this.thundering); // Paper - per-level WeatherData
             }
 
             @Override
