From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/animal/parrot/Parrot.java b/net/minecraft/world/entity/animal/parrot/Parrot.java
index 18f12371445e13d5b3fb3a145ec8452a003819b7..58bb6b8fa561033f0083cd405cedb75826cfc5d9 100644
--- a/net/minecraft/world/entity/animal/parrot/Parrot.java
+++ b/net/minecraft/world/entity/animal/parrot/Parrot.java
@@ -270,7 +270,7 @@ public class Parrot extends ShoulderRidingEntity {
             }
 
             if (!this.level().isClientSide()) {
-                if (this.random.nextInt(10) == 0) {
+                if (this.random.nextInt(10) == 0 && !org.bukkit.craftbukkit.event.CraftEventFactory.callEntityTameEvent(this, player).isCancelled()) { // CraftBukkit
                     this.tame(player);
                     this.level().broadcastEntityEvent(this, EntityEvent.TAMING_SUCCEEDED);
                 } else {
@@ -291,7 +291,7 @@ public class Parrot extends ShoulderRidingEntity {
             }
         } else {
             this.usePlayerItem(player, hand, itemStack);
-            this.addEffect(new MobEffectInstance(MobEffects.POISON, 900));
+            this.addEffect(new MobEffectInstance(MobEffects.POISON, 900), org.bukkit.event.entity.EntityPotionEffectEvent.Cause.FOOD); // CraftBukkit
             if (player.isCreative() || !this.isInvulnerable()) {
                 this.hurt(this.damageSources().playerAttack(player), Float.MAX_VALUE);
             }
@@ -384,8 +384,8 @@ public class Parrot extends ShoulderRidingEntity {
     }
 
     @Override
-    public boolean isPushable() {
-        return true;
+    public boolean isCollidable(boolean ignoreClimbing) { // Paper - Climbing should not bypass cramming gamerule
+        return super.isCollidable(ignoreClimbing); // CraftBukkit - collidable API // Paper - Climbing should not bypass cramming gamerule
     }
 
     @Override
@@ -401,8 +401,13 @@ public class Parrot extends ShoulderRidingEntity {
             return false;
         }
 
+        // CraftBukkit start
+        if (!super.hurtServer(level, source, damage)) {
+            return false;
+        }
+        // CraftBukkit end
         this.setOrderedToSit(false);
-        return super.hurtServer(level, source, damage);
+        return true; // CraftBukkit
     }
 
     public Parrot.Variant getVariant() {
