From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/dedicated/DedicatedServerSettings.java b/net/minecraft/server/dedicated/DedicatedServerSettings.java
index d8f758c59877ec2a006a93b7eb3e34dc622b2f53..842b81a94c538b30c42d1eb101919669f17fe989 100644
--- a/net/minecraft/server/dedicated/DedicatedServerSettings.java
+++ b/net/minecraft/server/dedicated/DedicatedServerSettings.java
@@ -7,9 +7,11 @@ public class DedicatedServerSettings {
     private final Path source;
     private DedicatedServerProperties properties;
 
-    public DedicatedServerSettings(final Path source) {
-        this.source = source;
-        this.properties = DedicatedServerProperties.fromFile(source);
+    // CraftBukkit start
+    public DedicatedServerSettings(joptsimple.OptionSet options) {
+        this.source = ((java.io.File) options.valueOf("config")).toPath();
+        this.properties = DedicatedServerProperties.fromFile(this.source, options);
+        // CraftBukkit end
     }
 
     public DedicatedServerProperties getProperties() {
