From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/dedicated/ServerWatchdog.java b/net/minecraft/server/dedicated/ServerWatchdog.java
index 31705276206ff73bd887e56db3bde8d39e2d1f04..dfab2ade5c37e310cbafed83a014247ab43aee95 100644
--- a/net/minecraft/server/dedicated/ServerWatchdog.java
+++ b/net/minecraft/server/dedicated/ServerWatchdog.java
@@ -51,7 +51,7 @@ public class ServerWatchdog implements Runnable {
                 CrashReport report = createWatchdogCrashReport("Watching Server", this.server.getRunningThread().threadId());
                 this.server.fillSystemReport(report.getSystemReport());
                 CrashReportCategory serverStats = report.addCategory("Performance stats");
-                serverStats.setDetail("Random tick rate", () -> this.server.getGameRules().getAsString(GameRules.RANDOM_TICK_SPEED));
+                serverStats.setDetail("Overworld random tick rate", () -> this.server.overworld().getGameRules().getAsString(GameRules.RANDOM_TICK_SPEED)); // Paper - per-level GameRules
                 serverStats.setDetail(
                     "Level stats",
                     () -> Streams.stream(this.server.getAllLevels())
