From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/projectile/arrow/Arrow.java b/net/minecraft/world/entity/projectile/arrow/Arrow.java
index d045ae5f86a395371eef75e2d23d0fd1bc00c71f..7c62081d5af4f47d053674479188c85b04f5df9a 100644
--- a/net/minecraft/world/entity/projectile/arrow/Arrow.java
+++ b/net/minecraft/world/entity/projectile/arrow/Arrow.java
@@ -116,7 +116,7 @@ public class Arrow extends AbstractArrow {
         Entity effectSource = this.getEffectSource();
         PotionContents potionContents = this.getPotionContents();
         float durationScale = this.getPotionDurationScale();
-        potionContents.forEachEffect(effect -> mob.addEffect(effect, effectSource), durationScale);
+        potionContents.forEachEffect(effect -> mob.addEffect(effect, effectSource, org.bukkit.event.entity.EntityPotionEffectEvent.Cause.ARROW), durationScale); // CraftBukkit
     }
 
     @Override
