From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/projectile/arrow/SpectralArrow.java b/net/minecraft/world/entity/projectile/arrow/SpectralArrow.java
index 210a8be39e889ba6cd5f71aed778689d3a801c69..ba218cef512fab903aac04ce5b2e649b2ea4d33c 100644
--- a/net/minecraft/world/entity/projectile/arrow/SpectralArrow.java
+++ b/net/minecraft/world/entity/projectile/arrow/SpectralArrow.java
@@ -44,7 +44,7 @@ public class SpectralArrow extends AbstractArrow {
     protected void doPostHurtEffects(final LivingEntity mob) {
         super.doPostHurtEffects(mob);
         MobEffectInstance effect = new MobEffectInstance(MobEffects.GLOWING, this.duration, 0);
-        mob.addEffect(effect, this.getEffectSource());
+        mob.addEffect(effect, this.getEffectSource(), org.bukkit.event.entity.EntityPotionEffectEvent.Cause.ARROW); // CraftBukkit
     }
 
     @Override
