From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/jsonrpc/methods/BanlistService.java b/net/minecraft/server/jsonrpc/methods/BanlistService.java
index 7d3dcb672d3a4b575d1f0661d9e10921305f3a75..95bb5ac3bfe9e4e8d7e0ff5aa1ad1abae29bde74 100644
--- a/net/minecraft/server/jsonrpc/methods/BanlistService.java
+++ b/net/minecraft/server/jsonrpc/methods/BanlistService.java
@@ -45,7 +45,7 @@ public class BanlistService {
                 minecraftApi.banListService().addUserBan(userBan.toBanEntry(), clientInfo);
                 ServerPlayer player = minecraftApi.playerListService().getPlayer(ban.get().player().id());
                 if (player != null) {
-                    player.connection.disconnect(Component.translatable("multiplayer.disconnect.banned"));
+                    player.connection.disconnect(Component.translatable("multiplayer.disconnect.banned"), org.bukkit.event.player.PlayerKickEvent.Cause.BANNED); // Paper - kick event cause
                 }
             }
         }
@@ -90,7 +90,7 @@ public class BanlistService {
             minecraftApi.banListService().addUserBan(ban.toBanEntry(), clientInfo);
             ServerPlayer player = minecraftApi.playerListService().getPlayer(ban.player().id());
             if (player != null) {
-                player.connection.disconnect(Component.translatable("multiplayer.disconnect.banned"));
+                player.connection.disconnect(Component.translatable("multiplayer.disconnect.banned"), org.bukkit.event.player.PlayerKickEvent.Cause.BANNED); // Paper - kick event cause
             }
         });
         return get(minecraftApi);
