From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/jsonrpc/methods/IpBanlistService.java b/net/minecraft/server/jsonrpc/methods/IpBanlistService.java
index 1f41ab89eba6480ed8c9cbbfe6beae167d5b7649..94472181457c43426154c554a113009094214ce6 100644
--- a/net/minecraft/server/jsonrpc/methods/IpBanlistService.java
+++ b/net/minecraft/server/jsonrpc/methods/IpBanlistService.java
@@ -33,7 +33,7 @@ public class IpBanlistService {
         bans.stream()
             .map(ban -> banIp(minecraftApi, ban, clientInfo))
             .flatMap(Collection::stream)
-            .forEach(player -> player.connection.disconnect(Component.translatable("multiplayer.disconnect.ip_banned")));
+            .forEach(player -> player.connection.disconnect(Component.translatable("multiplayer.disconnect.ip_banned"), org.bukkit.event.player.PlayerKickEvent.Cause.IP_BANNED)); // Paper - kick event cause
         return get(minecraftApi);
     }
 
@@ -83,7 +83,7 @@ public class IpBanlistService {
         finalBanlist.stream()
             .filter(ban -> !currentBans.contains(ban))
             .flatMap(ban -> minecraftApi.playerListService().getPlayersWithAddress(ban.ip()).stream())
-            .forEach(player -> player.connection.disconnect(Component.translatable("multiplayer.disconnect.ip_banned")));
+            .forEach(player -> player.connection.disconnect(Component.translatable("multiplayer.disconnect.ip_banned"), org.bukkit.event.player.PlayerKickEvent.Cause.IP_BANNED)); // Paper - kick event cause
         return get(minecraftApi);
     }
 
