From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/redstone/DefaultRedstoneWireEvaluator.java b/net/minecraft/world/level/redstone/DefaultRedstoneWireEvaluator.java
index dd72fb620de48cf955295f0a46e011a90f456c21..e01604a7913601e9d84d39509c24210c810dcefd 100644
--- a/net/minecraft/world/level/redstone/DefaultRedstoneWireEvaluator.java
+++ b/net/minecraft/world/level/redstone/DefaultRedstoneWireEvaluator.java
@@ -20,7 +20,13 @@ public class DefaultRedstoneWireEvaluator extends RedstoneWireEvaluator {
         final Level level, final BlockPos pos, final BlockState state, final @Nullable Orientation orientation, final boolean skipShapeUpdates
     ) {
         int targetStrength = this.calculateTargetStrength(level, pos);
-        if (state.getValue(RedStoneWireBlock.POWER) != targetStrength) {
+        // Paper start - Call BlockRedstoneEvent
+        int previousStrength = state.getValue(RedStoneWireBlock.POWER);
+        if (previousStrength != targetStrength && level.getBlockState(pos) == state) {
+            targetStrength = org.bukkit.craftbukkit.event.CraftEventFactory.callRedstoneChange(level, pos, previousStrength, targetStrength).getNewCurrent();
+        }
+        if (previousStrength != targetStrength) {
+            // Paper end - Call BlockRedstoneEvent
             if (level.getBlockState(pos) == state) {
                 level.setBlock(pos, state.setValue(RedStoneWireBlock.POWER, targetStrength), Block.UPDATE_CLIENTS);
             }
