From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/goal/Goal.java b/net/minecraft/world/entity/ai/goal/Goal.java
index c50e8aea838dd27e4925445e46e16aa2d80ef363..aa5f0e00daf26650b943a24676638581e494a112 100644
--- a/net/minecraft/world/entity/ai/goal/Goal.java
+++ b/net/minecraft/world/entity/ai/goal/Goal.java
@@ -46,6 +46,17 @@ public abstract class Goal {
         return this.flags;
     }
 
+
+    // Paper start - Mob Goal API
+    public boolean hasFlag(final Goal.Flag flag) {
+        return this.flags.contains(flag);
+    }
+
+    public void addFlag(final Goal.Flag flag) {
+        this.flags.add(flag);
+    }
+    // Paper end - Mob Goal API
+
     protected int adjustedTickDelay(final int ticks) {
         return this.requiresUpdateEveryTick() ? ticks : reducedTickDelay(ticks);
     }
@@ -62,7 +73,19 @@ public abstract class Goal {
         return (ServerLevel)level;
     }
 
+    // Paper start - Mob goal api
+    private com.destroystokyo.paper.entity.ai.PaperGoal<?> paperGoal;
+    public <T extends org.bukkit.entity.Mob> com.destroystokyo.paper.entity.ai.Goal<T> asPaperGoal() {
+        if (this.paperGoal == null) {
+            this.paperGoal = new com.destroystokyo.paper.entity.ai.PaperGoal<>(this);
+        }
+        //noinspection unchecked
+        return (com.destroystokyo.paper.entity.ai.Goal<T>) this.paperGoal;
+    }
+    // Paper end - Mob goal api
+
     public enum Flag {
+        UNKNOWN_BEHAVIOR, // Paper - add UNKNOWN_BEHAVIOR
         MOVE,
         LOOK,
         JUMP,
