From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/goal/RunAroundLikeCrazyGoal.java b/net/minecraft/world/entity/ai/goal/RunAroundLikeCrazyGoal.java
index 639d4c47c1ff9b819707087bcc0b17d07470bb90..3645f1a669bf18cf66fbeebf437b8846753b7fd9 100644
--- a/net/minecraft/world/entity/ai/goal/RunAroundLikeCrazyGoal.java
+++ b/net/minecraft/world/entity/ai/goal/RunAroundLikeCrazyGoal.java
@@ -59,7 +59,7 @@ public class RunAroundLikeCrazyGoal extends Goal {
             if (passenger instanceof Player player) {
                 int temper = this.horse.getTemper();
                 int maxTemper = this.horse.getMaxTemper();
-                if (maxTemper > 0 && this.horse.getRandom().nextInt(maxTemper) < temper) {
+                if (maxTemper > 0 && this.horse.getRandom().nextInt(maxTemper) < temper && !org.bukkit.craftbukkit.event.CraftEventFactory.callEntityTameEvent(this.horse, player).isCancelled()) { // CraftBukkit - fire EntityTameEvent
                     this.horse.tameWithName(player);
                     return;
                 }
