From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/goal/TemptGoal.java b/net/minecraft/world/entity/ai/goal/TemptGoal.java
index 9eaa1ef31521691ed7c3b6859a502f13e73b5fc0..af2bd8a5cf760792a63dfb9c2370360f288beab6 100644
--- a/net/minecraft/world/entity/ai/goal/TemptGoal.java
+++ b/net/minecraft/world/entity/ai/goal/TemptGoal.java
@@ -23,7 +23,7 @@ public class TemptGoal extends Goal {
     private double pz;
     private double pRotX;
     private double pRotY;
-    protected @Nullable Player player;
+    protected @Nullable LivingEntity player; // CraftBukkit
     private int calmDown;
     private boolean isRunning;
     private final Predicate<ItemStack> items;
@@ -56,6 +56,15 @@ public class TemptGoal extends Goal {
         } else {
             this.player = getServerLevel(this.mob)
                 .getNearestPlayer(this.targetingConditions.range(this.mob.getAttributeValue(Attributes.TEMPT_RANGE)), this.mob);
+            // CraftBukkit start
+            if (this.player != null) {
+                org.bukkit.event.entity.EntityTargetLivingEntityEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callEntityTargetLivingEvent(this.mob, this.player, org.bukkit.event.entity.EntityTargetEvent.TargetReason.TEMPT);
+                if (event.isCancelled()) {
+                    return false;
+                }
+                this.player = (event.getTarget() == null) ? null : ((org.bukkit.craftbukkit.entity.CraftLivingEntity) event.getTarget()).getHandle();
+            }
+            // CraftBukkit end
             return this.player != null;
         }
     }
@@ -122,7 +131,7 @@ public class TemptGoal extends Goal {
         this.mob.getNavigation().stop();
     }
 
-    protected void navigateTowards(final Player player) {
+    protected void navigateTowards(final LivingEntity player) { // Paper
         this.mob.getNavigation().moveTo(player, this.speedModifier);
     }
 
@@ -141,7 +150,7 @@ public class TemptGoal extends Goal {
         }
 
         @Override
-        protected void navigateTowards(final Player player) {
+        protected void navigateTowards(final LivingEntity player) { // Paper
             Vec3 target = player.getEyePosition().subtract(this.mob.position()).scale(this.mob.getRandom().nextDouble()).add(this.mob.position());
             this.mob.getMoveControl().setWantedPosition(target.x, target.y, target.z, this.speedModifier);
         }
