From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/chat/contents/ObjectContents.java b/net/minecraft/network/chat/contents/ObjectContents.java
index d21a2359215f496b957a2c00168a80d9a7ef28fc..4f04b0dff91a0c39836bf5d306e74dd9fa4b3690 100644
--- a/net/minecraft/network/chat/contents/ObjectContents.java
+++ b/net/minecraft/network/chat/contents/ObjectContents.java
@@ -32,7 +32,7 @@ public record ObjectContents(ObjectInfo contents, Optional<Component> fallback)
 
     @Override
     public MutableComponent resolve(final ResolutionContext context, final int recursionDepth) throws CommandSyntaxException {
-        Optional<MutableComponent> fallback = ComponentUtils.resolve(context, this.fallback, recursionDepth);
+        Optional<MutableComponent> fallback = ComponentUtils.resolveSeparator(context, this.fallback, recursionDepth); // Paper - validate separator
         ObjectInfo validatedContents = context.validate(this.contents);
         return validatedContents == null
             ? fallback.orElseGet(() -> Component.literal(this.contents.defaultFallback()))
