From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/chat/contents/TranslatableContents.java b/net/minecraft/network/chat/contents/TranslatableContents.java
index 04bd33681ad4f6491e89f95a88c597538da33824..b6973f9b48092676029ff58485ec8b8dece7387b 100644
--- a/net/minecraft/network/chat/contents/TranslatableContents.java
+++ b/net/minecraft/network/chat/contents/TranslatableContents.java
@@ -177,6 +177,16 @@ public class TranslatableContents implements ComponentContents {
 
     @Override
     public <T> Optional<T> visit(final FormattedText.ContentConsumer<T> output) {
+        // Paper start - Count visited parts
+        try {
+            return this.visit(new TranslatableContentConsumer<>(output));
+        } catch (IllegalArgumentException ignored) {
+            return output.accept("...");
+        }
+    }
+
+    private <T> Optional<T> visit(TranslatableContentConsumer<T> output) {
+        // Paper end - Count visited parts
         this.decompose();
 
         for (FormattedText part : this.decomposedParts) {
@@ -189,6 +199,27 @@ public class TranslatableContents implements ComponentContents {
         return Optional.empty();
     }
 
+    // Paper start - Count visited parts
+    private static final class TranslatableContentConsumer<T> implements FormattedText.ContentConsumer<T> {
+        private static final IllegalArgumentException NESTED_TOO_LONG = new IllegalArgumentException("Too long");
+
+        private final FormattedText.ContentConsumer<T> visitor;
+        private int visited;
+
+        private TranslatableContentConsumer(final FormattedText.ContentConsumer<T> visitor) {
+            this.visitor = visitor;
+        }
+
+        @Override
+        public Optional<T> accept(final String asString) {
+            if (this.visited++ > 32) {
+                throw NESTED_TOO_LONG;
+            }
+            return this.visitor.accept(asString);
+        }
+    }
+    // Paper end - Count visited parts
+
     @Override
     public MutableComponent resolve(final ResolutionContext context, final int recursionDepth) throws CommandSyntaxException {
         Object[] argsCopy = new Object[this.args.length];
