From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/advancements/AdvancementHolder.java b/net/minecraft/advancements/AdvancementHolder.java
index 3a0891109dd18e2d2b7304fbb8d96522172f9ea8..4aaa9fd163acc55a40373f221151451b6995a408 100644
--- a/net/minecraft/advancements/AdvancementHolder.java
+++ b/net/minecraft/advancements/AdvancementHolder.java
@@ -26,4 +26,10 @@ public record AdvancementHolder(Identifier id, Advancement value) {
     public String toString() {
         return this.id.toString();
     }
+
+    // CraftBukkit start
+    public final org.bukkit.advancement.Advancement toBukkit() {
+        return new org.bukkit.craftbukkit.advancement.CraftAdvancement(this);
+    }
+    // CraftBukkit end
 }
