From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/advancements/AdvancementTree.java b/net/minecraft/advancements/AdvancementTree.java
index f531de20d7e3a860b85e460d17df5092d28deffd..3e8d68de1653c3b4b8e7c3ba3b735a98e06c1f2f 100644
--- a/net/minecraft/advancements/AdvancementTree.java
+++ b/net/minecraft/advancements/AdvancementTree.java
@@ -25,7 +25,7 @@ public class AdvancementTree {
             this.remove(child);
         }
 
-        LOGGER.info("Forgot about advancement {}", node.holder());
+        LOGGER.debug("Forgot about advancement {}", node.holder()); // Paper - Improve logging and errors
         this.nodes.remove(node.holder().id());
         if (node.parent() == null) {
             this.roots.remove(node);
@@ -61,7 +61,7 @@ public class AdvancementTree {
             }
         }
 
-        LOGGER.info("Loaded {} advancements", this.nodes.size());
+        // LOGGER.info("Loaded {} advancements", this.nodes.size()); // CraftBukkit - moved to ServerAdvancementManager#apply // Paper - Improve logging and errors; you say it was moved... but it wasn't :) it should be moved however, since this is called when the API creates an advancement
     }
 
     private boolean tryInsert(final AdvancementHolder holder) {
