From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/util/parsing/packrat/Scope.java b/net/minecraft/util/parsing/packrat/Scope.java
index 95ae183548d85c7149b003d151983e66391309fb..4880f16347c99dc5c1a06fbbf0af197c95f3c9e7 100644
--- a/net/minecraft/util/parsing/packrat/Scope.java
+++ b/net/minecraft/util/parsing/packrat/Scope.java
@@ -265,4 +265,31 @@ public final class Scope {
 
         return true;
     }
+
+    // Paper start - track depth
+    private int depth;
+
+    private static final Term<?> INCREASING_DEPTH_TERM = (parseState, scope, control) -> {
+        if (++scope.depth > 512) {
+            parseState.errorCollector().store(parseState.mark(), new IllegalStateException("Too deep"));
+            return false;
+        }
+        return true;
+    };
+
+    @SuppressWarnings({"unchecked"})
+    public static <S> Term<S> increaseDepth() {
+        return (Term<S>) INCREASING_DEPTH_TERM;
+    }
+
+    private static final Term<?> DECREASING_DEPTH_TERM = (parseState, scope, control) -> {
+        scope.depth--;
+        return true;
+    };
+
+    @SuppressWarnings({"unchecked"})
+    public static <S> Term<S> decreaseDepth() {
+        return (Term<S>) DECREASING_DEPTH_TERM;
+    }
+    // Paper end - track depth
 }
