From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/level/progress/LoggingLevelLoadListener.java b/net/minecraft/server/level/progress/LoggingLevelLoadListener.java
index 7013ca8be9437f54b3cfd49cd9f6125dc1765afe..0b05d6d05d007962d43a2e47c8244bff2dbf1ee9 100644
--- a/net/minecraft/server/level/progress/LoggingLevelLoadListener.java
+++ b/net/minecraft/server/level/progress/LoggingLevelLoadListener.java
@@ -17,17 +17,21 @@ public class LoggingLevelLoadListener implements LevelLoadListener {
     private long startTime = Long.MAX_VALUE;
     private long nextLogTime = Long.MAX_VALUE;
 
-    public LoggingLevelLoadListener(final boolean includePlayerChunks) {
+    // Paper start - add level
+    private final net.minecraft.server.level.ServerLevel level;
+    public LoggingLevelLoadListener(final boolean includePlayerChunks, net.minecraft.server.level.ServerLevel level) {
+        this.level = level;
+        // Paper end - add level
         this.includePlayerChunks = includePlayerChunks;
         this.progressTracker = new LevelLoadProgressTracker(includePlayerChunks);
     }
 
     public static LoggingLevelLoadListener forDedicatedServer() {
-        return new LoggingLevelLoadListener(false);
+        return new LoggingLevelLoadListener(false, null); // Paper
     }
 
     public static LoggingLevelLoadListener forSingleplayer() {
-        return new LoggingLevelLoadListener(true);
+        return new LoggingLevelLoadListener(true, null); // Paper
     }
 
     @Override
@@ -42,10 +46,20 @@ public class LoggingLevelLoadListener implements LevelLoadListener {
             this.progressTracker.start(stage, totalChunks);
             switch (stage) {
                 case PREPARE_GLOBAL_SPAWN:
+                    // Paper start - log dimension
+                    if (this.level != null) {
+                        LOGGER.info("Selecting spawn point for level '{}'...", this.level.dimension().identifier());
+                    } else {
                     LOGGER.info("Selecting global world spawn...");
+                    }
                     break;
                 case LOAD_INITIAL_CHUNKS:
+                    if (this.level != null) {
+                        LOGGER.info("Loading {} persistent chunks for level '{}'...", totalChunks, this.level.dimension().identifier());
+                    } else {
                     LOGGER.info("Loading {} persistent chunks...", totalChunks);
+                    }
+                    // Paper end - log dimension
                     break;
                 case LOAD_PLAYER_CHUNKS:
                     LOGGER.info("Loading {} chunks for player spawn...", totalChunks);
@@ -73,7 +87,7 @@ public class LoggingLevelLoadListener implements LevelLoadListener {
                 ? LevelLoadListener.Stage.LOAD_PLAYER_CHUNKS
                 : LevelLoadListener.Stage.LOAD_INITIAL_CHUNKS;
             if (stage == finalStage) {
-                LOGGER.info("Time elapsed: {} ms", Util.getMillis() - this.startTime);
+                LOGGER.info("Prepared spawn area in {} ms", Util.getMillis() - this.startTime); // Paper
                 this.nextLogTime = Long.MAX_VALUE;
                 this.closed = true;
             }
