From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/resources/NetworkRegistryLoadTask.java b/net/minecraft/resources/NetworkRegistryLoadTask.java
index 64dbf8e06fb862513fada31fe2d39564e044974c..2eaf22d8c95451e7757824262e665e5c5b477f2b 100644
--- a/net/minecraft/resources/NetworkRegistryLoadTask.java
+++ b/net/minecraft/resources/NetworkRegistryLoadTask.java
@@ -80,7 +80,7 @@ public class NetworkRegistryLoadTask<T> extends RegistryLoadTask<T> {
         }
 
         return Util.sequence(elements).thenAcceptAsync(pendingRegistrations -> {
-            this.registerElements(pendingRegistrations.stream());
+            this.registerElements(pendingRegistrations.stream(), null); // Paper - this code is only used on the client
             Map<TagKey<T>, List<Holder<T>>> pendingTags = TagLoader.loadTagsFromNetwork(registryEntries.tags(), this.readOnlyRegistry());
             this.registerTags(pendingTags);
         }, executor);
