From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/notifications/EmptyNotificationService.java b/net/minecraft/server/notifications/EmptyNotificationService.java
index d04e1f5afcc6f86d712d9a30c20868cf04498569..5056f1e601604b9db19037b9a2395b125b10679a 100644
--- a/net/minecraft/server/notifications/EmptyNotificationService.java
+++ b/net/minecraft/server/notifications/EmptyNotificationService.java
@@ -69,7 +69,7 @@ public class EmptyNotificationService implements NotificationService {
     }
 
     @Override
-    public <T> void onGameRuleChanged(final GameRule<T> gameRule, final T value) {
+    public <T> void onGameRuleChanged(net.minecraft.server.level.ServerLevel level, final GameRule<T> gameRule, final T value) { // Paper - per-world game rules (add level param)
     }
 
     @Override
