From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/notifications/NotificationManager.java b/net/minecraft/server/notifications/NotificationManager.java
index d2b70adf50a7e64ed47f09a80dd02aef933438b5..5638b5c7a02f3c19a130af33bfc20d502d13b592 100644
--- a/net/minecraft/server/notifications/NotificationManager.java
+++ b/net/minecraft/server/notifications/NotificationManager.java
@@ -107,8 +107,8 @@ public class NotificationManager implements NotificationService {
     }
 
     @Override
-    public <T> void onGameRuleChanged(final GameRule<T> gameRule, final T value) {
-        this.notificationServices.forEach(notificationService -> notificationService.onGameRuleChanged(gameRule, value));
+    public <T> void onGameRuleChanged(net.minecraft.server.level.ServerLevel level, final GameRule<T> gameRule, final T value) { // Paper - per-world game rules (add level param)
+        this.notificationServices.forEach(notificationService -> notificationService.onGameRuleChanged(level, gameRule, value)); // Paper - per-world game rules (add level param)
     }
 
     @Override
