From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/FireChargeItem.java b/net/minecraft/world/item/FireChargeItem.java
index d7e93b62e44eb181cad72ae4c03cea6dd75f707a..5d436e01bdb23a44af64a0cad920411ba27a40fa 100644
--- a/net/minecraft/world/item/FireChargeItem.java
+++ b/net/minecraft/world/item/FireChargeItem.java
@@ -36,12 +36,28 @@ public class FireChargeItem extends Item implements ProjectileItem {
         if (!CampfireBlock.canLight(blockState) && !CandleBlock.canLight(blockState) && !CandleCakeBlock.canLight(blockState)) {
             pos = pos.relative(context.getClickedFace());
             if (BaseFireBlock.canBePlacedAt(level, pos, context.getHorizontalDirection())) {
+                // CraftBukkit start - fire BlockIgniteEvent
+                if (org.bukkit.craftbukkit.event.CraftEventFactory.callBlockIgniteEvent(level, pos, org.bukkit.event.block.BlockIgniteEvent.IgniteCause.FIREBALL, context.getPlayer()).isCancelled()) {
+                    if (!context.getPlayer().getAbilities().instabuild) {
+                        context.getItemInHand().shrink(1);
+                    }
+                    return InteractionResult.PASS;
+                }
+                // CraftBukkit end
                 this.playSound(level, pos);
                 level.setBlockAndUpdate(pos, BaseFireBlock.getState(level, pos));
                 level.gameEvent(context.getPlayer(), GameEvent.BLOCK_PLACE, pos);
                 used = true;
             }
         } else {
+            // CraftBukkit start - fire BlockIgniteEvent
+            if (org.bukkit.craftbukkit.event.CraftEventFactory.callBlockIgniteEvent(level, pos, org.bukkit.event.block.BlockIgniteEvent.IgniteCause.FIREBALL, context.getPlayer()).isCancelled()) {
+                if (!context.getPlayer().getAbilities().instabuild) {
+                    context.getItemInHand().shrink(1);
+                }
+                return InteractionResult.PASS;
+            }
+            // CraftBukkit end
             this.playSound(level, pos);
             level.setBlockAndUpdate(pos, blockState.setValue(BlockStateProperties.LIT, true));
             level.gameEvent(context.getPlayer(), GameEvent.BLOCK_CHANGE, pos);
