From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/FishingRodItem.java b/net/minecraft/world/item/FishingRodItem.java
index 3e8c36f7807c5f2beb8fbb949402a72833c023f7..33d0dae9a1e132d220148b24b508795b6b13ec77 100644
--- a/net/minecraft/world/item/FishingRodItem.java
+++ b/net/minecraft/world/item/FishingRodItem.java
@@ -23,7 +23,7 @@ public class FishingRodItem extends Item {
         ItemStack itemStack = player.getItemInHand(hand);
         if (player.fishing != null) {
             if (!level.isClientSide()) {
-                int dmg = player.fishing.retrieve(itemStack);
+                int dmg = player.fishing.retrieve(itemStack, hand); // Paper - Add hand parameter to PlayerFishEvent
                 itemStack.hurtAndBreak(dmg, player, hand.asEquipmentSlot());
             }
 
@@ -39,20 +39,31 @@ public class FishingRodItem extends Item {
             );
             itemStack.causeUseVibration(player, GameEvent.ITEM_INTERACT_FINISH);
         } else {
-            level.playSound(
-                null,
-                player.getX(),
-                player.getY(),
-                player.getZ(),
-                SoundEvents.FISHING_BOBBER_THROW,
-                SoundSource.NEUTRAL,
-                0.5F,
-                0.4F / (level.getRandom().nextFloat() * 0.4F + 0.8F)
-            );
+            // CraftBukkit - moved down
             if (level instanceof ServerLevel serverLevel) {
                 int lureSpeed = (int)(EnchantmentHelper.getFishingTimeReduction(serverLevel, itemStack, player) * 20.0F);
                 int luck = EnchantmentHelper.getFishingLuckBonus(serverLevel, itemStack, player);
-                Projectile.spawnProjectile(new FishingHook(player, level, luck, lureSpeed), serverLevel, itemStack);
+                // CraftBukkit start
+                FishingHook fishingHook = new FishingHook(player, level, luck, lureSpeed);
+                org.bukkit.event.player.PlayerFishEvent playerFishEvent = new org.bukkit.event.player.PlayerFishEvent((org.bukkit.entity.Player) player.getBukkitEntity(), null, (org.bukkit.entity.FishHook) fishingHook.getBukkitEntity(), org.bukkit.craftbukkit.CraftEquipmentSlot.getHand(hand), org.bukkit.event.player.PlayerFishEvent.State.FISHING);
+                level.getCraftServer().getPluginManager().callEvent(playerFishEvent);
+
+                if (playerFishEvent.isCancelled()) {
+                    player.fishing = null;
+                    return InteractionResult.PASS;
+                }
+                level.playSound(
+                    null,
+                    player.getX(),
+                    player.getY(),
+                    player.getZ(),
+                    SoundEvents.FISHING_BOBBER_THROW,
+                    SoundSource.NEUTRAL,
+                    0.5F,
+                    0.4F / (level.getRandom().nextFloat() * 0.4F + 0.8F)
+                );
+                Projectile.spawnProjectile(fishingHook, serverLevel, itemStack);
+                // CraftBukkit end
             }
 
             player.awardStat(Stats.ITEM_USED.get(this));
