From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/ItemCooldowns.java b/net/minecraft/world/item/ItemCooldowns.java
index 54d2dbd89a363d58f9848664952e082708e26416..9f1285c172b4c3a19587da287c3336d0f49d0172 100644
--- a/net/minecraft/world/item/ItemCooldowns.java
+++ b/net/minecraft/world/item/ItemCooldowns.java
@@ -56,6 +56,13 @@ public class ItemCooldowns {
     }
 
     public void addCooldown(final Identifier cooldownGroup, final int time) {
+        // Paper start - Item cooldown events
+        this.addCooldown(cooldownGroup, time, true);
+    }
+
+    public void addCooldown(Identifier cooldownGroup, int time, boolean callEvent) {
+        // Event called in server override
+        // Paper end - Item cooldown events
         this.cooldowns.put(cooldownGroup, new ItemCooldowns.CooldownInstance(this.tickCount, this.tickCount + time));
         this.onCooldownStarted(cooldownGroup, time);
     }
@@ -73,4 +80,14 @@ public class ItemCooldowns {
 
     private record CooldownInstance(int startTime, int endTime) {
     }
+
+    // Paper start - util
+    public int getRemainingCooldown(final Identifier cooldownGroup) {
+        final CooldownInstance cooldown = this.cooldowns.get(cooldownGroup);
+        if (cooldown == null) {
+            return 0;
+        }
+        return Math.max(0, cooldown.endTime() - this.tickCount);
+    }
+    // Paper end
 }
