From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/LeadItem.java b/net/minecraft/world/item/LeadItem.java
index 8bfab0fcc3e77d8cafa71bff924dda53f254dc9c..5a03357629e1450155c816180b5954064422a7e0 100644
--- a/net/minecraft/world/item/LeadItem.java
+++ b/net/minecraft/world/item/LeadItem.java
@@ -27,14 +27,14 @@ public class LeadItem extends Item {
         if (state.is(BlockTags.FENCES)) {
             Player player = context.getPlayer();
             if (!level.isClientSide() && player != null) {
-                return bindPlayerMobs(player, level, pos);
+                return bindPlayerMobs(player, level, pos, context.getHand()); // CraftBukkit - Pass hand
             }
         }
 
         return InteractionResult.PASS;
     }
 
-    public static InteractionResult bindPlayerMobs(final Player player, final Level level, final BlockPos pos) {
+    public static InteractionResult bindPlayerMobs(final Player player, final Level level, final BlockPos pos, final net.minecraft.world.InteractionHand hand) { // CraftBukkit - Add InteractionHand
         List<Leashable> entitiesToLeash = Leashable.leashableInArea(level, Vec3.atCenterOf(pos), l -> l.getLeashHolder() == player);
         if (entitiesToLeash.isEmpty()) {
             return InteractionResult.PASS;
@@ -42,10 +42,22 @@ public class LeadItem extends Item {
 
         Optional<LeashFenceKnotEntity> existingKnot = LeashFenceKnotEntity.getKnot(level, pos);
         LeashFenceKnotEntity activeKnot = existingKnot.orElseGet(() -> LeashFenceKnotEntity.createKnot(level, pos));
+        // CraftBukkit start - fire HangingPlaceEvent
+        if (existingKnot.isEmpty()) {
+            org.bukkit.inventory.EquipmentSlot handSlot = org.bukkit.craftbukkit.CraftEquipmentSlot.getHand(hand);
+            org.bukkit.event.hanging.HangingPlaceEvent event = new org.bukkit.event.hanging.HangingPlaceEvent((org.bukkit.entity.Hanging) activeKnot.getBukkitEntity(), player != null ? (org.bukkit.entity.Player) player.getBukkitEntity() : null, org.bukkit.craftbukkit.block.CraftBlock.at(level, pos), org.bukkit.block.BlockFace.SELF, handSlot);
+            level.getCraftServer().getPluginManager().callEvent(event);
+
+            if (event.isCancelled()) {
+                activeKnot.discard();
+                return InteractionResult.PASS;
+            }
+        }
+        // CraftBukkit end
         boolean anyLeashed = false;
 
         for (Leashable leashable : entitiesToLeash) {
-            if (leashable.canHaveALeashAttachedTo(activeKnot)) {
+            if (leashable.canHaveALeashAttachedTo(activeKnot) && org.bukkit.craftbukkit.event.CraftEventFactory.handlePlayerLeashEntityEvent(leashable, activeKnot, player, hand)) { // Paper - leash event
                 leashable.setLeashedTo(activeKnot, true);
                 anyLeashed = true;
             }
@@ -63,4 +75,10 @@ public class LeadItem extends Item {
 
         return InteractionResult.PASS;
     }
+
+    // CraftBukkit start
+    public static InteractionResult bindPlayerMobs(Player player, Level world, BlockPos pos) {
+        return LeadItem.bindPlayerMobs(player, world, pos, net.minecraft.world.InteractionHand.MAIN_HAND);
+    }
+    // CraftBukkit end
 }
