From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/MapItem.java b/net/minecraft/world/item/MapItem.java
index 2c302b9aedb95df240a43242ca07a3982a528f8c..337b6aa7ee170a1ec9310a72d4d5f25176246c6b 100644
--- a/net/minecraft/world/item/MapItem.java
+++ b/net/minecraft/world/item/MapItem.java
@@ -103,10 +103,10 @@ public class MapItem extends Item {
                             int averagingAreaMinX = (centerX / scale + imgX - 64) * scale;
                             int averagingAreaMinZ = (centerZ / scale + imgY - 64) * scale;
                             Multiset<MapColor> colorCount = LinkedHashMultiset.create();
-                            LevelChunk chunk = level.getChunk(
+                            LevelChunk chunk = level.getChunkIfLoaded( // Paper - Maps shouldn't load chunks
                                 SectionPos.blockToSectionCoord(averagingAreaMinX), SectionPos.blockToSectionCoord(averagingAreaMinZ)
                             );
-                            if (!chunk.isEmpty()) {
+                            if (chunk != null && !chunk.isEmpty()) { // Paper - Maps shouldn't load chunks
                                 int waterDepth = 0;
                                 double averageAreaHeight = 0.0;
                                 if (level.dimensionType().hasCeiling()) {
@@ -213,7 +213,7 @@ public class MapItem extends Item {
 
                 for (int row = 0; row < 128; row++) {
                     for (int column = 0; column < 128; column++) {
-                        Holder<Biome> biome = level.getBiome(pos.set((unscaledStartX + column) * scale, 0, (unscaledStartZ + row) * scale));
+                        Holder<Biome> biome = level.getUncachedNoiseBiome(net.minecraft.core.QuartPos.fromBlock((unscaledStartX + column) * scale), net.minecraft.core.QuartPos.fromBlock(0), net.minecraft.core.QuartPos.fromBlock((unscaledStartZ + row) * scale)); // Paper - Perf: Use seed based lookup for treasure maps
                         isBiomeWatery[row * 128 + column] = biome.is(BiomeTags.WATER_ON_MAP_OUTLINES);
                     }
                 }
