From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/SplashPotionItem.java b/net/minecraft/world/item/SplashPotionItem.java
index 28f68b61b4f877eba9195f118425d7885ba6c673..5d1e3c88d1d4e20d0e69243779d149a807f71759 100644
--- a/net/minecraft/world/item/SplashPotionItem.java
+++ b/net/minecraft/world/item/SplashPotionItem.java
@@ -19,6 +19,10 @@ public class SplashPotionItem extends ThrowablePotionItem {
 
     @Override
     public InteractionResult use(final Level level, final Player player, final InteractionHand hand) {
+        // Paper start - PlayerLaunchProjectileEvent
+        final InteractionResult wrapper = super.use(level, player, hand);
+        if (wrapper instanceof InteractionResult.Fail) return wrapper;
+        // Paper end - PlayerLaunchProjectileEvent
         level.playSound(
             null,
             player.getX(),
@@ -29,7 +33,7 @@ public class SplashPotionItem extends ThrowablePotionItem {
             0.5F,
             0.4F / (level.getRandom().nextFloat() * 0.4F + 0.8F)
         );
-        return super.use(level, player, hand);
+        return wrapper; // Paper - PlayerLaunchProjectileEvent
     }
 
     @Override
