From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/chunk/ImposterProtoChunk.java b/net/minecraft/world/level/chunk/ImposterProtoChunk.java
index 32d06e9655629b9416dfe79637ec3966fd96fe8f..e8097b9abebf1496fdab3b845aa143b18029fd84 100644
--- a/net/minecraft/world/level/chunk/ImposterProtoChunk.java
+++ b/net/minecraft/world/level/chunk/ImposterProtoChunk.java
@@ -48,6 +48,13 @@ public class ImposterProtoChunk extends ProtoChunk {
         return this.wrapped.getBlockState(pos);
     }
 
+    // Paper start
+    @Override
+    public final BlockState getBlockState(final int x, final int y, final int z) {
+        return this.wrapped.getBlockStateFinal(x, y, z);
+    }
+    // Paper end
+
     @Override
     public FluidState getFluidState(final BlockPos pos) {
         return this.wrapped.getFluidState(pos);
