From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/chunk/ProtoChunk.java b/net/minecraft/world/level/chunk/ProtoChunk.java
index e460757018c4e38f1805cfd18dfde96e0c56dd75..3e19d1bf3b43135f1e132c9d8fa421d42bd8d34e 100644
--- a/net/minecraft/world/level/chunk/ProtoChunk.java
+++ b/net/minecraft/world/level/chunk/ProtoChunk.java
@@ -90,15 +90,32 @@ public class ProtoChunk extends ChunkAccess {
         return new ChunkAccess.PackedTicks(this.blockTicks.pack(currentTick), this.fluidTicks.pack(currentTick));
     }
 
+    // Paper start - If loaded util
+    @Override
+    public final FluidState getFluidIfLoaded(BlockPos pos) {
+        return this.getFluidState(pos);
+    }
+
+    @Override
+    public final BlockState getBlockStateIfLoaded(BlockPos pos) {
+        return this.getBlockState(pos);
+    }
+    // Paper end
+
     @Override
     public BlockState getBlockState(final BlockPos pos) {
-        int y = pos.getY();
+        // Paper start
+        return this.getBlockState(pos.getX(), pos.getY(), pos.getZ());
+    }
+
+    public BlockState getBlockState(final int x, final int y, final int z) {
+        // Paper end
         if (this.isOutsideBuildHeight(y)) {
             return Blocks.VOID_AIR.defaultBlockState();
         }
 
         LevelChunkSection section = this.getSection(this.getSectionIndex(y));
-        return section.hasOnlyAir() ? Blocks.AIR.defaultBlockState() : section.getBlockState(pos.getX() & 15, y & 15, pos.getZ() & 15);
+        return section.hasOnlyAir() ? Blocks.AIR.defaultBlockState() : section.getBlockState(x & 15, y & 15, z & 15); // Paper
     }
 
     @Override
