From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/equipment/Equippable.java b/net/minecraft/world/item/equipment/Equippable.java
index 0c6d44c2f3ff11f3cc9b0a70da420ccae294899d..b0c7b15632bf08808fb9d943f7b3462400f89f56 100644
--- a/net/minecraft/world/item/equipment/Equippable.java
+++ b/net/minecraft/world/item/equipment/Equippable.java
@@ -131,6 +131,19 @@ public record Equippable(
             ItemStack inEquipmentSlot = player.getItemBySlot(this.slot);
             if ((!EnchantmentHelper.has(inEquipmentSlot, EnchantmentEffectComponents.PREVENT_ARMOR_CHANGE) || player.isCreative())
                 && !ItemStack.isSameItemSameComponents(inHand, inEquipmentSlot)) {
+                // Paper start
+                final io.papermc.paper.event.player.PlayerSwapWithEquipmentSlotEvent event = new io.papermc.paper.event.player.PlayerSwapWithEquipmentSlotEvent(
+                    (org.bukkit.entity.Player) player.getBukkitEntity(),
+                    org.bukkit.craftbukkit.inventory.CraftItemStack.asBukkitCopy(inHand),
+                    org.bukkit.craftbukkit.CraftEquipmentSlot.getSlot(this.slot),
+                    org.bukkit.craftbukkit.inventory.CraftItemStack.asBukkitCopy(inEquipmentSlot)
+                );
+                event.callEvent();
+                if (event.isCancelled()) {
+                    ((org.bukkit.entity.Player) player.getBukkitEntity()).updateInventory();
+                    return InteractionResult.FAIL;
+                }
+                // Paper end
                 if (!player.level().isClientSide()) {
                     player.awardStat(Stats.ITEM_USED.get(inHand.getItem()));
                 }
