From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/raid/Raids.java b/net/minecraft/world/entity/raid/Raids.java
index d6952d4be0026e0a9144dcdd8af9430e26c48ec1..bc0f31a16f7b055e85f10eda377026a927032535 100644
--- a/net/minecraft/world/entity/raid/Raids.java
+++ b/net/minecraft/world/entity/raid/Raids.java
@@ -52,6 +52,7 @@ public class Raids extends SavedData {
     private Raids(final List<Raids.RaidWithId> raids, final int nextId, final int tick) {
         for (Raids.RaidWithId raid : raids) {
             this.raidMap.put(raid.id, raid.raid);
+            raid.raid.idOrNegativeOne = raid.id; // Paper - expose id of raids while method is kept around as deprecated for removal
         }
 
         this.nextId = nextId;
@@ -132,11 +133,23 @@ public class Raids extends SavedData {
         }
 
         Raid raid = this.getOrCreateRaid(level, raidCenterPos);
-        if (!raid.isStarted() && !this.raidMap.containsValue(raid)) {
-            this.raidMap.put(this.getUniqueId(), raid);
-        }
+        // CraftBukkit - moved down
+        // if (!raid.isStarted() && !this.raidMap.containsValue(raid)) {
+        //     this.raidMap.put(this.getUniqueId(), raid);
+        // }
+
+        if (!raid.isStarted() || (raid.isInProgress() && raid.getRaidOmenLevel() < raid.getMaxRaidOmenLevel())) { // CraftBukkit - fixed a bug with raid: players could add up Bad Omen level even when the raid had finished
+            // CraftBukkit start
+            if (!org.bukkit.craftbukkit.event.CraftEventFactory.callRaidTriggerEvent(level, raid, player)) {
+                player.removeEffect(net.minecraft.world.effect.MobEffects.RAID_OMEN);
+                return null;
+            }
 
-        if (!raid.isStarted() || raid.getRaidOmenLevel() < raid.getMaxRaidOmenLevel()) {
+            if (!raid.isStarted() && !this.raidMap.containsValue(raid)) {
+                this.raidMap.put(this.getUniqueId(), raid);
+                raid.idOrNegativeOne = this.nextId; // Paper - expose id of raids while method is kept around as deprecated for removal
+            }
+            // CraftBukkit end
             raid.absorbRaidOmen(player);
         }
 
