From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/boss/enderdragon/phases/DragonStrafePlayerPhase.java b/net/minecraft/world/entity/boss/enderdragon/phases/DragonStrafePlayerPhase.java
index 95786cea4b8da6009c4add4862793edd8778d877..b07a36b1b2f4d009b10b9df864f285f7be100709 100644
--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonStrafePlayerPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonStrafePlayerPhase.java
@@ -69,13 +69,16 @@ public class DragonStrafePlayerPhase extends AbstractDragonPhaseInstance {
                         double ydd = this.attackTarget.getY(0.5) - startingY;
                         double zdd = this.attackTarget.getZ() - startingZ;
                         Vec3 direction = new Vec3(xdd, ydd, zdd);
-                        if (!this.dragon.isSilent()) {
+                        if (false && !this.dragon.isSilent()) { // Paper - EnderDragon Events; Fire after shoot fireball event
                             level.levelEvent(null, LevelEvent.SOUND_DRAGON_FIREBALL, this.dragon.blockPosition(), 0);
                         }
 
                         DragonFireball entity = new DragonFireball(level, this.dragon, direction.normalize());
                         entity.snapTo(startingX, startingY, startingZ, 0.0F, 0.0F);
+                        if (new com.destroystokyo.paper.event.entity.EnderDragonShootFireballEvent((org.bukkit.entity.EnderDragon) this.dragon.getBukkitEntity(), (org.bukkit.entity.DragonFireball) entity.getBukkitEntity()).callEvent()) { // Paper - EnderDragon Events
+                            if (!this.dragon.isSilent()) level.levelEvent(null, LevelEvent.SOUND_DRAGON_FIREBALL, this.dragon.blockPosition(), 0); // Paper - EnderDragon Events; Fire after shoot fireball event
                         level.addFreshEntity(entity);
+                        } else entity.discard(null); // Paper - EnderDragon Events
                         this.fireballCharge = 0;
                         if (this.currentPath != null) {
                             while (!this.currentPath.isDone()) {
