From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingFlamingPhase.java b/net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingFlamingPhase.java
index 9b90e39609b18dde40250eed8f0284af17994591..d368ae059ca0367bf6ec8cb7698d2463742eeee0 100644
--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingFlamingPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingFlamingPhase.java
@@ -87,7 +87,13 @@ public class DragonSittingFlamingPhase extends AbstractDragonSittingPhase {
             this.flame.setCustomParticle(PowerParticleOption.create(ParticleTypes.DRAGON_BREATH, 1.0F));
             this.flame.setPotionDurationScale(0.25F);
             this.flame.addEffect(new MobEffectInstance(MobEffects.INSTANT_DAMAGE));
+            if (new com.destroystokyo.paper.event.entity.EnderDragonFlameEvent((org.bukkit.entity.EnderDragon) this.dragon.getBukkitEntity(), (org.bukkit.entity.AreaEffectCloud) this.flame.getBukkitEntity()).callEvent()) { // Paper - EnderDragon Events
             level.addFreshEntity(this.flame);
+            // Paper start - EnderDragon Events
+            } else {
+                this.end();
+            }
+            // Paper end - EnderDragon Events
         }
     }
 
@@ -100,7 +106,7 @@ public class DragonSittingFlamingPhase extends AbstractDragonSittingPhase {
     @Override
     public void end() {
         if (this.flame != null) {
-            this.flame.discard();
+            this.flame.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.DESPAWN); // CraftBukkit - add Bukkit remove cause
             this.flame = null;
         }
     }
