From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/boss/enderdragon/phases/EnderDragonPhaseManager.java b/net/minecraft/world/entity/boss/enderdragon/phases/EnderDragonPhaseManager.java
index f9a17b67152a70bc164f4cdc62cbaa57f538dc0f..21ff455aee8be31b5e905dced613978ea6ffa2cd 100644
--- a/net/minecraft/world/entity/boss/enderdragon/phases/EnderDragonPhaseManager.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/EnderDragonPhaseManager.java
@@ -17,12 +17,24 @@ public class EnderDragonPhaseManager {
         this.setPhase(EnderDragonPhase.HOVERING);
     }
 
-    public void setPhase(final EnderDragonPhase<?> target) {
+    public void setPhase(EnderDragonPhase<?> target) { // CraftBukkit - Call EnderDragonChangePhaseEvent
         if (this.currentPhase == null || target != this.currentPhase.getPhase()) {
             if (this.currentPhase != null) {
                 this.currentPhase.end();
             }
 
+            // CraftBukkit start - Call EnderDragonChangePhaseEvent
+            org.bukkit.event.entity.EnderDragonChangePhaseEvent event = new org.bukkit.event.entity.EnderDragonChangePhaseEvent(
+                (org.bukkit.craftbukkit.entity.CraftEnderDragon) this.dragon.getBukkitEntity(),
+                (this.currentPhase == null) ? null : org.bukkit.craftbukkit.entity.CraftEnderDragon.getBukkitPhase(this.currentPhase.getPhase()),
+                org.bukkit.craftbukkit.entity.CraftEnderDragon.getBukkitPhase(target)
+            );
+            if (!event.callEvent()) {
+                return;
+            }
+            target = org.bukkit.craftbukkit.entity.CraftEnderDragon.getMinecraftPhase(event.getNewPhase());
+            // CraftBukkit end
+
             this.currentPhase = this.getPhase((EnderDragonPhase<DragonPhaseInstance>)target);
             if (!this.dragon.level().isClientSide()) {
                 this.dragon.getEntityData().set(EnderDragon.DATA_PHASE, target.getId());
