From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/behavior/GoToWantedItem.java b/net/minecraft/world/entity/ai/behavior/GoToWantedItem.java
index 82d180a9b061fb70a6d3a3042072210c3c0219d4..aff2db230f992b1cd139de0997497322f0efc419 100644
--- a/net/minecraft/world/entity/ai/behavior/GoToWantedItem.java
+++ b/net/minecraft/world/entity/ai/behavior/GoToWantedItem.java
@@ -37,6 +37,21 @@ public class GoToWantedItem {
                                 && item.closerThan(body, maxDistToWalk)
                                 && body.level().getWorldBorder().isWithinBounds(item.blockPosition())
                                 && body.canPickUpLoot()) {
+                                // CraftBukkit start
+                                if (body instanceof net.minecraft.world.entity.animal.allay.Allay) {
+                                    org.bukkit.event.entity.EntityTargetEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callEntityTargetEvent(body, item, org.bukkit.event.entity.EntityTargetEvent.TargetReason.CLOSEST_ENTITY);
+
+                                    if (event.isCancelled()) {
+                                        return false;
+                                    }
+                                    if (!(event.getTarget() instanceof org.bukkit.craftbukkit.entity.CraftItem targetItem)) { // Paper - only erase allay memory on non-item targets
+                                        wantedItem.erase();
+                                        return false; // Paper - only erase allay memory on non-item targets
+                                    }
+
+                                    item = targetItem.getHandle();
+                                }
+                                // CraftBukkit end
                                 WalkTarget target = new WalkTarget(new EntityTracker(item, false), speedModifier, 0);
                                 lookTarget.set(new EntityTracker(item, true));
                                 walkTarget.set(target);
