From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java b/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
index 1ec957fc806389710c1647dbfdce8aef3491b0f1..8ab7f9d3ae6f4fd4dbad9a7149f0db96d19d3909 100644
--- a/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
+++ b/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
@@ -56,6 +56,12 @@ public class InteractWithDoor {
                             if (fromState.is(BlockTags.MOB_INTERACTABLE_DOORS, s -> s.getBlock() instanceof DoorBlock)) {
                                 DoorBlock fromBlock = (DoorBlock)fromState.getBlock();
                                 if (!fromBlock.isOpen(fromState)) {
+                                    // CraftBukkit start - entities opening doors
+                                    org.bukkit.event.entity.EntityInteractEvent event = new org.bukkit.event.entity.EntityInteractEvent(body.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.at(body.level(), fromPos));
+                                    if (!event.callEvent()) {
+                                        return false;
+                                    }
+                                    // CraftBukkit end - entities opening doors
                                     fromBlock.setOpen(body, level, fromState, fromPos, true);
                                 }
 
@@ -67,6 +73,12 @@ public class InteractWithDoor {
                             if (toState.is(BlockTags.MOB_INTERACTABLE_DOORS, s -> s.getBlock() instanceof DoorBlock)) {
                                 DoorBlock door = (DoorBlock)toState.getBlock();
                                 if (!door.isOpen(toState)) {
+                                    // CraftBukkit start - entities opening doors
+                                    org.bukkit.event.entity.EntityInteractEvent event = new org.bukkit.event.entity.EntityInteractEvent(body.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.at(body.level(), toPos));
+                                    if (!event.callEvent()) {
+                                        return false;
+                                    }
+                                    // CraftBukkit end - entities opening doors
                                     door.setOpen(body, level, toState, toPos, true);
                                     doors = rememberDoorToClose(doorsMemory, doors, level, toPos);
                                 }
