From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java b/net/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
index e3c70d35bd9f45050d4edf1719ccea49bd126d1f..c507cbb1b0d89f8fe5a00e67eea510f7516fe040 100644
--- a/net/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
+++ b/net/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
@@ -22,13 +22,32 @@ public class PoiCompetitorScan {
                         level.getPoiManager()
                             .getType(pos.pos())
                             .ifPresent(
-                                poiType -> i.<List<LivingEntity>>get(nearestEntities)
-                                    .stream()
-                                    .filter(v -> v instanceof Villager && v != body)
-                                    .map(v -> (Villager)v)
-                                    .filter(LivingEntity::isAlive)
-                                    .filter(nearbyVillager -> competesForSameJobsite(pos, poiType, nearbyVillager))
-                                    .reduce(body, PoiCompetitorScan::selectWinner)
+                                // Paper start - Improve performance of PoiCompetitorScan by unrolling stream
+                                // The previous logic used Stream#reduce to simulate a form of single-iteration bubble sort
+                                // in which the "winning" villager would maintain MemoryModuleType.JOB_SITE while all others
+                                // would lose said memory module type by passing each "current winner" and incoming next
+                                // villager to #selectWinner.
+                                poiType -> {
+                                    final List<LivingEntity> livingEntities = i.get(nearestEntities);
+
+                                    Villager winner = body;
+                                    for (final LivingEntity other : livingEntities) {
+                                        if (other == body) {
+                                            continue;
+                                        }
+                                        if (!(other instanceof final Villager otherVillager)) {
+                                            continue;
+                                        }
+                                        if (!other.isAlive()) {
+                                            continue;
+                                        }
+                                        if (!competesForSameJobsite(pos, poiType, otherVillager)) {
+                                            continue;
+                                        }
+                                        winner = selectWinner(winner, otherVillager);
+                                    }
+                                }
+                                // Paper end - Improve performance of PoiCompetitorScan by unrolling stream
                             );
                         return true;
                     }
