From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/behavior/StartAttacking.java b/net/minecraft/world/entity/ai/behavior/StartAttacking.java
index 7d86e7d0d2c9ee69d3a9845b8be2a7d863c07ee8..f28b65e23b58d5d78b77111b2b2596021c298d65 100644
--- a/net/minecraft/world/entity/ai/behavior/StartAttacking.java
+++ b/net/minecraft/world/entity/ai/behavior/StartAttacking.java
@@ -32,6 +32,17 @@ public class StartAttacking {
                         return false;
                     }
 
+                    // CraftBukkit start
+                    org.bukkit.event.entity.EntityTargetEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callEntityTargetLivingEvent(body, targetEntity, (targetEntity instanceof net.minecraft.server.level.ServerPlayer) ? org.bukkit.event.entity.EntityTargetEvent.TargetReason.CLOSEST_PLAYER : org.bukkit.event.entity.EntityTargetEvent.TargetReason.CLOSEST_ENTITY);
+                    if (event.isCancelled()) {
+                        return false;
+                    }
+                    if (event.getTarget() == null) {
+                        attackTarget.erase();
+                        return true;
+                    }
+                    targetEntity = ((org.bukkit.craftbukkit.entity.CraftLivingEntity) event.getTarget()).getHandle();
+                    // CraftBukkit end
                     attackTarget.set(targetEntity);
                     cantReachSince.erase();
                     return true;
