From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/behavior/StopAttackingIfTargetInvalid.java b/net/minecraft/world/entity/ai/behavior/StopAttackingIfTargetInvalid.java
index 0c9808ac9535345131fa72625f2e7d6b291666a4..53574a1801344cea3d9ef9b364c411c8ada608dd 100644
--- a/net/minecraft/world/entity/ai/behavior/StopAttackingIfTargetInvalid.java
+++ b/net/minecraft/world/entity/ai/behavior/StopAttackingIfTargetInvalid.java
@@ -41,6 +41,30 @@ public class StopAttackingIfTargetInvalid {
                             return true;
                         }
 
+                        // Paper start - better track target change reason
+                        final org.bukkit.event.entity.EntityTargetEvent.TargetReason reason;
+                        if (!body.canAttack(target)) {
+                            reason = org.bukkit.event.entity.EntityTargetEvent.TargetReason.TARGET_INVALID;
+                        } else if (canGrowTiredOfTryingToReachTarget && StopAttackingIfTargetInvalid.isTiredOfTryingToReachTarget(body, i.tryGet(cantReachSince))) {
+                            reason = org.bukkit.event.entity.EntityTargetEvent.TargetReason.FORGOT_TARGET;
+                        } else if (!target.isAlive()) {
+                            reason = org.bukkit.event.entity.EntityTargetEvent.TargetReason.TARGET_DIED;
+                        } else if (target.level() != body.level()) {
+                            reason = org.bukkit.event.entity.EntityTargetEvent.TargetReason.TARGET_OTHER_LEVEL;
+                        } else {
+                            reason = org.bukkit.event.entity.EntityTargetEvent.TargetReason.TARGET_INVALID;
+                        }
+                        // Paper end
+                        // CraftBukkit start
+                        org.bukkit.event.entity.EntityTargetEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callEntityTargetLivingEvent(body, null, reason); // Paper
+                        if (event.isCancelled()) {
+                            return false;
+                        }
+                        if (event.getTarget() != null) {
+                            body.getBrain().setMemory(MemoryModuleType.ATTACK_TARGET, ((org.bukkit.craftbukkit.entity.CraftLivingEntity) event.getTarget()).getHandle());
+                            return true;
+                        }
+                        // CraftBukkit end
                         onTargetErased.accept(level, body, target);
                         attackTarget.erase();
                         return true;
