From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/ai/behavior/TransportItemsBetweenContainers.java b/net/minecraft/world/entity/ai/behavior/TransportItemsBetweenContainers.java
index c5baabf7b8ad4d8c608325d9bab08c4ac410aeea..10ff003e12032d0d6a595e24d055e81915ea39d6 100644
--- a/net/minecraft/world/entity/ai/behavior/TransportItemsBetweenContainers.java
+++ b/net/minecraft/world/entity/ai/behavior/TransportItemsBetweenContainers.java
@@ -329,7 +329,10 @@ public class TransportItemsBetweenContainers extends Behavior<PathfinderMob> {
 
         boolean isValidTarget = this.isWantedBlock(body, transportItemTarget.state)
             && !this.isPositionAlreadyVisited(visitedPositions, unreachablePositions, transportItemTarget, level)
-            && !this.isContainerLocked(transportItemTarget);
+            // Paper start - ItemTransportingEntityValidateTargetEvent
+            && !this.isContainerLocked(transportItemTarget)
+            && org.bukkit.craftbukkit.event.CraftEventFactory.callTransporterValidateTarget(body, level, transportItemTarget.pos);
+            // Paper end - ItemTransportingEntityValidateTargetEvent
         return isValidTarget ? transportItemTarget : null;
     }
 
