From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/consume_effects/TeleportRandomlyConsumeEffect.java b/net/minecraft/world/item/consume_effects/TeleportRandomlyConsumeEffect.java
index ade4ce11280cd28d2ae9fe094c60fc16b55b799c..23ba133fea6e44dfd31e564d9544ec43800d426b 100644
--- a/net/minecraft/world/item/consume_effects/TeleportRandomlyConsumeEffect.java
+++ b/net/minecraft/world/item/consume_effects/TeleportRandomlyConsumeEffect.java
@@ -54,7 +54,13 @@ public record TeleportRandomlyConsumeEffect(float diameter) implements ConsumeEf
             }
 
             Vec3 oldPos = user.position();
-            if (user.randomTeleport(xx, yy, zz, true)) {
+            // CraftBukkit start - handle canceled status of teleport event
+            java.util.Optional<Boolean> status = user.randomTeleport(xx, yy, zz, true, org.bukkit.event.player.PlayerTeleportEvent.TeleportCause.CONSUMABLE_EFFECT);
+
+            // teleport event was canceled, no more tries
+            if (status.isEmpty()) break;
+            if (status.get()) {
+                // CraftBukkit end
                 level.gameEvent(GameEvent.TELEPORT, oldPos, GameEvent.Context.of(user));
                 SoundSource soundSource;
                 SoundEvent soundEvent;
