From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/component/BlocksAttacks.java b/net/minecraft/world/item/component/BlocksAttacks.java
index 324735d065c86892a8cd6f92d6cfb470fc26560c..e262a5aab379097046d63a8a82cf260eecebcc3f 100644
--- a/net/minecraft/world/item/component/BlocksAttacks.java
+++ b/net/minecraft/world/item/component/BlocksAttacks.java
@@ -85,11 +85,15 @@ public record BlocksAttacks(
             );
     }
 
-    public void disable(final ServerLevel level, final LivingEntity user, final float baseSeconds, final ItemStack blockingWith) {
+    public void disable(final ServerLevel level, final LivingEntity user, final float baseSeconds, final ItemStack blockingWith, LivingEntity attacker) { // Paper
         int cooldownTicks = this.disableBlockingForTicks(baseSeconds);
         if (cooldownTicks > 0) {
             if (user instanceof Player player) {
-                player.getCooldowns().addCooldown(blockingWith, cooldownTicks);
+                // Paper start
+                final io.papermc.paper.event.player.PlayerShieldDisableEvent shieldDisableEvent = new io.papermc.paper.event.player.PlayerShieldDisableEvent((org.bukkit.entity.Player) player.getBukkitEntity(), attacker.getBukkitEntity(), cooldownTicks);
+                if (!shieldDisableEvent.callEvent()) return;
+                player.getCooldowns().addCooldown(blockingWith, shieldDisableEvent.getCooldown());
+                // Paper end
             }
 
             user.stopUsingItem();
