From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/component/ConsumableListener.java b/net/minecraft/world/item/component/ConsumableListener.java
index be8de7a53d12db86eaac477ce66baac6139aa879..663c045a58f7ee10ea472ae67b2239ecfce8ce34 100644
--- a/net/minecraft/world/item/component/ConsumableListener.java
+++ b/net/minecraft/world/item/component/ConsumableListener.java
@@ -6,4 +6,6 @@ import net.minecraft.world.level.Level;
 
 public interface ConsumableListener {
     void onConsume(final Level level, final LivingEntity user, final ItemStack stack, final Consumable consumable);
+
+    default void cancelUsingItem(net.minecraft.server.level.ServerPlayer player, ItemStack stack, java.util.List<net.minecraft.network.protocol.Packet<? super net.minecraft.network.protocol.game.ClientGamePacketListener>> collectedPackets) {} // CraftBukkit // Paper - properly resend entities - collect packets for bundle
 }
