From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/component/CustomData.java b/net/minecraft/world/item/component/CustomData.java
index 926b38a2b29b604f01cf0e07b6b0c3c9c92481e5..587e4bcd87a5d995037728772bd5b6c990b4c4a1 100644
--- a/net/minecraft/world/item/component/CustomData.java
+++ b/net/minecraft/world/item/component/CustomData.java
@@ -13,7 +13,17 @@ import net.minecraft.world.item.ItemStack;
 
 public final class CustomData {
     public static final CustomData EMPTY = new CustomData(new CompoundTag());
-    public static final Codec<CompoundTag> COMPOUND_TAG_CODEC = Codec.withAlternative(CompoundTag.CODEC, TagParser.FLATTENED_CODEC);
+    // Paper start - Item serialization as json
+    public static ThreadLocal<Boolean> SERIALIZE_CUSTOM_AS_SNBT = ThreadLocal.withInitial(() -> false);
+    public static final Codec<CompoundTag> COMPOUND_TAG_CODEC = Codec.either(CompoundTag.CODEC, TagParser.FLATTENED_CODEC)
+        .xmap(com.mojang.datafixers.util.Either::unwrap, data -> { // Both will be used for deserialization, but we decide which one to use for serialization
+            if (!SERIALIZE_CUSTOM_AS_SNBT.get()) {
+                return com.mojang.datafixers.util.Either.left(data); // First codec
+            } else {
+                return com.mojang.datafixers.util.Either.right(data); // Second codec
+            }
+        });
+    // Paper end - Item serialization as json
     public static final Codec<CustomData> CODEC = COMPOUND_TAG_CODEC.xmap(CustomData::new, data -> data.tag);
     @Deprecated
     public static final StreamCodec<ByteBuf, CustomData> STREAM_CODEC = ByteBufCodecs.COMPOUND_TAG.map(CustomData::new, data -> data.tag);
@@ -62,6 +72,17 @@ public final class CustomData {
         return this.tag.copy();
     }
 
+    // Paper start - expose unsafe internal compound tag for read only access
+    @Deprecated
+    public CompoundTag getUnsafe() {
+        return this.tag;
+    }
+
+    public boolean contains(String key) {
+        return this.tag.contains(key);
+    }
+    // Paper end - expose unsafe internal compound tag for read only access
+
     @Override
     public boolean equals(final Object obj) {
         return obj == this || obj instanceof CustomData customData && this.tag.equals(customData.tag);
