From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/item/component/WrittenBookContent.java b/net/minecraft/world/item/component/WrittenBookContent.java
index e45fa1529bced0c4b896828b158fa5cadffd6de7..d026fecee59f3507f421c6f6259a5dcfeac050bf 100644
--- a/net/minecraft/world/item/component/WrittenBookContent.java
+++ b/net/minecraft/world/item/component/WrittenBookContent.java
@@ -91,7 +91,7 @@ public record WrittenBookContent(Filterable<String> title, String author, int ge
 
     public static boolean resolveForItem(final ItemStack itemStack, final ResolutionContext context, final HolderLookup.Provider registries) {
         WrittenBookContent content = itemStack.get(DataComponents.WRITTEN_BOOK_CONTENT);
-        if (content != null && !content.resolved()) {
+        if (io.papermc.paper.configuration.GlobalConfiguration.get().itemValidation.resolveSelectorsInBooks && content != null && !content.resolved()) { // Paper - Disable component selector resolving in books by default
             WrittenBookContent resolvedContent = content.resolve(context, registries);
             if (resolvedContent != null) {
                 itemStack.set(DataComponents.WRITTEN_BOOK_CONTENT, resolvedContent);
