--- a/net/minecraft/network/protocol/game/ClientboundLevelChunkWithLightPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundLevelChunkWithLightPacket.java
@@ -40,6 +_,12 @@
         );
     }
 
+    // Paper start - Anti-Xray
+    public void setReady(final boolean ready) {
+        // Empty hook, updated by feature patch
+    }
+    // Paper end - Anti-Xray
+
     @Override
     public PacketType<ClientboundLevelChunkWithLightPacket> type() {
         return GamePacketTypes.CLIENTBOUND_LEVEL_CHUNK_WITH_LIGHT;
