From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java b/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
index b7d2e7b0fc6a2cd78d5428b69b4e76cf518aa5ce..05baba9103362ce3bd840867695a76c323785180 100644
--- a/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
@@ -19,6 +19,13 @@ public class ClientboundSetEquipmentPacket implements Packet<ClientGamePacketLis
     private final List<Pair<EquipmentSlot, ItemStack>> slots;
 
     public ClientboundSetEquipmentPacket(final int entity, final List<Pair<EquipmentSlot, ItemStack>> slots) {
+    // Paper start - data sanitization
+        this(entity, slots, false);
+    }
+    private boolean sanitize;
+    public ClientboundSetEquipmentPacket(int entity, List<Pair<EquipmentSlot, ItemStack>> slots, boolean sanitize) {
+        this.sanitize = sanitize;
+    // Paper end - data sanitization
         this.entity = entity;
         this.slots = slots;
     }
@@ -40,6 +47,7 @@ public class ClientboundSetEquipmentPacket implements Packet<ClientGamePacketLis
         output.writeVarInt(this.entity);
         int size = this.slots.size();
 
+        try (final io.papermc.paper.util.sanitizer.ItemObfuscationSession ignored = io.papermc.paper.util.sanitizer.ItemObfuscationSession.start(this.sanitize ? io.papermc.paper.configuration.GlobalConfiguration.get().anticheat.obfuscation.items.binding.level : io.papermc.paper.util.sanitizer.ItemObfuscationSession.ObfuscationLevel.NONE)) { // Paper - data sanitization
         for (int i = 0; i < size; i++) {
             Pair<EquipmentSlot, ItemStack> e = this.slots.get(i);
             EquipmentSlot slotType = e.getFirst();
@@ -48,6 +56,7 @@ public class ClientboundSetEquipmentPacket implements Packet<ClientGamePacketLis
             output.writeByte(shouldContinue ? slotId | -128 : slotId);
             ItemStack.OPTIONAL_STREAM_CODEC.encode(output, e.getSecond());
         }
+        } // Paper - data sanitization
     }
 
     @Override
