From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/protocol/game/ClientboundSetPlayerTeamPacket.java b/net/minecraft/network/protocol/game/ClientboundSetPlayerTeamPacket.java
index ad3cfaa19b1512d1492441a2a02e460e5abb79c0..c5b2d63119a96e6783ff2171af6c6a9a77340cba 100644
--- a/net/minecraft/network/protocol/game/ClientboundSetPlayerTeamPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetPlayerTeamPacket.java
@@ -61,6 +61,12 @@ public class ClientboundSetPlayerTeamPacket implements Packet<ClientGamePacketLi
         );
     }
 
+    // Paper start - Multiple Entries with Scoreboards
+    public static ClientboundSetPlayerTeamPacket createMultiplePlayerPacket(PlayerTeam team, Collection<String> players, ClientboundSetPlayerTeamPacket.Action action) {
+        return new ClientboundSetPlayerTeamPacket(team.getName(), action == ClientboundSetPlayerTeamPacket.Action.ADD ? METHOD_JOIN : METHOD_LEAVE, Optional.empty(), players);
+    }
+    // Paper end - Multiple Entries with Scoreboards
+
     private ClientboundSetPlayerTeamPacket(final RegistryFriendlyByteBuf input) {
         this.name = input.readUtf();
         this.method = input.readByte();
@@ -179,5 +185,12 @@ public class ClientboundSetPlayerTeamPacket implements Packet<ClientGamePacketLi
                 team.packOptions()
             );
         }
+
+        // Paper start
+        @Override
+        public Team.CollisionRule collisionRule() {
+            return !io.papermc.paper.configuration.GlobalConfiguration.get().collisions.enablePlayerCollisions ? PlayerTeam.CollisionRule.NEVER : this.collisionRule; // Paper - Configurable player collision
+        }
+        // Paper end
     }
 }
