From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/protocol/game/ServerboundSetCreativeModeSlotPacket.java b/net/minecraft/network/protocol/game/ServerboundSetCreativeModeSlotPacket.java
index 077d59467a74a1bc510ad801770256d70739b3f2..4e136b93826f67dc1924254aab78358a99fad605 100644
--- a/net/minecraft/network/protocol/game/ServerboundSetCreativeModeSlotPacket.java
+++ b/net/minecraft/network/protocol/game/ServerboundSetCreativeModeSlotPacket.java
@@ -11,7 +11,7 @@ public record ServerboundSetCreativeModeSlotPacket(short slotNum, ItemStack item
     public static final StreamCodec<RegistryFriendlyByteBuf, ServerboundSetCreativeModeSlotPacket> STREAM_CODEC = StreamCodec.composite(
         ByteBufCodecs.SHORT,
         ServerboundSetCreativeModeSlotPacket::slotNum,
-        ItemStack.validatedStreamCodec(ItemStack.OPTIONAL_UNTRUSTED_STREAM_CODEC),
+        ItemStack.validatedStreamCodec(ItemStack.OPTIONAL_UNTRUSTED_STREAM_CODEC).apply(ByteBufCodecs::trackDepth), // Paper - Track codec depth
         ServerboundSetCreativeModeSlotPacket::itemStack,
         ServerboundSetCreativeModeSlotPacket::new
     );
