From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/protocol/game/ServerboundUseItemOnPacket.java b/net/minecraft/network/protocol/game/ServerboundUseItemOnPacket.java
index ca761d4a7be11a7926267b0e8a759306d9184f84..ba6d0c714ea6789fc6bdda94876c2d4e38e7069a 100644
--- a/net/minecraft/network/protocol/game/ServerboundUseItemOnPacket.java
+++ b/net/minecraft/network/protocol/game/ServerboundUseItemOnPacket.java
@@ -14,6 +14,7 @@ public class ServerboundUseItemOnPacket implements Packet<ServerGamePacketListen
     private final BlockHitResult blockHit;
     private final InteractionHand hand;
     private final int sequence;
+    public long timestamp; // Spigot
 
     public ServerboundUseItemOnPacket(final InteractionHand hand, final BlockHitResult blockHit, final int sequence) {
         this.hand = hand;
@@ -22,6 +23,7 @@ public class ServerboundUseItemOnPacket implements Packet<ServerGamePacketListen
     }
 
     private ServerboundUseItemOnPacket(final FriendlyByteBuf input) {
+        this.timestamp = System.currentTimeMillis(); // Spigot
         this.hand = input.readEnum(InteractionHand.class);
         this.blockHit = input.readBlockHitResult();
         this.sequence = input.readVarInt();
