From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/network/protocol/game/ServerboundUseItemPacket.java b/net/minecraft/network/protocol/game/ServerboundUseItemPacket.java
index ed7e7602461af4ee10a3a8de185c54d611f8e400..5d4a082ba8be10c3253352306def2ec21bcf49f0 100644
--- a/net/minecraft/network/protocol/game/ServerboundUseItemPacket.java
+++ b/net/minecraft/network/protocol/game/ServerboundUseItemPacket.java
@@ -14,6 +14,7 @@ public class ServerboundUseItemPacket implements Packet<ServerGamePacketListener
     private final int sequence;
     private final float yRot;
     private final float xRot;
+    public long timestamp; // Spigot
 
     public ServerboundUseItemPacket(final InteractionHand hand, final int sequence, final float yRot, final float xRot) {
         this.hand = hand;
@@ -23,6 +24,7 @@ public class ServerboundUseItemPacket implements Packet<ServerGamePacketListener
     }
 
     private ServerboundUseItemPacket(final FriendlyByteBuf input) {
+        this.timestamp = System.currentTimeMillis(); // Spigot
         this.hand = input.readEnum(InteractionHand.class);
         this.sequence = input.readVarInt();
         this.yRot = input.readFloat();
