From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/util/filefix/fixes/LegacyStructureFileFix.java b/net/minecraft/util/filefix/fixes/LegacyStructureFileFix.java
index 3957b5d1a2ace9ecf75752612cfb877b64f3ebcf..5f321102cd8bb4edfffa838753cd2a43bab38ce8 100644
--- a/net/minecraft/util/filefix/fixes/LegacyStructureFileFix.java
+++ b/net/minecraft/util/filefix/fixes/LegacyStructureFileFix.java
@@ -147,7 +147,7 @@ public class LegacyStructureFileFix extends FileFix {
             }
 
             Optional<Identifier> generatorIdentifier = Optional.ofNullable(Identifier.tryParse(chunkGeneratorType));
-            CompoundTag dataFixContext = ChunkMap.getChunkDataFixContextTag(dimensionKey, generatorIdentifier);
+            CompoundTag dataFixContext = ChunkMap.getChunkDataFixContextTag(dimensionKey, generatorIdentifier, ResourceKey.create(Registries.LEVEL_STEM, dimensionKey.identifier())); // Paper - pass level stem
             storeLegacyStructureDataToChunks(dimensionFixEntry.structures, chunkNbt, dataFixContext, upgradeProgress);
         }
     }
