From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/commands/BanPlayerCommands.java b/net/minecraft/server/commands/BanPlayerCommands.java
index 86deb4fdfd7b5378a82ae7f5e1a749db63468231..c87bc86c57fde9d92e3927bdead5cdaf2e935744 100644
--- a/net/minecraft/server/commands/BanPlayerCommands.java
+++ b/net/minecraft/server/commands/BanPlayerCommands.java
@@ -47,7 +47,7 @@ public class BanPlayerCommands {
                 source.sendSuccess(() -> Component.translatable("commands.ban.success", Component.literal(player.name()), entry.getReasonMessage()), true);
                 ServerPlayer online = source.getServer().getPlayerList().getPlayer(player.id());
                 if (online != null) {
-                    online.connection.disconnect(Component.translatable("multiplayer.disconnect.banned"));
+                    online.connection.disconnect(Component.translatable("multiplayer.disconnect.banned"), org.bukkit.event.player.PlayerKickEvent.Cause.BANNED); // Paper - kick event cause
                 }
             }
         }
