From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/commands/DebugCommand.java b/net/minecraft/server/commands/DebugCommand.java
index 1e64ac2bec04c3b70046c8b4a41bca73bce95faf..812659bc14a356bde510d245557f94746bc576cc 100644
--- a/net/minecraft/server/commands/DebugCommand.java
+++ b/net/minecraft/server/commands/DebugCommand.java
@@ -263,6 +263,13 @@ public class DebugCommand {
             return true;
         }
 
+        // Paper start
+        @Override
+        public org.bukkit.command.CommandSender getBukkitSender(final CommandSourceStack wrapper) {
+            throw new UnsupportedOperationException();
+        }
+        // Paper end
+
         @Override
         public void close() {
             IOUtils.closeQuietly(this.output);
