From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/commands/DifficultyCommand.java b/net/minecraft/server/commands/DifficultyCommand.java
index 3c16fa1242024d2a2f796f76924180af7451189a..d4fea441a97799d7e2904f304b2a8635bb83776a 100644
--- a/net/minecraft/server/commands/DifficultyCommand.java
+++ b/net/minecraft/server/commands/DifficultyCommand.java
@@ -31,11 +31,11 @@ public class DifficultyCommand {
 
     public static int setDifficulty(final CommandSourceStack source, final Difficulty difficulty) throws CommandSyntaxException {
         MinecraftServer server = source.getServer();
-        if (server.getWorldData().getDifficulty() == difficulty) {
+        if (source.getLevel().getDifficulty() == difficulty) { // CraftBukkit
             throw ERROR_ALREADY_SAME_DIFFICULTY.create(difficulty.getDisplayName());
         }
 
-        server.setDifficulty(difficulty, true);
+        server.setDifficulty(source.getLevel(), difficulty, source, true); // Paper - per level difficulty; don't skip other difficulty-changing logic (fix upstream's fix); WorldDifficultyChangeEvent
         source.sendSuccess(() -> Component.translatable("commands.difficulty.success", difficulty.getDisplayName()), true);
         return 0;
     }
