--- a/net/minecraft/server/commands/EffectCommands.java
+++ b/net/minecraft/server/commands/EffectCommands.java
@@ -184,7 +_,7 @@
         for (Entity entity : entities) {
             if (entity instanceof LivingEntity livingEntity) {
                 MobEffectInstance instance = new MobEffectInstance(effectHolder, duration, amplifier, false, particles);
-                tracker.track(livingEntity, livingEntity.addEffect(instance, source.getEntity()));
+                if (livingEntity.addEffect(instance, source.getEntity(), org.bukkit.event.entity.EntityPotionEffectEvent.Cause.COMMAND)) { // CraftBukkit
             }
         }
 
@@ -207,7 +_,7 @@
         CommandResponseTracker<LivingEntity> tracker = CommandResponseTracker.create();
 
         for (Entity entity : entities) {
-            if (entity instanceof LivingEntity livingEntity) {
+            if (entity instanceof LivingEntity livingEntity && livingEntity.removeEffect(effectHolder, org.bukkit.event.entity.EntityPotionEffectEvent.Cause.COMMAND)) { // CraftBukkit
                 tracker.track(livingEntity, livingEntity.removeAllEffects());
             }
         }
@@ -220,7 +_,7 @@
         CommandResponseTracker<LivingEntity> tracker = CommandResponseTracker.create();
 
         for (Entity entity : entities) {
-            if (entity instanceof LivingEntity livingEntity) {
+            if (entity instanceof LivingEntity livingEntity && livingEntity.removeAllEffects(org.bukkit.event.entity.EntityPotionEffectEvent.Cause.COMMAND)) { // CraftBukkit
                 tracker.track(livingEntity, livingEntity.removeEffect(effectHolder));
             }
         }
