--- a/net/minecraft/server/commands/GameRuleCommand.java
+++ b/net/minecraft/server/commands/GameRuleCommand.java
@@ -40,7 +_,7 @@
 
     private static <T> int setRule(final CommandContext<CommandSourceStack> context, final GameRule<T> gameRule) throws CommandSyntaxException {
         CommandSourceStack source = context.getSource();
-        T value = context.getArgument("value", gameRule.valueClass());
+        T value = org.bukkit.craftbukkit.event.CraftEventFactory.handleGameRuleSet(gameRule, context.getArgument("value", gameRule.valueClass()), source.getLevel(), source.getBukkitSender()).value(); // Paper - per-world game rules and event
         GameRules gameRules = source.getLevel().getGameRules();
         String seralizedValue = gameRule.serialize(value);
         if (gameRules.get(gameRule).equals(value)) {
