--- a/net/minecraft/server/commands/GiveCommand.java
+++ b/net/minecraft/server/commands/GiveCommand.java
@@ -75,7 +_,7 @@
                 ItemStack copyToDrop = prototypeItemStack.copyWithCount(size);
                 boolean added = player.getInventory().add(copyToDrop);
                 if (added && copyToDrop.isEmpty()) {
-                    ItemEntity drop = player.createItemStackToDrop(prototypeItemStack.copy(), false, false);
+                    ItemEntity drop = player.createItemStackToDrop(prototypeItemStack.copy(), false, false, false, null); // Paper - do not fire PlayerDropItemEvent for /give command
                     if (drop != null) {
                         drop.makeFakeItem();
                         drop.level().addFreshEntity(drop);
@@ -94,7 +_,7 @@
                         );
                     player.containerMenu.broadcastChanges();
                 } else {
-                    ItemEntity drop = player.createItemStackToDrop(copyToDrop, false, false);
+                    ItemEntity drop = player.createItemStackToDrop(copyToDrop, false, false, false, null); // Paper - do not fire PlayerDropItemEvent for /give command
                     if (drop != null) {
                         drop.setNoPickUpDelay();
                         drop.setTarget(player.getUUID());
