From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/decoration/painting/Painting.java b/net/minecraft/world/entity/decoration/painting/Painting.java
index 2090fb61a71d847dad0d8cf79a5e7dfaad4d55fa..f6882a8392998e1964205635ea0b060a7a192157 100644
--- a/net/minecraft/world/entity/decoration/painting/Painting.java
+++ b/net/minecraft/world/entity/decoration/painting/Painting.java
@@ -150,21 +150,31 @@ public class Painting extends HangingEntity {
 
     @Override
     protected AABB calculateBoundingBox(final BlockPos pos, final Direction direction) {
+        // CraftBukkit start
+        PaintingVariant variant = (PaintingVariant) this.getVariant().value();
+        return Painting.calculateBoundingBoxStatic(pos, direction, variant.width(), variant.height());
+    }
+
+    public static AABB calculateBoundingBoxStatic(final BlockPos pos, final Direction direction, final int width, final int height) {
+        // CraftBukkit end
         float shiftToBlockWall = 0.46875F;
         Vec3 attachedToWall = Vec3.atCenterOf(pos).relative(direction, -0.46875);
-        PaintingVariant variant = this.getVariant().value();
-        double horizontalOffset = this.offsetForPaintingSize(variant.width());
-        double verticalOffset = this.offsetForPaintingSize(variant.height());
+        // CraftBukkit start
+        double horizontalOffset = Painting.offsetForPaintingSize(width);
+        double verticalOffset = Painting.offsetForPaintingSize(height);
+        // CraftBukkit end
         Direction left = direction.getCounterClockWise();
         Vec3 position = attachedToWall.relative(left, horizontalOffset).relative(Direction.UP, verticalOffset);
         Direction.Axis axis = direction.getAxis();
-        double xSize = axis == Direction.Axis.X ? 0.0625 : variant.width();
-        double ySize = variant.height();
-        double zSize = axis == Direction.Axis.Z ? 0.0625 : variant.width();
+        // CraftBukkit start
+        double xSize = axis == Direction.Axis.X ? 0.0625 : width;
+        double ySize = height;
+        double zSize = axis == Direction.Axis.Z ? 0.0625 : width;
+        // CraftBukkit end
         return AABB.ofSize(position, xSize, ySize, zSize);
     }
 
-    private double offsetForPaintingSize(final int size) {
+    private static double offsetForPaintingSize(final int size) { // CraftBukkit - static
         return size % 2 == 0 ? 0.5 : 0.0;
     }
 
