From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/stats/ServerStatsCounter.java b/net/minecraft/stats/ServerStatsCounter.java
index 75f463a99f88d54043b2deae3b38c0c3c8e03f3c..f5beb001b938072e78c0ead7e0ad0499784ed1cf 100644
--- a/net/minecraft/stats/ServerStatsCounter.java
+++ b/net/minecraft/stats/ServerStatsCounter.java
@@ -75,9 +75,21 @@ public class ServerStatsCounter extends StatsCounter {
                 LOGGER.error("Couldn't parse statistics file {}", file, e);
             }
         }
+        // Paper start - Moved after stat fetching for player state file
+        // Moves the loading after vanilla loading, so it overrides the values.
+        // Disables saving any forced stats, so it stays at the same value (without enabling disableStatSaving)
+        // Fixes stat initialization to not cause a NullPointerException
+        // Spigot start
+        for (Map.Entry<net.minecraft.resources.Identifier, Integer> entry : org.spigotmc.SpigotConfig.forcedStats.entrySet()) {
+            Stat<net.minecraft.resources.Identifier> wrapper = Stats.CUSTOM.get(java.util.Objects.requireNonNull(BuiltInRegistries.CUSTOM_STAT.getValue(entry.getKey()))); // Paper - ensured by SpigotConfig#stats
+            this.stats.put(wrapper, entry.getValue().intValue());
+        }
+        // Spigot end
+        // Paper end - Moved after stat fetching for player state file
     }
 
     public void save() {
+        if (org.spigotmc.SpigotConfig.disableStatSaving) return; // Spigot
         try {
             FileUtil.createDirectoriesSafe(this.file.getParent());
 
@@ -91,6 +103,8 @@ public class ServerStatsCounter extends StatsCounter {
 
     @Override
     public void setValue(final Player player, final Stat<?> stat, final int count) {
+        if (org.spigotmc.SpigotConfig.disableStatSaving) return; // Spigot
+        if (stat.getType() == Stats.CUSTOM && stat.getValue() instanceof final net.minecraft.resources.Identifier key && org.spigotmc.SpigotConfig.forcedStats.get(key) != null) return; // Paper - disable saving forced stats
         super.setValue(player, stat, count);
         this.dirty.add(stat);
     }
