From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/stats/ServerRecipeBook.java b/net/minecraft/stats/ServerRecipeBook.java
index 2cd50af0340b54aaf6c6afbf99c29f49bf8fd81f..ac3fbaf20c3ed8ca981c9071f1bf95f2ad169f85 100644
--- a/net/minecraft/stats/ServerRecipeBook.java
+++ b/net/minecraft/stats/ServerRecipeBook.java
@@ -64,17 +64,21 @@ public class ServerRecipeBook extends RecipeBook {
         for (RecipeHolder<?> recipe : recipes) {
             ResourceKey<Recipe<?>> id = recipe.id();
             if (!this.known.contains(id) && !recipe.value().isSpecial()) {
+                // Paper start - PlayerRecipeDiscoverEvent event
+                final org.bukkit.event.player.PlayerRecipeDiscoverEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callPlayerRecipeListUpdateEvent(player, recipe);
+                if (event.isCancelled()) continue;
+                // Paper end - PlayerRecipeDiscoverEvent event
                 this.add(id);
                 this.addHighlight(id);
                 this.displayResolver
                     .displaysForRecipe(
-                        id, display -> recipesToAdd.add(new ClientboundRecipeBookAddPacket.Entry(display, recipe.value().showNotification(), true))
+                        id, display -> recipesToAdd.add(new ClientboundRecipeBookAddPacket.Entry(display, event.shouldShowNotification(), true)) // Paper - set notification from the event
                     );
                 CriteriaTriggers.RECIPE_UNLOCKED.trigger(player, recipe);
             }
         }
 
-        if (!recipesToAdd.isEmpty()) {
+        if (!recipesToAdd.isEmpty() && player.connection != null) { // SPIGOT-4478 during PlayerLoginEvent
             player.connection.send(new ClientboundRecipeBookAddPacket(recipesToAdd, false));
         }
 
@@ -92,7 +96,7 @@ public class ServerRecipeBook extends RecipeBook {
             }
         }
 
-        if (!recipesToRemove.isEmpty()) {
+        if (!recipesToRemove.isEmpty() && player.connection != null) { // SPIGOT-4478 during PlayerLoginEvent
             player.connection.send(new ClientboundRecipeBookRemovePacket(recipesToRemove));
         }
 
