From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/animal/feline/Cat.java b/net/minecraft/world/entity/animal/feline/Cat.java
index d090ba5702f29a097bd08ecae021da026b04bc25..d810202ba190091b17a5a68fd17a0d86cf6588cd 100644
--- a/net/minecraft/world/entity/animal/feline/Cat.java
+++ b/net/minecraft/world/entity/animal/feline/Cat.java
@@ -412,6 +412,11 @@ public class Cat extends TamableAnimal {
                 if (itemStack.is(ItemTags.CAT_COLLAR_DYES)) {
                     DyeColor color = itemStack.get(DataComponents.DYE);
                     if (color != null && color != this.getCollarColor()) {
+                        // Paper start - Add EntityDyeEvent and CollarColorable interface
+                        final io.papermc.paper.event.entity.EntityDyeEvent event = new io.papermc.paper.event.entity.EntityDyeEvent(this.getBukkitEntity(), org.bukkit.DyeColor.getByWoolData((byte) color.getId()), ((net.minecraft.server.level.ServerPlayer) player).getBukkitEntity());
+                        if (!event.callEvent()) return InteractionResult.FAIL;
+                        color = DyeColor.byId(event.getColor().getWoolData());
+                        // Paper end - Add EntityDyeEvent and CollarColorable interface
                         if (!this.level().isClientSide()) {
                             this.setCollarColor(color);
                             itemStack.consume(1, player);
@@ -483,7 +488,7 @@ public class Cat extends TamableAnimal {
     }
 
     private void tryToTame(final Player player) {
-        if (this.random.nextInt(3) == 0) {
+        if (this.random.nextInt(3) == 0 && !org.bukkit.craftbukkit.event.CraftEventFactory.callEntityTameEvent(this, player).isCancelled()) { // CraftBukkit
             this.tame(player);
             this.setOrderedToSit(true);
             this.level().broadcastEntityEvent(this, EntityEvent.TAMING_SUCCEEDED);
@@ -615,15 +620,20 @@ public class Cat extends TamableAnimal {
                 .dropFromGiftLootTable(
                     getServerLevel(this.cat),
                     BuiltInLootTables.CAT_MORNING_GIFT,
-                    (level, itemStack) -> level.addFreshEntity(
-                        new ItemEntity(
+                    // CraftBukkit start
+                    (level, itemStack) -> {
+                        final ItemEntity item = new ItemEntity(
                             level,
                             (double)catPos.getX() - Mth.sin(this.cat.yBodyRot * Mth.DEG_TO_RAD),
                             catPos.getY(),
                             (double)catPos.getZ() + Mth.cos(this.cat.yBodyRot * Mth.DEG_TO_RAD),
                             itemStack
-                        )
-                    )
+                        );
+                        org.bukkit.event.entity.EntityDropItemEvent event = new org.bukkit.event.entity.EntityDropItemEvent(this.cat.getBukkitEntity(), (org.bukkit.entity.Item) item.getBukkitEntity());
+                        if (!event.callEvent()) return;
+                        level.addFreshEntity(item);
+                    }
+                    // CraftBukkit end
                 );
         }
 
@@ -649,7 +659,7 @@ public class Cat extends TamableAnimal {
     }
 
     private static class CatTemptGoal extends TemptGoal {
-        private @Nullable Player selectedPlayer;
+        private @Nullable LivingEntity selectedPlayer; // CraftBukkit
         private final Cat cat;
 
         public CatTemptGoal(final Cat mob, final double speedModifier, final Predicate<ItemStack> items, final boolean canScare) {
