From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/storage/LevelStorageSource.java b/net/minecraft/world/level/storage/LevelStorageSource.java
index 7b247f89b3fd0cb8719acb27f00564611b576595..6f24eb34e0ecc2b03371c585a9538e9337a2295a 100644
--- a/net/minecraft/world/level/storage/LevelStorageSource.java
+++ b/net/minecraft/world/level/storage/LevelStorageSource.java
@@ -145,13 +145,14 @@ public class LevelStorageSource {
         final WorldDataConfiguration dataConfiguration,
         final Registry<LevelStem> datapackDimensions,
         final HolderLookup.Provider registryAccess
+        , final ResourceKey<Level> dimension // Paper - pass dimension
     ) {
         if (DataFixers.getFileFixer().requiresFileFixing(NbtUtils.getDataVersion(levelDataTag))) {
             throw new IllegalStateException("Cannot get level data without file fixing first");
         }
 
         Dynamic<?> dataTag = RegistryOps.injectRegistryContext(levelDataTag, registryAccess);
-        WorldGenSettings worldGenSettings = readExistingSavedData(worldAccess, registryAccess, WorldGenSettings.TYPE)
+        WorldGenSettings worldGenSettings = readExistingSavedData(worldAccess, dimension, registryAccess, WorldGenSettings.TYPE) // Paper - pass dimension
             .mapOrElse(
                 Function.identity(),
                 error -> {
@@ -170,9 +171,9 @@ public class LevelStorageSource {
     }
 
     public static <T extends SavedData> DataResult<T> readExistingSavedData(
-        final LevelStorageSource.LevelStorageAccess access, final HolderLookup.Provider registryAccess, final SavedDataType<T> savedDataType
+        final LevelStorageSource.LevelStorageAccess access, final ResourceKey<Level> dimension, final HolderLookup.Provider registryAccess, final SavedDataType<T> savedDataType // Paper - pass dimension
     ) {
-        Path dataLocation = savedDataType.id().withSuffix(".dat").resolveAgainst(access.getLevelPath(LevelResource.DATA));
+        Path dataLocation = savedDataType.id().withSuffix(".dat").resolveAgainst(access.getDimensionPath(dimension).resolve(LevelResource.DATA.id())); // Paper - use dimension scoped data (we 'demote' some data from global and add our own dimension scoped data)
 
         CompoundTag fileContents;
         try {
@@ -393,7 +394,7 @@ public class LevelStorageSource {
 
     public LevelStorageSource.LevelStorageAccess validateAndCreateAccess(final String levelId) throws IOException, ContentValidationException {
         Path levelPath = this.getLevelPath(levelId);
-        List<ForbiddenSymlinkInfo> validationResults = this.worldDirValidator.validateDirectory(levelPath, true);
+        List<ForbiddenSymlinkInfo> validationResults = Boolean.getBoolean("paper.disableWorldSymlinkValidation") ? List.of() : this.worldDirValidator.validateDirectory(levelPath, true); // Paper - add skipping of symlinks scan
         if (!validationResults.isEmpty()) {
             throw new ContentValidationException(levelPath, validationResults);
         } else {
