From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/storage/LevelData.java b/net/minecraft/world/level/storage/LevelData.java
index a51aa51ff6b039aa280c6eea3b50df7e549f2bcd..7f46f4529de72f2a0e3b38c4a8a401d3fee93829 100644
--- a/net/minecraft/world/level/storage/LevelData.java
+++ b/net/minecraft/world/level/storage/LevelData.java
@@ -62,5 +62,21 @@ public interface LevelData {
         public BlockPos pos() {
             return this.globalPos.pos();
         }
+
+        // Paper start
+        /**
+         * Equals without checking dimension.
+         *
+         * @param other other object
+         * @return true if position and rotation are equal
+         */
+        public boolean positionEquals(Object other) {
+            if (other == this) return true;
+            if (!(other instanceof LevelData.RespawnData otherRespawn)) return false;
+            return this.pos().equals(otherRespawn.pos())
+                && this.yaw == otherRespawn.yaw
+                && this.pitch == otherRespawn.pitch;
+        }
+        // Paper end
     }
 }
