From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/storage/PlayerDataStorage.java b/net/minecraft/world/level/storage/PlayerDataStorage.java
index aa3160786b502ede052ec62e82785f129d3b6691..8eda63e1f07f7998875713447c5f39eea3f85ca4 100644
--- a/net/minecraft/world/level/storage/PlayerDataStorage.java
+++ b/net/minecraft/world/level/storage/PlayerDataStorage.java
@@ -31,6 +31,7 @@ public class PlayerDataStorage {
     }
 
     public void save(final Player player) {
+        if (org.spigotmc.SpigotConfig.disablePlayerDataSaving) return; // Spigot
         try (ProblemReporter.ScopedCollector reporter = new ProblemReporter.ScopedCollector(player.problemPath(), LOGGER)) {
             TagValueOutput output = TagValueOutput.createWithContext(reporter, player.registryAccess());
             player.saveWithoutId(output);
@@ -42,7 +43,7 @@ public class PlayerDataStorage {
             Path oldFile = playerDirPath.resolve(player.getStringUUID() + ".dat_old");
             Util.safeReplaceFile(realFile, tmpFile, oldFile);
         } catch (Exception ignored) {
-            LOGGER.warn("Failed to save player data for {}", player.getPlainTextName());
+            LOGGER.warn("Failed to save player data for {}", player.getPlainTextName(), ignored); // Paper - Print exception
         }
     }
 
@@ -62,9 +63,25 @@ public class PlayerDataStorage {
 
     private Optional<CompoundTag> load(final NameAndId nameAndId, final String suffix) {
         File realFile = new File(this.playerDir, nameAndId.id() + suffix);
+        // Spigot start
+        boolean usingWrongFile = false;
+        if (org.bukkit.Bukkit.getOnlineMode() && !realFile.exists()) { // Paper - Check online mode first
+            realFile = new File(this.playerDir, net.minecraft.core.UUIDUtil.createOfflinePlayerUUID(nameAndId.name()) + suffix);
+            if (realFile.exists()) {
+                usingWrongFile = true;
+                LOGGER.warn("Using offline mode UUID file for player {} as it is the only copy we can find.", nameAndId.name());
+            }
+        }
+        // Spigot end
         if (realFile.exists() && realFile.isFile()) {
             try {
-                return Optional.of(NbtIo.readCompressed(realFile.toPath(), NbtAccounter.unlimitedHeap()));
+                // Spigot start
+                Optional<CompoundTag> optional = Optional.of(NbtIo.readCompressed(realFile.toPath(), NbtAccounter.unlimitedHeap()));
+                if (usingWrongFile) {
+                    realFile.renameTo(new File(realFile.getPath() + ".offline-read"));
+                }
+                return optional;
+                // Spigot end
             } catch (Exception ignored) {
                 LOGGER.warn("Failed to load player data for {}", nameAndId.name());
             }
@@ -84,4 +101,10 @@ public class PlayerDataStorage {
             return DataFixTypes.PLAYER.updateToCurrentVersion(this.fixerUpper, tag, version);
         });
     }
+
+    // CraftBukkit start
+    public File getPlayerDir() {
+        return this.playerDir;
+    }
+    // CraftBukkit end
 }
