From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/storage/TagValueOutput.java b/net/minecraft/world/level/storage/TagValueOutput.java
index c7dc0bc2f727c2c9e977bde8035c6ad1e54a479a..007bd7addfe854f28fcf53cb638f51b7601b1c92 100644
--- a/net/minecraft/world/level/storage/TagValueOutput.java
+++ b/net/minecraft/world/level/storage/TagValueOutput.java
@@ -24,6 +24,23 @@ public class TagValueOutput implements ValueOutput {
         this.output = output;
     }
 
+    // Paper start - utility methods
+    public static TagValueOutput createWrappingGlobal(
+        final ProblemReporter problemReporter,
+        final CompoundTag output
+    ) {
+        return new TagValueOutput(problemReporter, NbtOps.INSTANCE, output);
+    }
+
+    public static TagValueOutput createWrappingWithContext(
+        final ProblemReporter problemReporter,
+        final HolderLookup.Provider lookup,
+        final CompoundTag output
+    ) {
+        return new TagValueOutput(problemReporter, lookup.createSerializationContext(NbtOps.INSTANCE), output);
+    }
+    // Paper end - utility methods
+
     public static TagValueOutput createWithContext(final ProblemReporter problemReporter, final HolderLookup.Provider provider) {
         return new TagValueOutput(problemReporter, provider.createSerializationContext(NbtOps.INSTANCE), new CompoundTag());
     }
