From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/monster/illager/Pillager.java b/net/minecraft/world/entity/monster/illager/Pillager.java
index 13083622766683a4d9e2d6d9b62398e3f74c2fcb..2fb3b7f3928a459d089bd8680673aa589118f464 100644
--- a/net/minecraft/world/entity/monster/illager/Pillager.java
+++ b/net/minecraft/world/entity/monster/illager/Pillager.java
@@ -215,7 +215,7 @@ public class Pillager extends AbstractIllager implements CrossbowAttackMob, Inve
             this.onItemPickup(entity);
             ItemStack remainder = this.inventory.addItem(itemStack);
             if (remainder.isEmpty()) {
-                entity.discard();
+                entity.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.PICKUP); // CraftBukkit - add Bukkit remove cause
             } else {
                 itemStack.setCount(remainder.getCount());
             }
