From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/monster/illager/SpellcasterIllager.java b/net/minecraft/world/entity/monster/illager/SpellcasterIllager.java
index ea01ffebd0f237b2563b8659969e0960e5fb7f41..5fcd7c0d18352ef6ff574a1e3df20cb1d9cc0a4f 100644
--- a/net/minecraft/world/entity/monster/illager/SpellcasterIllager.java
+++ b/net/minecraft/world/entity/monster/illager/SpellcasterIllager.java
@@ -208,6 +208,11 @@ public abstract class SpellcasterIllager extends AbstractIllager {
         public void tick() {
             this.attackWarmupDelay--;
             if (this.attackWarmupDelay == 0) {
+                // CraftBukkit start
+                if (!org.bukkit.craftbukkit.event.CraftEventFactory.handleEntitySpellCastEvent(SpellcasterIllager.this, this.getSpell())) {
+                    return;
+                }
+                // CraftBukkit end
                 this.performSpellCasting();
                 SpellcasterIllager.this.playSound(SpellcasterIllager.this.getCastingSoundEvent(), 1.0F, 1.0F);
             }
