From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/monster/breeze/Breeze.java b/net/minecraft/world/entity/monster/breeze/Breeze.java
index b17bdccd984c0a58a06e0b9271e480531f81a1fb..eaa7eb69bdc819228bbab05b053844f1578294ba 100644
--- a/net/minecraft/world/entity/monster/breeze/Breeze.java
+++ b/net/minecraft/world/entity/monster/breeze/Breeze.java
@@ -243,6 +243,11 @@ public class Breeze extends Monster {
 
     @Override
     public boolean canAttack(final LivingEntity target) {
+        // TODO - snapshot - reimplement but without reintroducing MC-199589
+        // CraftBukkit start - SPIGOT-7957: Allow attack if target from brain was set
+        // LivingEntity targetFromBrain = this.getBrain().getMemory(MemoryModuleType.ATTACK_TARGET).orElse(null);
+        // if (targetFromBrain != null) return target.is(targetFromBrain);
+        // CraftBukkit end
         return (target.is(EntityTypes.PLAYER) || target.is(EntityTypes.IRON_GOLEM)) && super.canAttack(target);
     }
 
