From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/vehicle/ContainerEntity.java b/net/minecraft/world/entity/vehicle/ContainerEntity.java
index 294bfe7bb651354bc4f9623ef84edc67037dfd1e..c80a44dbc54e0bf00daa525bef20422b15ce8a85 100644
--- a/net/minecraft/world/entity/vehicle/ContainerEntity.java
+++ b/net/minecraft/world/entity/vehicle/ContainerEntity.java
@@ -58,12 +58,12 @@ public interface ContainerEntity extends Container, MenuProvider {
     default void addChestVehicleSaveData(final ValueOutput output) {
         if (this.getContainerLootTable() != null) {
             output.putString("LootTable", this.getContainerLootTable().identifier().toString());
+            this.lootableData().saveNbt(output); // Paper
             if (this.getContainerLootTableSeed() != 0L) {
                 output.putLong("LootTableSeed", this.getContainerLootTableSeed());
             }
-        } else {
-            ContainerHelper.saveAllItems(output, this.getItemStacks());
         }
+        ContainerHelper.saveAllItems(output, this.getItemStacks()); // Paper - always save the items, table may still remain
     }
 
     default void readChestVehicleSaveData(final ValueInput input) {
@@ -71,7 +71,12 @@ public interface ContainerEntity extends Container, MenuProvider {
         ResourceKey<LootTable> lootTable = input.read("LootTable", LootTable.KEY_CODEC).orElse(null);
         this.setContainerLootTable(lootTable);
         this.setContainerLootTableSeed(input.getLongOr("LootTableSeed", 0L));
-        if (lootTable == null) {
+        // Paper start - LootTable API
+        if (this.getContainerLootTable() != null) {
+            this.lootableData().loadNbt(input);
+        }
+        // Paper end - LootTable API
+        if (true || lootTable == null) { // Paper - always read the items, table may still remain
             ContainerHelper.loadAllItems(input, this.getItemStacks());
         }
     }
@@ -86,19 +91,27 @@ public interface ContainerEntity extends Container, MenuProvider {
     }
 
     default InteractionResult interactWithContainerVehicle(final Player player) {
-        player.openMenu(this);
+        // Paper start - Fix InventoryOpenEvent cancellation
+        if (player.openMenu(this).isEmpty()) {
+            return InteractionResult.PASS;
+        }
+        // Paper end - Fix InventoryOpenEvent cancellation
         return InteractionResult.SUCCESS;
     }
 
     default void unpackChestVehicleLootTable(final @Nullable Player player) {
         MinecraftServer server = this.level().getServer();
-        if (this.getContainerLootTable() != null && server != null) {
+        if (server != null && this.lootableData().shouldReplenish(this, com.destroystokyo.paper.loottable.PaperLootableInventoryData.ENTITY, player)) { // Paper - LootTable API
             LootTable lootTable = server.reloadableRegistries().getLootTable(this.getContainerLootTable());
             if (player != null) {
                 CriteriaTriggers.GENERATE_LOOT.trigger((ServerPlayer)player, this.getContainerLootTable());
             }
 
-            this.setContainerLootTable(null);
+            // Paper start - LootTable API
+            if (this.lootableData().shouldClearLootTable(this, com.destroystokyo.paper.loottable.PaperLootableInventoryData.ENTITY, player)) {
+                this.setContainerLootTable(null);
+            }
+            // Paper end - LootTable API
             LootParams.Builder builder = new LootParams.Builder((ServerLevel)this.level()).withParameter(LootContextParams.ORIGIN, this.position());
             if (player != null) {
                 builder.withLuck(player.getLuck()).withParameter(LootContextParams.THIS_ENTITY, player);
@@ -168,4 +181,14 @@ public interface ContainerEntity extends Container, MenuProvider {
     default boolean isChestVehicleStillValid(final Player player) {
         return !this.isRemoved() && player.isWithinEntityInteractionRange(this.getBoundingBox(), 4.0);
     }
+
+    // Paper start - LootTable API
+    default com.destroystokyo.paper.loottable.PaperLootableInventoryData lootableData() {
+        throw new UnsupportedOperationException("Implement this method");
+    }
+
+    default com.destroystokyo.paper.loottable.PaperLootableInventory getLootableInventory() {
+        return ((com.destroystokyo.paper.loottable.PaperLootableInventory) ((Entity) this).getBukkitEntity());
+    }
+    // Paper end - LootTable API
 }
