From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/rcon/thread/RconClient.java b/net/minecraft/server/rcon/thread/RconClient.java
index 48916bf68a48020f709059d091c8d13fa9654246..2995d915a7cb433cc72084c341351ebc390c4f2d 100644
--- a/net/minecraft/server/rcon/thread/RconClient.java
+++ b/net/minecraft/server/rcon/thread/RconClient.java
@@ -23,11 +23,14 @@ public class RconClient extends GenericThread {
     private final Socket client;
     private final byte[] buf = new byte[1460];
     private final String rconPassword;
-    private final ServerInterface serverInterface;
+    // CraftBukkit start
+    private final net.minecraft.server.dedicated.DedicatedServer serverInterface;
+    private final net.minecraft.server.rcon.RconConsoleSource rconConsoleSource;
+    // CraftBukkit end
 
     public RconClient(final ServerInterface serverInterface, final String rconPassword, final Socket socket) {
         super("RCON Client " + socket.getInetAddress());
-        this.serverInterface = serverInterface;
+        this.serverInterface = (net.minecraft.server.dedicated.DedicatedServer) serverInterface; // CraftBukkit
         this.client = socket;
 
         try {
@@ -37,6 +40,7 @@ public class RconClient extends GenericThread {
         }
 
         this.rconPassword = rconPassword;
+        this.rconConsoleSource = new net.minecraft.server.rcon.RconConsoleSource(this.serverInterface, client.getRemoteSocketAddress()); // CraftBukkit
     }
 
     @Override
@@ -67,7 +71,7 @@ public class RconClient extends GenericThread {
                                 String command = PktUtils.stringFromByteArray(this.buf, offset, read);
 
                                 try {
-                                    this.sendCmdResponse(requestid, this.serverInterface.runCommand(command));
+                                    this.sendCmdResponse(requestid, this.serverInterface.runCommand(this.rconConsoleSource, command)); // CraftBukkit
                                 } catch (Exception e) {
                                     this.sendCmdResponse(requestid, "Error executing: " + command + " (" + e.getMessage() + ")");
                                 }
