From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/rcon/thread/RconThread.java b/net/minecraft/server/rcon/thread/RconThread.java
index f029c0a8999318f614243c085262b10b06b8cf2f..4bcd075ff359a4b11cd4ff362ae770585ad64cfb 100644
--- a/net/minecraft/server/rcon/thread/RconThread.java
+++ b/net/minecraft/server/rcon/thread/RconThread.java
@@ -56,7 +56,7 @@ public class RconThread extends GenericThread {
 
     public static @Nullable RconThread create(final ServerInterface serverInterface) {
         DedicatedServerProperties settings = serverInterface.getProperties();
-        String serverIp = serverInterface.getServerIp();
+        String serverIp = settings.rconIp; // Paper - Configurable rcon ip
         if (serverIp.isEmpty()) {
             serverIp = "0.0.0.0";
         }
@@ -103,6 +103,14 @@ public class RconThread extends GenericThread {
 
         this.clients.clear();
     }
+    // Paper start - don't wait for remote connections
+    public void stopNonBlocking() {
+        this.running = false;
+        for (RconClient client : this.clients) {
+            client.running = false;
+        }
+    }
+    // Paper end - don't wait for remote connections
 
     private void closeSocket(final ServerSocket socket) {
         LOGGER.debug("closeSocket: {}", socket);
