From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/monster/Monster.java b/net/minecraft/world/entity/monster/Monster.java
index 5fa8c0bf4f62c7848473fba8b913781a3c0591eb..28bab5bc11d98d8836b33d06524549e6e3d15acf 100644
--- a/net/minecraft/world/entity/monster/Monster.java
+++ b/net/minecraft/world/entity/monster/Monster.java
@@ -88,7 +88,7 @@ public abstract class Monster extends PathfinderMob implements Enemy {
         }
 
         DimensionType dimensionType = level.dimensionType();
-        int blockLightLimit = dimensionType.monsterSpawnBlockLightLimit();
+        int blockLightLimit = level.getLevel().paperConfig().entities.spawning.monsterSpawnMaxLightLevel.or(dimensionType.monsterSpawnBlockLightLimit()); // Paper - Configurable max block light for monster spawning
         if (blockLightLimit < 15 && level.getBrightness(LightLayer.BLOCK, pos) > blockLightLimit) {
             return false;
         }
