From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/monster/Ravager.java b/net/minecraft/world/entity/monster/Ravager.java
index a90e5f6a165247082ee4306dc904cbe0874107d4..241f80500a5f3942c9d96d74fb37117906b1c3a3 100644
--- a/net/minecraft/world/entity/monster/Ravager.java
+++ b/net/minecraft/world/entity/monster/Ravager.java
@@ -154,12 +154,19 @@ public class Ravager extends Raider {
                     BlockState state = serverLevel.getBlockState(pos);
                     Block block = state.getBlock();
                     if (block instanceof LeavesBlock) {
+                        // CraftBukkit start
+                        if (!org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(this, pos, state.getFluidState().createLegacyBlock())) { // Paper - fix wrong block state
+                            continue;
+                        }
+                        // CraftBukkit end
                         destroyedBlock = serverLevel.destroyBlock(pos, true, this) || destroyedBlock;
                     }
                 }
 
                 if (!destroyedBlock && this.onGround()) {
+                    if (new com.destroystokyo.paper.event.entity.EntityJumpEvent(getBukkitLivingEntity()).callEvent()) { // Paper - Entity Jump API
                     this.jumpFromGround();
+                    } else { this.setJumping(false); } // Paper - Entity Jump API; setJumping(false) stops a potential loop
                 }
             }
 
@@ -250,7 +257,7 @@ public class Ravager extends Raider {
         double xd = entity.getX() - this.getX();
         double zd = entity.getZ() - this.getZ();
         double dd = Math.max(xd * xd + zd * zd, 0.001);
-        entity.push(xd / dd * 4.0, 0.2, zd / dd * 4.0);
+        entity.push(xd / dd * 4.0, 0.2, zd / dd * 4.0, this); // Paper - Add EntityKnockbackByEntityEvent and EntityPushedByEntityAttackEvent
     }
 
     @Override
