--- a/net/minecraft/world/level/levelgen/DensityFunction.java
+++ b/net/minecraft/world/level/levelgen/DensityFunction.java
@@ -1,4 +_,4 @@
-package net.minecraft.world.level.levelgen;
+tpackage net.minecraft.world.level.levelgen;
 
 import com.mojang.serialization.Codec;
 import com.mojang.serialization.MapCodec;
@@ -17,15 +_,15 @@
 
 public interface DensityFunction {
     Codec<Holder<DensityFunction>> REFERENCE_CODEC = RegistryCodecs.holder(Registries.DENSITY_FUNCTION);
-    Codec<DensityFunction> CODEC = RegistryCodecs.holder(Registries.DENSITY_FUNCTION, DensityFunctions.DIRECT_CODEC).xmap(holder -> {
+    Codec<DensityFunction> CODEC = RegistryCodecs.holder(Registries.DENSITY_FUNCTION, net.minecraft.world.level.levelgen.DensityFunctions.DIRECT_CODEC).xmap(holder -> {
         return switch (holder) {
             case Holder.Direct<DensityFunction> direct -> (DensityFunction)direct.value();
-            case Holder.Reference<DensityFunction> reference -> new DensityFunctions.HolderHolder(reference);
+            case Holder.Reference<DensityFunction> reference -> new net.minecraft.world.level.levelgen.DensityFunctions.HolderHolder(reference);
             default -> throw new MatchException(null, null);
         };
     }, value -> {
         return switch (value) {
-            case DensityFunctions.HolderHolder(Holder<DensityFunction> function) -> function;
+            case net.minecraft.world.level.levelgen.DensityFunctions.HolderHolder(Holder<DensityFunction> function) -> function;
             default -> Holder.direct(value);
         };
     });
@@ -72,87 +_,87 @@
     MapCodec<? extends DensityFunction> codec();
 
     default DensityFunction clamp(final float min, final float max) {
-        return DensityFunctions.clamp(this, min, max);
+        return net.minecraft.world.level.levelgen.DensityFunctions.clamp(this, min, max);
     }
 
     default DensityFunction abs() {
-        return DensityFunctions.abs(this);
+        return net.minecraft.world.level.levelgen.DensityFunctions.abs(this);
     }
 
     default DensityFunction square() {
-        return DensityFunctions.square(this);
+        return net.minecraft.world.level.levelgen.DensityFunctions.square(this);
     }
 
     default DensityFunction cube() {
-        return DensityFunctions.cube(this);
+        return net.minecraft.world.level.levelgen.DensityFunctions.cube(this);
     }
 
     default DensityFunction sqrt() {
-        return DensityFunctions.sqrt(this);
+        return net.minecraft.world.level.levelgen.DensityFunctions.sqrt(this);
     }
 
     default DensityFunction halfNegative() {
-        return DensityFunctions.halfNegative(this);
+        return net.minecraft.world.level.levelgen.DensityFunctions.halfNegative(this);
     }
 
     default DensityFunction quarterNegative() {
-        return DensityFunctions.quarterNegative(this);
+        return net.minecraft.world.level.levelgen.DensityFunctions.quarterNegative(this);
     }
 
     default DensityFunction reciprocal() {
-        return DensityFunctions.reciprocal(this);
+        return net.minecraft.world.level.levelgen.DensityFunctions.reciprocal(this);
     }
 
     default DensityFunction negate() {
-        return DensityFunctions.negate(this);
+        return net.minecraft.world.level.levelgen.DensityFunctions.negate(this);
     }
 
     default DensityFunction squeeze() {
-        return DensityFunctions.squeeze(this);
+        return net.minecraft.world.level.levelgen.DensityFunctions.squeeze(this);
     }
 
     default DensityFunction log() {
-        return DensityFunctions.log(this);
+        return net.minecraft.world.level.levelgen.DensityFunctions.log(this);
     }
 
     default DensityFunction sign() {
-        return DensityFunctions.sign(this);
+        return net.minecraft.world.level.levelgen.DensityFunctions.sign(this);
     }
 
     default DensityFunction add(final DensityFunction right) {
-        return DensityFunctions.add(this, right);
+        return net.minecraft.world.level.levelgen.DensityFunctions.add(this, right);
     }
 
     default DensityFunction add(final float right) {
-        return DensityFunctions.add(this, DensityFunctions.constant(right));
+        return net.minecraft.world.level.levelgen.DensityFunctions.add(this, net.minecraft.world.level.levelgen.DensityFunctions.constant(right));
     }
 
     default DensityFunction sub(final DensityFunction right) {
-        return DensityFunctions.sub(this, right);
+        return net.minecraft.world.level.levelgen.DensityFunctions.sub(this, right);
     }
 
     default DensityFunction sub(final float right) {
-        return DensityFunctions.sub(this, DensityFunctions.constant(right));
+        return net.minecraft.world.level.levelgen.DensityFunctions.sub(this, net.minecraft.world.level.levelgen.DensityFunctions.constant(right));
     }
 
     default DensityFunction mul(final DensityFunction right) {
-        return DensityFunctions.mul(this, right);
+        return net.minecraft.world.level.levelgen.DensityFunctions.mul(this, right);
     }
 
     default DensityFunction mul(final float right) {
-        return DensityFunctions.mul(this, DensityFunctions.constant(right));
+        return net.minecraft.world.level.levelgen.DensityFunctions.mul(this, net.minecraft.world.level.levelgen.DensityFunctions.constant(right));
     }
 
     default DensityFunction div(final DensityFunction right) {
-        return DensityFunctions.div(this, right);
+        return net.minecraft.world.level.levelgen.DensityFunctions.div(this, right);
     }
 
     default DensityFunction div(final float right) {
-        return DensityFunctions.div(this, DensityFunctions.constant(right));
+        return net.minecraft.world.level.levelgen.DensityFunctions.div(this, net.minecraft.world.level.levelgen.DensityFunctions.constant(right));
     }
 
     default DensityFunction pow(final DensityFunction exponent) {
-        return DensityFunctions.pow(this, exponent);
+        return net.minecraft.world.level.levelgen.DensityFunctions.pow(this, exponent);
     }
 
     default DensityFunction pow(final float exponent) {
@@ -161,7 +_,7 @@
         } else if (exponent == 2.0F) {
             return this.square();
         } else {
-            return exponent == 3.0F ? this.cube() : DensityFunctions.pow(this, DensityFunctions.constant(exponent));
+            return exponent == 3.0F ? this.cube() : net.minecraft.world.level.levelgen.DensityFunctions.pow(this, net.minecraft.world.level.levelgen.DensityFunctions.constant(exponent));
         }
     }
 
