--- a/net/minecraft/world/level/levelgen/WorldGenerationContext.java
+++ b/net/minecraft/world/level/levelgen/WorldGenerationContext.java
@@ -8,6 +_,8 @@
 public class WorldGenerationContext {
     private final int minY;
     private final int height;
+    // Paper start - Flat bedrock generator settings
+    private final net.minecraft.world.level.@org.jspecify.annotations.Nullable Level level;
     private final int seaLevel;
 
     public static WorldGenerationContext of(final LevelAccessor level) {
@@ -17,12 +_,18 @@
     }
 
     public WorldGenerationContext(final ChunkGenerator generator, final LevelHeightAccessor heightAccessor) {
+        this(generator, heightAccessor, null);
+    }
+    public WorldGenerationContext(final ChunkGenerator generator, final LevelHeightAccessor heightAccessor, final net.minecraft.world.level.@org.jspecify.annotations.Nullable Level level) {
+        this.level = level;
+        // Paper end - Flat bedrock generator settings
         this.minY = Math.max(heightAccessor.getMinY(), generator.getMinY());
         this.height = Math.min(heightAccessor.getHeight(), generator.getGenDepth());
         this.seaLevel = generator.getSeaLevel();
     }
 
     private WorldGenerationContext(final LevelAccessor level) {
+        this.level = null;
         this.minY = level.getMinY();
         this.height = level.getHeight();
         this.seaLevel = level.getSeaLevel();
@@ -39,4 +_,17 @@
     public int seaLevel() {
         return this.seaLevel;
     }
+
+    // Paper start - Flat bedrock generator settings
+    public net.minecraft.world.level.Level level() {
+        if (this.level == null) {
+            throw new NullPointerException("WorldGenerationContext was initialized without a Level, but WorldGenerationContext#level was called");
+        }
+        return this.level;
+    }
+
+    public net.minecraft.world.level.@org.jspecify.annotations.Nullable Level levelOrNull() {
+        return this.level;
+    }
+    // Paper end - Flat bedrock generator settings
 }
