From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/level/levelgen/WorldgenRandom.java b/net/minecraft/world/level/levelgen/WorldgenRandom.java
index cbdaf188c5fe8450e2b7cb493308fe5829e9c8a9..bf644302ad190b2e491a6137e03dafd0c4244470 100644
--- a/net/minecraft/world/level/levelgen/WorldgenRandom.java
+++ b/net/minecraft/world/level/levelgen/WorldgenRandom.java
@@ -64,7 +64,7 @@ public class WorldgenRandom extends LegacyRandomSource {
     }
 
     public void setLargeFeatureWithSalt(final long seed, final int x, final int z, final int blend) {
-        long result = x * 341873128712L + z * 132897987541L + seed + blend;
+        long result = x * 341873128712L + z * 132897987541L + seed + blend; // Paper - diff on change for CustomChunkGenerator
         this.setSeed(result);
     }
 
