From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/util/thread/BlockableEventLoop.java b/net/minecraft/util/thread/BlockableEventLoop.java
index 46ee892e67d994c7da3583ece06776954d965e94..3dea2dae8c9eb3ed021068162a726aa1e432369c 100644
--- a/net/minecraft/util/thread/BlockableEventLoop.java
+++ b/net/minecraft/util/thread/BlockableEventLoop.java
@@ -38,6 +38,45 @@ public abstract class BlockableEventLoop<R extends Runnable> implements Executor
         MetricsRegistry.INSTANCE.add(this);
     }
 
+    // Paper start
+    public final boolean executeAllRecentInternalTasks() {
+        final int pending = this.pendingRunnables.size();
+
+        // note: due to possible recursive execution, we may execute more tasks than we want to
+
+        int ran = 0;
+
+        for (int i = 0; i < pending; ++i) {
+            final R run = this.pendingRunnables.poll();
+            if (run == null) {
+                // recursion
+                break;
+            }
+            this.doRunTask(run);
+            ++ran;
+        }
+
+        return ran > 0;
+    }
+
+    public boolean runTaskIf(final java.util.function.Predicate<? super R> predicate) {
+        if (!this.isSameThread()) {
+            throw new IllegalStateException();
+        }
+
+        final R run = this.pendingRunnables.peek();
+        if (run == null || (predicate != null && !predicate.test(run))) {
+            return false;
+        }
+
+        this.pendingRunnables.remove();
+
+        this.doRunTask(run);
+
+        return true;
+    }
+    // Paper end
+
     protected abstract boolean shouldRun(final R task);
 
     public boolean isSameThread() {
@@ -88,6 +127,14 @@ public abstract class BlockableEventLoop<R extends Runnable> implements Executor
         }
     }
 
+    // Paper start
+    public void scheduleOnMain(Runnable runnable) {
+        // postToMainThread does not work the same as older versions of mc
+        // This method is actually used to create a TickTask, which can then be posted onto main
+        this.schedule(this.wrapRunnable(runnable));
+    }
+    // Paper end
+
     @Override
     public void schedule(final R r) {
         this.pendingRunnables.add(r);
