From dcaabe6525c573bbd390ff8f3f593c5e0ad6de87 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/world/entity/monster/skeleton/Bogged.java b/net/minecraft/world/entity/monster/skeleton/Bogged.java
index f8fd6f1e6ca4ebacd6a7b61653cb14b446e7ebac..faff25abf54e00f38b10bdb47e37ab2d100f32fb 100644
--- a/net/minecraft/world/entity/monster/skeleton/Bogged.java
+++ b/net/minecraft/world/entity/monster/skeleton/Bogged.java
@@ -72,7 +72,15 @@ public class Bogged extends AbstractSkeleton implements Shearable {
         ItemStack itemStack = player.getItemInHand(hand);
         if (itemStack.is(Items.SHEARS) && this.readyForShearing()) {
             if (this.level() instanceof ServerLevel level) {
-                this.shear(level, SoundSource.PLAYERS, itemStack);
+                // Paper start - call PlayerShearEntityEvent
+                java.util.List<ItemStack> drops = this.generateDefaultDrops(level, itemStack);
+                org.bukkit.event.player.PlayerShearEntityEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.handlePlayerShearEntityEvent(player, this, itemStack, hand, drops);
+                if (event.isCancelled()) {
+                    return InteractionResult.PASS;
+                }
+                drops = org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(event.getDrops());
+                this.shear(level, SoundSource.PLAYERS, itemStack, drops);
+                // Paper end - call PlayerShearEntityEvent
                 this.gameEvent(GameEvent.SHEAR, player);
                 itemStack.hurtAndBreak(1, player, hand.asEquipmentSlot());
             }
@@ -125,13 +133,31 @@ public class Bogged extends AbstractSkeleton implements Shearable {
 
     @Override
     public void shear(final ServerLevel level, final SoundSource soundSource, final ItemStack tool) {
+        // Paper start - custom shear drops
+        this.shear(level, soundSource, tool, this.generateDefaultDrops(level, tool));
+    }
+
+    @Override
+    public java.util.List<ItemStack> generateDefaultDrops(final ServerLevel level, final ItemStack tool) {
+        final java.util.List<ItemStack> drops = new it.unimi.dsi.fastutil.objects.ObjectArrayList<>();
+        this.dropFromShearingLootTable(level, BuiltInLootTables.BOGGED_SHEAR, tool, (ignored, stack) -> {
+            drops.add(stack);
+        });
+        return drops;
+    }
+
+    @Override
+    public void shear(final ServerLevel level, final SoundSource soundSource, final ItemStack tool, final java.util.List<ItemStack> drops) {
+        // Paper end - custom shear drops
         level.playSound(null, this, SoundEvents.BOGGED_SHEAR, soundSource, 1.0F, 1.0F);
-        this.spawnShearedMushrooms(level, tool);
+        this.spawnShearedMushrooms(level, tool, drops); // Paper - custom shear drops
         this.setSheared(true);
     }
 
-    private void spawnShearedMushrooms(final ServerLevel level, final ItemStack tool) {
-        this.dropFromShearingLootTable(level, BuiltInLootTables.BOGGED_SHEAR, tool, (l, drop) -> this.spawnAtLocation(l, drop, this.getBbHeight()));
+    // Paper start - custom shear drops
+    private void spawnShearedMushrooms(final ServerLevel level, final ItemStack tool, java.util.List<ItemStack> drops) {
+        drops.forEach(drop -> this.spawnAtLocation(level, drop, this.getBbHeight()));
+        // Paper end - custom shear drops
     }
 
     @Override
