--- a/net/minecraft/world/entity/monster/skeleton/Skeleton.java
+++ b/net/minecraft/world/entity/monster/skeleton/Skeleton.java
@@ -17,7 +_,8 @@
 public class Skeleton extends AbstractSkeleton {
     private static final EntityDataAccessor<Boolean> DATA_STRAY_CONVERSION_ID = SynchedEntityData.defineId(Skeleton.class, EntityDataSerializers.BOOLEAN);
     private final ConversionTracker<AbstractSkeleton> freezingTracker = new ConversionTracker<>(
-        this, DATA_STRAY_CONVERSION_ID, () -> EntityTypes.STRAY, () -> 1048, () -> this.isInPowderSnow, "FreezingTime", 140, "StrayConversionTime", 300, null
+        this, DATA_STRAY_CONVERSION_ID, () -> EntityTypes.STRAY, () -> 1048, () -> this.isInPowderSnow, "FreezingTime", 140, "StrayConversionTime", 300, null,
+        org.bukkit.event.entity.EntityTransformEvent.TransformReason.FROZEN, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.FROZEN // Paper
     );
 
     public Skeleton(final EntityType<? extends Skeleton> type, final Level level) {
@@ -56,6 +_,18 @@
     @VisibleForTesting
     public void startFreezeConversion(final int time) {
         this.freezingTracker.startConversion(time);
+    }
+
+    public boolean isFreezeConverting() {
+        return this.freezingTracker.isConverting();
+    }
+
+    public void setFreezeConverting(final boolean freezeConverting) {
+        this.getEntityData().set(DATA_STRAY_CONVERSION_ID, freezeConverting);
+    }
+
+    public int getConversionTime() {
+        return this.freezingTracker.getConversionTime();
     }
 
     @Override
