From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/ServerTickRateManager.java b/net/minecraft/server/ServerTickRateManager.java
index 48c6077dff5c5b13e6db63ac6e88385b70298bc8..eeb2f88723b37bb3cada04bf3098dd46f0ed7316 100644
--- a/net/minecraft/server/ServerTickRateManager.java
+++ b/net/minecraft/server/ServerTickRateManager.java
@@ -15,6 +15,7 @@ public class ServerTickRateManager extends TickRateManager {
     private long scheduledCurrentSprintTicks = 0L;
     private boolean previousIsFrozen = false;
     private final MinecraftServer server;
+    private boolean silent; // Paper - silence feedback when API requests sprint
 
     public ServerTickRateManager(final MinecraftServer server) {
         this.server = server;
@@ -68,6 +69,13 @@ public class ServerTickRateManager extends TickRateManager {
     }
 
     public boolean requestGameToSprint(final int time) {
+        // Paper start - silence feedback when API requests sprint
+        return requestGameToSprint(time, false);
+    }
+
+    public boolean requestGameToSprint(final int time, final boolean silent) {
+        if (!isSprinting()) this.silent = silent;
+        // Paper end - silence feedback when API requests sprint
         boolean interrupted = this.remainingSprintTicks > 0L;
         this.sprintTimeSpend = 0L;
         this.scheduledCurrentSprintTicks = time;
@@ -86,9 +94,12 @@ public class ServerTickRateManager extends TickRateManager {
         );
         this.scheduledCurrentSprintTicks = 0L;
         this.sprintTimeSpend = 0L;
-        this.server
+        // Paper start - silence feedback when API requests sprint
+        if (!this.silent) this.server
             .createCommandSourceStack()
             .sendSuccess(() -> Component.translatable("commands.tick.sprint.report", ticksPerSecond, millisecondsPerTick), true);
+        this.silent = false;
+        // Paper end - silence feedback when API requests sprint
         this.remainingSprintTicks = 0L;
         this.setFrozen(this.previousIsFrozen);
         this.server.onTickRateChanged();
