From f632b9eb8d5c757eaffb596f5fe062258ab24e42 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 15:37:42 +0200
Subject: [PATCH] paper File Patches


diff --git a/net/minecraft/server/ServerFunctionLibrary.java b/net/minecraft/server/ServerFunctionLibrary.java
index 762ee210f4c562520160026f27dfd50c4a08524c..744d06948b5c79b3e24c05ea844b81c9da1dd8ae 100644
--- a/net/minecraft/server/ServerFunctionLibrary.java
+++ b/net/minecraft/server/ServerFunctionLibrary.java
@@ -111,7 +111,7 @@ public class ServerFunctionLibrary implements PreparableReloadListener {
                         return null;
                     }).join());
                     this.functions = newFunctions.build();
-                    this.tags = this.tagsLoader.build((Map<Identifier, List<TagLoader.EntryWithSource>>)data.getFirst());
+                    this.tags = this.tagsLoader.build((Map<Identifier, List<TagLoader.EntryWithSource>>)data.getFirst(), null); // Paper - command function tags are not implemented yet
                 },
                 reloadExecutor
             );
